{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# VCF to csv using scikit-allel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import allel"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Data read"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'samples': array(['200852'], dtype=object),\n",
       " 'calldata/GT': array([[[0, 1]],\n",
       " \n",
       "        [[0, 1]],\n",
       " \n",
       "        [[0, 1]],\n",
       " \n",
       "        [[0, 1]],\n",
       " \n",
       "        [[0, 1]],\n",
       " \n",
       "        [[0, 1]]], dtype=int8),\n",
       " 'variants/ALT': array([['T', '', ''],\n",
       "        ['A', '', ''],\n",
       "        ['C', '', ''],\n",
       "        ['A', '', ''],\n",
       "        ['A', '', ''],\n",
       "        ['T', '', '']], dtype=object),\n",
       " 'variants/CHROM': array(['chr1', 'chr1', 'chr1', 'chr1', 'chr1', 'chr1'], dtype=object),\n",
       " 'variants/FILTER_PASS': array([ True,  True,  True,  True,  True,  True]),\n",
       " 'variants/ID': array(['rs2691315', 'rs201535981', 'rs71260069', 'rs367730352',\n",
       "        'rs71267774', 'rs1396210256'], dtype=object),\n",
       " 'variants/POS': array([ 15820,  17385,  17697, 133129, 183629, 184267]),\n",
       " 'variants/QUAL': array([nan, nan, nan, nan, nan, nan], dtype=float32),\n",
       " 'variants/REF': array(['G', 'G', 'G', 'G', 'G', 'C'], dtype=object)}"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = allel.read_vcf(\"data/EE_sample/EE_sample.vcf\")\n",
    "data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Data to dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT_1</th>\n",
       "      <th>ALT_2</th>\n",
       "      <th>ALT_3</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER_PASS</th>\n",
       "      <th>FILTER_SB</th>\n",
       "      <th>...</th>\n",
       "      <th>gnomadGenomes_AC_Hemi_3</th>\n",
       "      <th>gnomadGenomes_AC_Hom_1</th>\n",
       "      <th>gnomadGenomes_AC_Hom_2</th>\n",
       "      <th>gnomadGenomes_AC_Hom_3</th>\n",
       "      <th>gnomadGenomes_AF_1</th>\n",
       "      <th>gnomadGenomes_AF_2</th>\n",
       "      <th>gnomadGenomes_AF_3</th>\n",
       "      <th>gnomadGenomes_AF_ethnic</th>\n",
       "      <th>hgvs</th>\n",
       "      <th>CSQ</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>15820</td>\n",
       "      <td>rs2691315</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>-1</td>\n",
       "      <td>18059</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>0.219648</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.093807</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17385</td>\n",
       "      <td>rs201535981</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>-1</td>\n",
       "      <td>21384</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>0.191644</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.213724</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17697</td>\n",
       "      <td>rs71260069</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>-1</td>\n",
       "      <td>17539</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>0.151133</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.178196</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>133129</td>\n",
       "      <td>rs367730352</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>-1</td>\n",
       "      <td>72154</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>0.547875</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.597389</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>183629</td>\n",
       "      <td>rs71267774</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>-1</td>\n",
       "      <td>14902</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>0.133296</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.220891</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A|downstream_gene_variant|MODIFIER|MIR6859-2|E...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>184267</td>\n",
       "      <td>rs1396210256</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>-1</td>\n",
       "      <td>887</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>0.006127</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.006310</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|MIR6859-2|E...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6 rows Ã— 113 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM     POS            ID REF ALT_1  ALT_2  ALT_3  QUAL  FILTER_PASS  \\\n",
       "0  chr1   15820     rs2691315   G     T    NaN    NaN   NaN         True   \n",
       "1  chr1   17385   rs201535981   G     A    NaN    NaN   NaN         True   \n",
       "2  chr1   17697    rs71260069   G     C    NaN    NaN   NaN         True   \n",
       "3  chr1  133129   rs367730352   G     A    NaN    NaN   NaN         True   \n",
       "4  chr1  183629    rs71267774   G     A    NaN    NaN   NaN         True   \n",
       "5  chr1  184267  rs1396210256   C     T    NaN    NaN   NaN         True   \n",
       "\n",
       "   FILTER_SB  ...  gnomadGenomes_AC_Hemi_3  gnomadGenomes_AC_Hom_1  \\\n",
       "0      False  ...                       -1                   18059   \n",
       "1      False  ...                       -1                   21384   \n",
       "2      False  ...                       -1                   17539   \n",
       "3      False  ...                       -1                   72154   \n",
       "4      False  ...                       -1                   14902   \n",
       "5      False  ...                       -1                     887   \n",
       "\n",
       "   gnomadGenomes_AC_Hom_2  gnomadGenomes_AC_Hom_3  gnomadGenomes_AF_1  \\\n",
       "0                      -1                      -1            0.219648   \n",
       "1                      -1                      -1            0.191644   \n",
       "2                      -1                      -1            0.151133   \n",
       "3                      -1                      -1            0.547875   \n",
       "4                      -1                      -1            0.133296   \n",
       "5                      -1                      -1            0.006127   \n",
       "\n",
       "   gnomadGenomes_AF_2  gnomadGenomes_AF_3 gnomadGenomes_AF_ethnic hgvs  \\\n",
       "0                 NaN                 NaN                0.093807  NaN   \n",
       "1                 NaN                 NaN                0.213724  NaN   \n",
       "2                 NaN                 NaN                0.178196  NaN   \n",
       "3                 NaN                 NaN                0.597389  NaN   \n",
       "4                 NaN                 NaN                0.220891  NaN   \n",
       "5                 NaN                 NaN                0.006310  NaN   \n",
       "\n",
       "                                                 CSQ  \n",
       "0  T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...  \n",
       "1  A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...  \n",
       "2  C|downstream_gene_variant|MODIFIER|DDX11L1|ENS...  \n",
       "3  A|non_coding_transcript_exon_variant|MODIFIER|...  \n",
       "4  A|downstream_gene_variant|MODIFIER|MIR6859-2|E...  \n",
       "5  T|downstream_gene_variant|MODIFIER|MIR6859-2|E...  \n",
       "\n",
       "[6 rows x 113 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = allel.vcf_to_dataframe(\"data/EE_sample/EE_sample.vcf\", [\"CHROM\", \"POS\", \"ID\", \"REF\", \"ALT\", \"QUAL\", \"FILTER\", \"INFO\"])\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Data to csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "allel.vcf_to_csv(\"data/EE_sample/EE_sample.vcf\", \"data/EE_sample/EE_sample_allel.csv\", [\"CHROM\", \"POS\", \"ID\", \"REF\", \"ALT\", \"QUAL\", \"FILTER\", \"INFO\"])"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "pathogen",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
