{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_genotype = pd.read_csv(\"data/EE_015/EE_015_genotype.csv.gz\", sep=\";\", compression=\"gzip\")\n",
    "df_default = pd.read_csv(\"data/EE_015/EE_015_default.csv.gz\", sep=\";\", compression=\"gzip\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.concat([df_default.drop(\"Unnamed: 0\", axis=1), df_genotype.drop(\"Unnamed: 0\", axis=1)], axis=1, ignore_index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "object_columns = df.select_dtypes(object).columns\n",
    "\n",
    "i = 25\n",
    "for col in object_columns[i:i+1]:\n",
    "    # print(f\"Column {col}: {df[col].unique()}\")\n",
    "    print(col)\n",
    "    unique = df[col].unique()\n",
    "    print(len(unique))\n",
    "    # print([x for x in unique if len(x.split(\";\"))>1])\n",
    "\n",
    "    for unique_val in unique:\n",
    "        print(unique_val)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POS</th>\n",
       "      <th>ACMG_score</th>\n",
       "      <th>AMP_score</th>\n",
       "      <th>AMP_total_samples</th>\n",
       "      <th>ClinVarClass</th>\n",
       "      <th>ClinVarDisease</th>\n",
       "      <th>DANN_score</th>\n",
       "      <th>DP</th>\n",
       "      <th>ECNT</th>\n",
       "      <th>GERMQ</th>\n",
       "      <th>...</th>\n",
       "      <th>gnomadExomes_AF</th>\n",
       "      <th>gnomadExomes_AF_ethnic</th>\n",
       "      <th>gnomadExomes_AN</th>\n",
       "      <th>gnomadGenomesAC</th>\n",
       "      <th>gnomadGenomesAN</th>\n",
       "      <th>gnomadGenomesEthnic_AC_Hom</th>\n",
       "      <th>gnomadGenomes_AC_Hemi</th>\n",
       "      <th>gnomadGenomes_AC_Hom</th>\n",
       "      <th>gnomadGenomes_AF</th>\n",
       "      <th>gnomadGenomes_AF_ethnic</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>15820</td>\n",
       "      <td>0.193</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "      <td>1</td>\n",
       "      <td>55</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>18059</td>\n",
       "      <td>82218</td>\n",
       "      <td>233</td>\n",
       "      <td>0</td>\n",
       "      <td>18059</td>\n",
       "      <td>0.219648</td>\n",
       "      <td>0.093807</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>17385</td>\n",
       "      <td>-0.047</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>70</td>\n",
       "      <td>1</td>\n",
       "      <td>37</td>\n",
       "      <td>...</td>\n",
       "      <td>0.173001</td>\n",
       "      <td>0.199722</td>\n",
       "      <td>53936</td>\n",
       "      <td>21384</td>\n",
       "      <td>111582</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>21384</td>\n",
       "      <td>0.191644</td>\n",
       "      <td>0.213724</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>17697</td>\n",
       "      <td>0.193</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>1</td>\n",
       "      <td>67</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>17539</td>\n",
       "      <td>116050</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>17539</td>\n",
       "      <td>0.151133</td>\n",
       "      <td>0.178196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>133129</td>\n",
       "      <td>-0.047</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>1</td>\n",
       "      <td>92</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>72154</td>\n",
       "      <td>131698</td>\n",
       "      <td>7468</td>\n",
       "      <td>0</td>\n",
       "      <td>72154</td>\n",
       "      <td>0.547875</td>\n",
       "      <td>0.597389</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>183629</td>\n",
       "      <td>0.193</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>139</td>\n",
       "      <td>4</td>\n",
       "      <td>93</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>14902</td>\n",
       "      <td>111796</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>14902</td>\n",
       "      <td>0.133296</td>\n",
       "      <td>0.220891</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102512</th>\n",
       "      <td>56856512</td>\n",
       "      <td>-0.047</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>32</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102513</th>\n",
       "      <td>56856517</td>\n",
       "      <td>-0.047</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>5</td>\n",
       "      <td>32</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102514</th>\n",
       "      <td>56856554</td>\n",
       "      <td>0.193</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>5</td>\n",
       "      <td>33</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102515</th>\n",
       "      <td>56856581</td>\n",
       "      <td>-0.047</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>5</td>\n",
       "      <td>33</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>135</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.014815</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102516</th>\n",
       "      <td>56856587</td>\n",
       "      <td>-0.047</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>5</td>\n",
       "      <td>80</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>123</td>\n",
       "      <td>123</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>123</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>102517 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             POS  ACMG_score  AMP_score  AMP_total_samples  ClinVarClass  \\\n",
       "0          15820       0.193          0                  0             0   \n",
       "1          17385      -0.047          0                  0             0   \n",
       "2          17697       0.193          0                  0             0   \n",
       "3         133129      -0.047          0                  0             0   \n",
       "4         183629       0.193          0                  0             0   \n",
       "...          ...         ...        ...                ...           ...   \n",
       "102512  56856512      -0.047          0                  0             0   \n",
       "102513  56856517      -0.047          0                  0             0   \n",
       "102514  56856554       0.193          0                  0             0   \n",
       "102515  56856581      -0.047          0                  0             0   \n",
       "102516  56856587      -0.047          0                  0             0   \n",
       "\n",
       "        ClinVarDisease  DANN_score   DP  ECNT  GERMQ  ...  gnomadExomes_AF  \\\n",
       "0                    0           0   45     1     55  ...         0.000000   \n",
       "1                    0           0   70     1     37  ...         0.173001   \n",
       "2                    0           0   49     1     67  ...         0.000000   \n",
       "3                    0           0   22     1     92  ...         0.000000   \n",
       "4                    0           0  139     4     93  ...         0.000000   \n",
       "...                ...         ...  ...   ...    ...  ...              ...   \n",
       "102512               0           0   11     5     32  ...         0.000000   \n",
       "102513               0           0   14     5     32  ...         0.000000   \n",
       "102514               0           0   17     5     33  ...         0.000000   \n",
       "102515               0           0   16     5     33  ...         0.000000   \n",
       "102516               0           0   16     5     80  ...         0.000000   \n",
       "\n",
       "        gnomadExomes_AF_ethnic  gnomadExomes_AN  gnomadGenomesAC  \\\n",
       "0                     0.000000                0            18059   \n",
       "1                     0.199722            53936            21384   \n",
       "2                     0.000000                0            17539   \n",
       "3                     0.000000                0            72154   \n",
       "4                     0.000000                0            14902   \n",
       "...                        ...              ...              ...   \n",
       "102512                0.000000                0                0   \n",
       "102513                0.000000                0                0   \n",
       "102514                0.000000                0                0   \n",
       "102515                0.000000                0                2   \n",
       "102516                0.000000                0              123   \n",
       "\n",
       "        gnomadGenomesAN  gnomadGenomesEthnic_AC_Hom  gnomadGenomes_AC_Hemi  \\\n",
       "0                 82218                         233                      0   \n",
       "1                111582                           1                      0   \n",
       "2                116050                           2                      0   \n",
       "3                131698                        7468                      0   \n",
       "4                111796                           1                      0   \n",
       "...                 ...                         ...                    ...   \n",
       "102512                0                           0                      0   \n",
       "102513                0                           0                      0   \n",
       "102514                0                           0                      0   \n",
       "102515              135                           0                      0   \n",
       "102516              123                           0                      0   \n",
       "\n",
       "        gnomadGenomes_AC_Hom  gnomadGenomes_AF  gnomadGenomes_AF_ethnic  \n",
       "0                      18059          0.219648                 0.093807  \n",
       "1                      21384          0.191644                 0.213724  \n",
       "2                      17539          0.151133                 0.178196  \n",
       "3                      72154          0.547875                 0.597389  \n",
       "4                      14902          0.133296                 0.220891  \n",
       "...                      ...               ...                      ...  \n",
       "102512                     0          0.000000                 0.000000  \n",
       "102513                     0          0.000000                 0.000000  \n",
       "102514                     0          0.000000                 0.000000  \n",
       "102515                     2          0.014815                 0.000000  \n",
       "102516                   123          1.000000                 1.000000  \n",
       "\n",
       "[102517 rows x 34 columns]"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[[c for c in df.columns if c not in df.select_dtypes(object).columns]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['PASS', 'clustered_events', 'haplotype']"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# sklearn.preprocessing.OneHotEncoder\n",
    "\n",
    "filters = [\"PASS\", \"SB\", \"UM\", \"clustered_events\", \"fragment\", \"germline\", \"haplotype\", \"multiallelic\", \"slippage\"]\n",
    "filter_cols = [\"FILTER_\" + f for f in filters]\n",
    "df[\"FILTER\"].astype(str).str.split(\";\").iloc[-6]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'0',\n",
       " \"3'flank\",\n",
       " \"3'utr\",\n",
       " \"5'flank\",\n",
       " \"5'utr\",\n",
       " 'NMD',\n",
       " 'coding',\n",
       " 'intronic',\n",
       " 'non-coding%40exon',\n",
       " 'splicing',\n",
       " 'splicing-ACMG'}"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "functions = [\"0\", \"NMD\", \"3'utr\", \"5'utr\", \"3'flank\", \"5'flank\", \"coding\", \"non-coding%40exon\", \"intronic\", \"splicing\", \"splicing-ACMG\"]\n",
    "function_cols = [\"function_\" + f for f in filters]\n",
    "\n",
    "uniq = set([])\n",
    "for x in df[\"function\"].astype(str).str.split(\",\"):\n",
    "    for y in x:\n",
    "        if y not in uniq:\n",
    "            uniq.add(y)\n",
    "\n",
    "uniq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['#CHROM', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'ACMG_class',\n",
       "       'ACMG_coding_impact', 'ACMG_gene', 'ACMG_rules', 'ACMG_transcript',\n",
       "       'AMP_matches', 'AMP_rules', 'AMP_tier', 'AS_FilterStatus',\n",
       "       'AS_SB_TABLE', 'CGDinheritance', 'Gene', 'MutationTaster_pred',\n",
       "       'MutationTaster_score', 'RU', 'SIFT_score', 'coding_impact',\n",
       "       'cosmicFathMMPrediction', 'function', 'hgvs'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "object_columns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Object columns:\n",
    "- #CHROM - no arrays\n",
    "- ID - id\n",
    "- REF, ALT - no arrays\n",
    "- QUAL - binary 1 or .\n",
    "- FILTER - [\"PASS\", \"SB\", \"UM\", \"clustered_events\", \"fragment\", \"germline\", \"haplotype\", \"multiallelic\", \"slippage\"]\n",
    "- ACMG_class - no arrays\n",
    "- ACMG_coding_impact - no arrays\n",
    "- ACMG_gene - no arrays\n",
    "- ACMG_rules - https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4544753/ I will leave this data unseparated for now \n",
    "- ACMG_transcript - no arrays\n",
    "- AMP_matches - no arrays\n",
    "- AMP_rules - https://varsome.com/about/resources/somatic-implementation/ I will leave this data unseparated for now\n",
    "- AMP_tier - no arrays\n",
    "- AS_FilterStatus - https://docs.varsome.com/en/vcf-attributes-explained I will leave this data unseparated for now\n",
    "- AS_SB_TABLE - https://docs.varsome.com/en/vcf-attributes-explained I will leave this data unseparated for now\n",
    "- CGDinheritance - I couldn't find good info, looks separable, left for now\n",
    "- Gene - gene symbols - might be separable, maybe not worth is if a lot of unique values, check relation to ACMG (target) class\n",
    "- MutationTaster_pred - I don't know which of the symbols represent which type and why there is a number of them, %3B is ';', left unseparated for now. https://www.mutationtaster.org/info/documentation.html. MutationTaster predicts an alteration as one of four possible types:\n",
    "    - disease causing - i.e. probably deleterious\n",
    "    - disease causing automatic - i.e. known to be deleterious, see section dbSNP / TGP / ClinVar / HGMD for details\n",
    "    - polymorphism - i.e. probably harmless\n",
    "    - polymorphism automatic - i.e. known to be harmless, see section dbSNP / TGP / ClinVar / HGMD for details\n",
    "- MutationTaster_score - respective score (certainty?)\n",
    "- RU - no arrays\n",
    "- SIFT_score - https://ionreporter.thermofisher.com/ionreporter/help/GUID-2097F236-C8A2-4E67-862D-0FB5875979AC.html, separable but I don't know how\n",
    "- coding_impact - https://www.ebi.ac.uk/training/online/courses/human-genetic-variation-introduction/what-is-genetic-variation/what-effect-do-variants-in-coding-regions-have/ seems like there are 3 important classes, couldn't find info on the other, leaving unseparated for now\n",
    "- cosmicFathMMPrediction - no arrays - also target\n",
    "- function - https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2872880/ - [\"0\", \"NMD\", \"3'utr\", \"5'utr\", \"3'flank\", \"5'flank\", \"coding\", \"non-coding%40exon\", \"intronic\", \"splicing\", \"splicing-ACMG\"]\n",
    "- hgvs - https://varnomen.hgvs.org/recommendations/general/ - again I don't know how to separate this well"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "pathogen",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
