{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "BESg3pXMj-z8"
      },
      "source": [
        "# VCF_Reading_Tutorial"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "ftNV_7O7klxP"
      },
      "source": [
        "Hi guys, this is my quick tutorial how to read vcf data. I am by no means an expert, so you are welcome to look up things on your own and expand/change this notebook. Regardless, for the purposes of this notebook I will be reading just one of the provided files, note that you do not need to extract the file and can instead use the .gz compressed version when working with the chosen library."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "dUt3ywuSnBww"
      },
      "source": [
        "## Installing pyvcf"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "713gdONGnFVz"
      },
      "source": [
        "For the purpose of reading vcf files I have chosen pyvcf library and as such, this is what the tutorial is centered around. Other libraries are available so if you find something better suited feel free to use it, but for now let's get the basics of pyvcf out of the way.\n",
        "\n",
        "Installing pyvcf is unironically one of the hardest parts of using it. For the record, I used Google Colab, so your experience when using some other IDE can be different. Also if you are working on it locally, please use a virtual environment. Regardless, first try a normal install, for example through standard pip command:\n",
        "* pip install pyvcf\n",
        "\n",
        "If it does not work(happens often), then use these commands:\n",
        "* pip install \"setuptools<58\" --upgrade\n",
        "* pip uninstall pyvcf\n",
        "* pip install pyvcf\n",
        "\n",
        "If you are working on a notebook environment then you can copy the corresponding cells below. If it still does not work, then upgrade pip and install necessary tools, then rerun the previous commands like so:\n",
        "* pip install --upgrade pip\n",
        "* !apt-get install -y python-dev libssl-dev libffi-dev libxml2 libxml2-dev libxslt1.1 libxslt1-dev zlib1g-dev\n",
        "* pip install \"setuptools<58\" --upgrade\n",
        "* pip uninstall pyvcf\n",
        "* pip install pyvcf"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 37,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "F9hPeTvHX0a-",
        "outputId": "0f46b7e0-7aef-42aa-cf6b-1bd69e91654a"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Requirement already satisfied: pip in /usr/local/lib/python3.10/dist-packages (23.3)\n",
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
            "\u001b[0m"
          ]
        }
      ],
      "source": [
        "!pip install --upgrade pip"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 38,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "LwcXpIoUZVA5",
        "outputId": "afccfd00-531f-4c75-f1bb-8bcd1b0158fe"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "Package python-dev is not available, but is referred to by another package.\n",
            "This may mean that the package is missing, has been obsoleted, or\n",
            "is only available from another source\n",
            "However the following packages replace it:\n",
            "  python2-dev python2 python-dev-is-python3\n",
            "\n",
            "E: Package 'python-dev' has no installation candidate\n"
          ]
        }
      ],
      "source": [
        "!apt-get install -y python-dev libssl-dev libffi-dev libxml2 libxml2-dev libxslt1.1 libxslt1-dev zlib1g-dev"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 39,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "LDM44okeam71",
        "outputId": "2be6c194-4b78-40fa-b140-8fc7ebd634d2"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Requirement already satisfied: setuptools<58 in /usr/local/lib/python3.10/dist-packages (57.5.0)\n",
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
            "\u001b[0mFound existing installation: PyVCF 0.6.8\n",
            "Uninstalling PyVCF-0.6.8:\n",
            "  Would remove:\n",
            "    /usr/local/bin/vcf_filter.py\n",
            "    /usr/local/bin/vcf_melt\n",
            "    /usr/local/bin/vcf_sample_filter.py\n",
            "    /usr/local/lib/python3.10/dist-packages/PyVCF-0.6.8.dist-info/*\n",
            "    /usr/local/lib/python3.10/dist-packages/vcf/*\n",
            "Proceed (Y/n)? \u001b[31mERROR: Operation cancelled by user\u001b[0m\u001b[31m\n",
            "\u001b[0m^C\n",
            "Requirement already satisfied: pyvcf in /usr/local/lib/python3.10/dist-packages (0.6.8)\n",
            "Requirement already satisfied: setuptools in /usr/local/lib/python3.10/dist-packages (from pyvcf) (57.5.0)\n",
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
            "\u001b[0m"
          ]
        }
      ],
      "source": [
        "!pip install \"setuptools<58\" --upgrade\n",
        "!pip uninstall pyvcf\n",
        "!pip install pyvcf"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "#### Further troubleshooting\n",
        "\n",
        "If the issue still presists (probably windows-related) try installing the package through conda"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "%%cmd\n",
        "\n",
        "C:\\Users\\USER\\anaconda3\\Scripts\\activate base\n",
        "conda activate pathogen\n",
        "conda install pyvcf"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "If this doesn't work try:"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "%%cmd\n",
        "\n",
        "C:\\Users\\USER\\anaconda3\\Scripts\\activate base\n",
        "conda activate pathogen\n",
        "\n",
        "conda config --add channels conda-forge\n",
        "conda config --set channel_priority strict\n",
        "conda install pyvcf"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Which finally worked for me"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "NgWGp5NtpQJ6"
      },
      "source": [
        "## Using pyvcf"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "ZOG-dgghpV0m"
      },
      "source": [
        "So I have never before worked with vcf files, but from what I have learned they have a certain \"built-in\"(that is not the official term but I do not know how else to call it) fields that are subject to certain constraints and standards and some \"custom\" fields that the user can define when writing the file. Regardless, let's focus on the built-in fields first. Below is the code for reading first 10 data points and showing information related to them."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "bpkV1q8abIal",
        "outputId": "5c255a8a-1bbf-4093-8aba-537bc816e5f4"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "CHROM: chr1\n",
            "POS: 15820\n",
            "ID: rs2691315\n",
            "REF: G\n",
            "ALT: [T]\n",
            "QUAL: None\n",
            "FILTER: []\n",
            "INFO: {'ACMG_class': ['Uncertain%40Significance'], 'ACMG_coding_impact': ['non%40coding'], 'ACMG_rules': ['BP4'], 'ACMG_score': [0.193], 'AMP_matches': ['V'], 'AMP_rules': ['Freq_II%2CType_IV%2CPred_IV'], 'AMP_tier': ['Tier%40IV'], 'AMP_total_samples': [0], 'AS_FilterStatus': 'SITE', 'AS_SB_TABLE': '12%2C23|7%2C2', 'DP': 45, 'ECNT': 1, 'GERMQ': 55, 'Gene': ['WASH7P'], 'MBQ': [29], 'MFRL': [219], 'MMQ': [34], 'MPOS': [30], 'POPAF': [7.3], 'TLOD': [10.9], 'function': ['non-coding%40exon'], 'gnomadGenomesAC': [18059], 'gnomadGenomesAN': [82218], 'gnomadGenomesEthnic_AC_Hom': [233], 'gnomadGenomes_AC_Hom': [18059], 'gnomadGenomes_AF': [0.219648], 'gnomadGenomes_AF_ethnic': [0.0938074], 'CSQ': ['T|downstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000450305.2|transcribed_unprocessed_pseudogene||||||||||rs2691315|2150|1||HGNC|HGNC:37102|YES||||||||||||Ensembl||G|G||||||0.4105|0.4849|0.2939|0.6052|0.2714|0.3354||||||||||0.2196|0.4324|0.1146|0.1638|0.1285|0.06176|0.06299|0.2151|0.09381|0.2235|0.1329|||||||||||||16.02|1.543896||||||||||||', 'T|downstream_gene_variant|MODIFIER|DDX11L2|ENSG00000290825|Transcript|ENST00000456328.2|lncRNA||||||||||rs2691315|1411|1||EntrezGene||YES|||1|||||||||Ensembl||G|G||||||0.4105|0.4849|0.2939|0.6052|0.2714|0.3354||||||||||0.2196|0.4324|0.1146|0.1638|0.1285|0.06176|0.06299|0.2151|0.09381|0.2235|0.1329|||||||||||||16.02|1.543896||||||||||||', 'T|non_coding_transcript_exon_variant|MODIFIER|WASH7P|ENSG00000227232|Transcript|ENST00000488147.1|unprocessed_pseudogene|9/11||ENST00000488147.1:n.1195C>A||1195|||||rs2691315||-1||HGNC|HGNC:38034|YES||||||||||||Ensembl||G|G||||||0.4105|0.4849|0.2939|0.6052|0.2714|0.3354||||||||||0.2196|0.4324|0.1146|0.1638|0.1285|0.06176|0.06299|0.2151|0.09381|0.2235|0.1329|||||||||||||16.02|1.543896||||||||||||', 'T|downstream_gene_variant|MODIFIER|MIR6859-1|ENSG00000278267|Transcript|ENST00000619216.1|miRNA||||||||||rs2691315|1549|-1||HGNC|HGNC:50039|YES||||||||||||Ensembl||G|G||||||0.4105|0.4849|0.2939|0.6052|0.2714|0.3354||||||||||0.2196|0.4324|0.1146|0.1638|0.1285|0.06176|0.06299|0.2151|0.09381|0.2235|0.1329|||||||||||||16.02|1.543896||||||||||||', 'T|non_coding_transcript_exon_variant|MODIFIER|WASH7P|653635|Transcript|NR_024540.1|transcribed_pseudogene|9/11||NR_024540.1:n.1208C>A||1208|||||rs2691315||-1||EntrezGene|HGNC:38034|YES||||||||||||RefSeq||G|G|OK|||||0.4105|0.4849|0.2939|0.6052|0.2714|0.3354||||||||||0.2196|0.4324|0.1146|0.1638|0.1285|0.06176|0.06299|0.2151|0.09381|0.2235|0.1329|||||||||||||16.02|1.543896||||||||||||', 'T|downstream_gene_variant|MODIFIER|DDX11L1|100287102|Transcript|NR_046018.2|transcribed_pseudogene||||||||||rs2691315|1411|1||EntrezGene|HGNC:37102|YES||||||||||||RefSeq||G|G||||||0.4105|0.4849|0.2939|0.6052|0.2714|0.3354||||||||||0.2196|0.4324|0.1146|0.1638|0.1285|0.06176|0.06299|0.2151|0.09381|0.2235|0.1329|||||||||||||16.02|1.543896||||||||||||', 'T|downstream_gene_variant|MODIFIER|MIR6859-1|102466751|Transcript|NR_106918.1|miRNA||||||||||rs2691315|1549|-1||EntrezGene|HGNC:50039|YES||||||||||||RefSeq||G|G||||||0.4105|0.4849|0.2939|0.6052|0.2714|0.3354||||||||||0.2196|0.4324|0.1146|0.1638|0.1285|0.06176|0.06299|0.2151|0.09381|0.2235|0.1329|||||||||||||16.02|1.543896||||||||||||', 'T|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00000344266|CTCF_binding_site||||||||||rs2691315|||||||||||||||||||||||||||0.4105|0.4849|0.2939|0.6052|0.2714|0.3354||||||||||0.2196|0.4324|0.1146|0.1638|0.1285|0.06176|0.06299|0.2151|0.09381|0.2235|0.1329|||||||||||||16.02|1.543896||||||||||||', 'T|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00001164745|enhancer||||||||||rs2691315|||||||||||||||||||||||||||0.4105|0.4849|0.2939|0.6052|0.2714|0.3354||||||||||0.2196|0.4324|0.1146|0.1638|0.1285|0.06176|0.06299|0.2151|0.09381|0.2235|0.1329|||||||||||||16.02|1.543896||||||||||||', 'T|TF_binding_site_variant|MODIFIER|||MotifFeature|ENSM00000173493|||||||||||rs2691315||1|||||||||||||||||||||||||0.4105|0.4849|0.2939|0.6052|0.2714|0.3354||||||||||0.2196|0.4324|0.1146|0.1638|0.1285|0.06176|0.06299|0.2151|0.09381|0.2235|0.1329||||||ENSPFM0042|17|N|-0.005|CTCF|||16.02|1.543896||||||||||||']}\n",
            "FORMAT: GT:AB:AD:AF:DP:F1R2:F2R1:SB\n",
            "Sample 200852: 0/1\n",
            "\n",
            "\n",
            "CHROM: chr1\n",
            "POS: 17385\n",
            "ID: rs201535981\n",
            "REF: G\n",
            "ALT: [A]\n",
            "QUAL: None\n",
            "FILTER: []\n",
            "INFO: {'ACMG_class': ['Uncertain%40Significance'], 'ACMG_coding_impact': ['non%40coding'], 'ACMG_rules': ['BP4'], 'ACMG_score': [-0.047], 'AMP_matches': [''], 'AMP_rules': ['Type_IV%2CPred_IV'], 'AMP_tier': ['Tier%40IV'], 'AMP_total_samples': [0], 'AS_FilterStatus': 'SITE', 'AS_SB_TABLE': '37%2C9|17%2C6', 'DP': 70, 'ECNT': 1, 'GERMQ': 37, 'Gene': ['MIR6859-1', 'WASH7P'], 'MBQ': [29], 'MFRL': [293], 'MMQ': [40], 'MPOS': [19], 'POPAF': [7.3], 'TLOD': [45.3], 'function': ['non-coding%40exon', 'intronic'], 'gnomadExomesEthnic_AC_Hom': [2], 'gnomadExomes_AC': [9331], 'gnomadExomes_AC_Hom': [9331], 'gnomadExomes_AF': [0.173001], 'gnomadExomes_AF_ethnic': [0.199722], 'gnomadExomes_AN': [53936], 'gnomadGenomesAC': [21384], 'gnomadGenomesAN': [111582], 'gnomadGenomesEthnic_AC_Hom': [1], 'gnomadGenomes_AC_Hom': [21384], 'gnomadGenomes_AF': [0.191644], 'gnomadGenomes_AF_ethnic': [0.213724], 'CSQ': ['A|downstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000450305.2|transcribed_unprocessed_pseudogene||||||||||rs201535981|3715|1||HGNC|HGNC:37102|YES||||||||||||Ensembl||G|G||||||||||||0.173|0.1443|0.2215|0.2532|0.02807|0.207|0.1997|0.1992|0.1629|0.1916|0.1547|0.2446|0.2082|0.261|0.03913|0.2582|0.2822|0.2137|0.2152|0.1667|||||||||||||7.934|0.644040||||||||||||', 'A|downstream_gene_variant|MODIFIER|DDX11L2|ENSG00000290825|Transcript|ENST00000456328.2|lncRNA||||||||||rs201535981|2976|1||EntrezGene||YES|||1|||||||||Ensembl||G|G||||||||||||0.173|0.1443|0.2215|0.2532|0.02807|0.207|0.1997|0.1992|0.1629|0.1916|0.1547|0.2446|0.2082|0.261|0.03913|0.2582|0.2822|0.2137|0.2152|0.1667|||||||||||||7.934|0.644040||||||||||||', 'A|splice_polypyrimidine_tract_variant&intron_variant&non_coding_transcript_variant|LOW|WASH7P|ENSG00000227232|Transcript|ENST00000488147.1|unprocessed_pseudogene||5/10|ENST00000488147.1:n.575-17C>T|||||||rs201535981||-1||HGNC|HGNC:38034|YES||||||||||||Ensembl||G|G||||||||||||0.173|0.1443|0.2215|0.2532|0.02807|0.207|0.1997|0.1992|0.1629|0.1916|0.1547|0.2446|0.2082|0.261|0.03913|0.2582|0.2822|0.2137|0.2152|0.1667|||||||||||||7.934|0.644040||||||||||||', 'A|mature_miRNA_variant|MODIFIER|MIR6859-1|ENSG00000278267|Transcript|ENST00000619216.1|miRNA|1/1||ENST00000619216.1:n.52C>T||52|||||rs201535981||-1||HGNC|HGNC:50039|YES||||||||||||Ensembl||G|G||||||||||||0.173|0.1443|0.2215|0.2532|0.02807|0.207|0.1997|0.1992|0.1629|0.1916|0.1547|0.2446|0.2082|0.261|0.03913|0.2582|0.2822|0.2137|0.2152|0.1667|||||||||||||7.934|0.644040||||||||||||', 'A|splice_polypyrimidine_tract_variant&intron_variant&non_coding_transcript_variant|LOW|WASH7P|653635|Transcript|NR_024540.1|transcribed_pseudogene||5/10|NR_024540.1:n.588-17C>T|||||||rs201535981||-1||EntrezGene|HGNC:38034|YES||||||||||||RefSeq||G|G|OK|||||||||||0.173|0.1443|0.2215|0.2532|0.02807|0.207|0.1997|0.1992|0.1629|0.1916|0.1547|0.2446|0.2082|0.261|0.03913|0.2582|0.2822|0.2137|0.2152|0.1667|||||||||||||7.934|0.644040||||||||||||', 'A|downstream_gene_variant|MODIFIER|DDX11L1|100287102|Transcript|NR_046018.2|transcribed_pseudogene||||||||||rs201535981|2976|1||EntrezGene|HGNC:37102|YES||||||||||||RefSeq||G|G||||||||||||0.173|0.1443|0.2215|0.2532|0.02807|0.207|0.1997|0.1992|0.1629|0.1916|0.1547|0.2446|0.2082|0.261|0.03913|0.2582|0.2822|0.2137|0.2152|0.1667|||||||||||||7.934|0.644040||||||||||||', 'A|non_coding_transcript_exon_variant|MODIFIER|MIR6859-1|102466751|Transcript|NR_106918.1|miRNA|1/1||NR_106918.1:n.52C>T||52|||||rs201535981||-1||EntrezGene|HGNC:50039|YES||||||||||||RefSeq||G|G||||||||||||0.173|0.1443|0.2215|0.2532|0.02807|0.207|0.1997|0.1992|0.1629|0.1916|0.1547|0.2446|0.2082|0.261|0.03913|0.2582|0.2822|0.2137|0.2152|0.1667|||||||||||||7.934|0.644040||||||||||||']}\n",
            "FORMAT: GT:AB:AD:AF:DP:F1R2:F2R1:SB\n",
            "Sample 200852: 0/1\n",
            "\n",
            "\n",
            "CHROM: chr1\n",
            "POS: 17697\n",
            "ID: rs71260069\n",
            "REF: G\n",
            "ALT: [C]\n",
            "QUAL: None\n",
            "FILTER: []\n",
            "INFO: {'ACMG_class': ['Benign'], 'ACMG_coding_impact': ['non%40coding'], 'ACMG_rules': ['BA1%2CBP4'], 'ACMG_score': [0.193], 'AMP_matches': ['E%2CV'], 'AMP_rules': ['Freq_II%2CType_IV%2CPred_IV'], 'AMP_tier': ['Tier%40IV'], 'AMP_total_samples': [0], 'AS_FilterStatus': 'SITE', 'AS_SB_TABLE': '0%2C39|0%2C8', 'DP': 49, 'ECNT': 1, 'GERMQ': 67, 'Gene': ['MIR6859-1', 'WASH7P'], 'MBQ': [29], 'MFRL': [331], 'MMQ': [40], 'MPOS': [17], 'POPAF': [7.3], 'TLOD': [13.2], 'function': [\"5'flank\", 'non-coding%40exon'], 'gnomadGenomesAC': [17539], 'gnomadGenomesAN': [116050], 'gnomadGenomesEthnic_AC_Hom': [2], 'gnomadGenomes_AC_Hom': [17539], 'gnomadGenomes_AF': [0.151133], 'gnomadGenomes_AF_ethnic': [0.178196], 'CSQ': ['C|downstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000450305.2|transcribed_unprocessed_pseudogene||||||||||rs71260069|4027|1||HGNC|HGNC:37102|YES||||||||||||Ensembl||G|G|||||||||||||||||||||0.1511|0.1238|0.0654|0.1197|0.1641|0.02216|0.2241|0.196|0.1782|0.1404|0.1298|||||||||||||12.16|1.043489||||||||||||', 'C|downstream_gene_variant|MODIFIER|DDX11L2|ENSG00000290825|Transcript|ENST00000456328.2|lncRNA||||||||||rs71260069|3288|1||EntrezGene||YES|||1|||||||||Ensembl||G|G|||||||||||||||||||||0.1511|0.1238|0.0654|0.1197|0.1641|0.02216|0.2241|0.196|0.1782|0.1404|0.1298|||||||||||||12.16|1.043489||||||||||||', 'C|non_coding_transcript_exon_variant|MODIFIER|WASH7P|ENSG00000227232|Transcript|ENST00000488147.1|unprocessed_pseudogene|5/11||ENST00000488147.1:n.483C>G||483|||||rs71260069||-1||HGNC|HGNC:38034|YES||||||||||||Ensembl||G|G|||||||||||||||||||||0.1511|0.1238|0.0654|0.1197|0.1641|0.02216|0.2241|0.196|0.1782|0.1404|0.1298|||||||||||||12.16|1.043489||||||||||||', 'C|upstream_gene_variant|MODIFIER|MIR6859-1|ENSG00000278267|Transcript|ENST00000619216.1|miRNA||||||||||rs71260069|261|-1||HGNC|HGNC:50039|YES||||||||||||Ensembl||G|G|||||||||||||||||||||0.1511|0.1238|0.0654|0.1197|0.1641|0.02216|0.2241|0.196|0.1782|0.1404|0.1298|||||||||||||12.16|1.043489||||||||||||', 'C|non_coding_transcript_exon_variant|MODIFIER|WASH7P|653635|Transcript|NR_024540.1|transcribed_pseudogene|5/11||NR_024540.1:n.496C>G||496|||||rs71260069||-1||EntrezGene|HGNC:38034|YES||||||||||||RefSeq||G|G|OK||||||||||||||||||||0.1511|0.1238|0.0654|0.1197|0.1641|0.02216|0.2241|0.196|0.1782|0.1404|0.1298|||||||||||||12.16|1.043489||||||||||||', 'C|downstream_gene_variant|MODIFIER|DDX11L1|100287102|Transcript|NR_046018.2|transcribed_pseudogene||||||||||rs71260069|3288|1||EntrezGene|HGNC:37102|YES||||||||||||RefSeq||G|G|||||||||||||||||||||0.1511|0.1238|0.0654|0.1197|0.1641|0.02216|0.2241|0.196|0.1782|0.1404|0.1298|||||||||||||12.16|1.043489||||||||||||', 'C|upstream_gene_variant|MODIFIER|MIR6859-1|102466751|Transcript|NR_106918.1|miRNA||||||||||rs71260069|261|-1||EntrezGene|HGNC:50039|YES||||||||||||RefSeq||G|G|||||||||||||||||||||0.1511|0.1238|0.0654|0.1197|0.1641|0.02216|0.2241|0.196|0.1782|0.1404|0.1298|||||||||||||12.16|1.043489||||||||||||']}\n",
            "FORMAT: GT:AB:AD:AF:DP:F1R2:F2R1:SB\n",
            "Sample 200852: 0/1\n",
            "\n",
            "\n",
            "CHROM: chr1\n",
            "POS: 133129\n",
            "ID: rs367730352\n",
            "REF: G\n",
            "ALT: [A]\n",
            "QUAL: None\n",
            "FILTER: []\n",
            "INFO: {'ACMG_class': ['Uncertain%40Significance'], 'ACMG_coding_impact': ['non%40coding'], 'ACMG_rules': ['BP4'], 'ACMG_score': [-0.047], 'AMP_matches': [''], 'AMP_rules': ['Type_IV%2CPred_IV'], 'AMP_tier': ['Tier%40IV'], 'AMP_total_samples': [0], 'AS_FilterStatus': 'SITE', 'AS_SB_TABLE': '0%2C0|18%2C2', 'DP': 22, 'ECNT': 1, 'GERMQ': 92, 'Gene': ['CICP27', 'ENSG00000238009'], 'MBQ': [27], 'MFRL': [348], 'MMQ': [33], 'MPOS': [20], 'POPAF': [7.3], 'TLOD': [57.1], 'function': ['non-coding%40exon', 'intronic'], 'gnomadGenomesAC': [72154], 'gnomadGenomesAN': [131698], 'gnomadGenomesEthnic_AC_Hom': [7468], 'gnomadGenomes_AC_Hom': [72154], 'gnomadGenomes_AF': [0.547875], 'gnomadGenomes_AF_ethnic': [0.597389], 'CSQ': ['A|non_coding_transcript_exon_variant|MODIFIER|CICP27|ENSG00000233750|Transcript|ENST00000442987.3|processed_pseudogene|1/1||ENST00000442987.3:n.2105G>A||2105|||||rs367730352||1||HGNC|HGNC:48835|YES||||||||||||Ensembl||G|G|||||||||||||||||||||0.5479|0.4621|0.5467|0.5007|0.5689|0.4942|0.6321|0.5|0.5974|0.5382|0.5257|||||||||||||5.750|0.429409||||||||||||', 'A|intron_variant&non_coding_transcript_variant|MODIFIER||ENSG00000238009|Transcript|ENST00000453576.2|lncRNA||1/1|ENST00000453576.2:n.193+245C>T|||||||rs367730352||-1|||||||2|||||||||Ensembl||G|G|||||||||||||||||||||0.5479|0.4621|0.5467|0.5007|0.5689|0.4942|0.6321|0.5|0.5974|0.5382|0.5257|||||||||||||5.750|0.429409||||||||||||', 'A|upstream_gene_variant|MODIFIER||ENSG00000238009|Transcript|ENST00000471248.1|lncRNA||||||||||rs367730352|3956|-1|||||||5|||||||||Ensembl||G|G|||||||||||||||||||||0.5479|0.4621|0.5467|0.5007|0.5689|0.4942|0.6321|0.5|0.5974|0.5382|0.5257|||||||||||||5.750|0.429409||||||||||||', 'A|upstream_gene_variant|MODIFIER||ENSG00000238009|Transcript|ENST00000477740.5|lncRNA||||||||||rs367730352|3912|-1||||YES|||5|||||||||Ensembl||G|G|||||||||||||||||||||0.5479|0.4621|0.5467|0.5007|0.5689|0.4942|0.6321|0.5|0.5974|0.5382|0.5257|||||||||||||5.750|0.429409||||||||||||', 'A|downstream_gene_variant|MODIFIER||ENSG00000268903|Transcript|ENST00000494149.2|processed_pseudogene||||||||||rs367730352|2012|-1||||YES||||||||||||Ensembl||G|G|||||||||||||||||||||0.5479|0.4621|0.5467|0.5007|0.5689|0.4942|0.6321|0.5|0.5974|0.5382|0.5257|||||||||||||5.750|0.429409||||||||||||', 'A|downstream_gene_variant|MODIFIER||ENSG00000269981|Transcript|ENST00000595919.1|processed_pseudogene||||||||||rs367730352|4553|-1||||YES||||||||||||Ensembl||G|G|||||||||||||||||||||0.5479|0.4621|0.5467|0.5007|0.5689|0.4942|0.6321|0.5|0.5974|0.5382|0.5257|||||||||||||5.750|0.429409||||||||||||', 'A|intron_variant&non_coding_transcript_variant|MODIFIER||ENSG00000238009|Transcript|ENST00000610542.1|lncRNA||1/3|ENST00000610542.1:n.350+245C>T|||||||rs367730352||-1|||||||5|||||||||Ensembl||G|G|||||||||||||||||||||0.5479|0.4621|0.5467|0.5007|0.5689|0.4942|0.6321|0.5|0.5974|0.5382|0.5257|||||||||||||5.750|0.429409||||||||||||', 'A|downstream_gene_variant|MODIFIER|LOC729737|729737|Transcript|NR_039983.2|lncRNA||||||||||rs367730352|1644|-1||EntrezGene||YES||||||||||||RefSeq||G|G|||||||||||||||||||||0.5479|0.4621|0.5467|0.5007|0.5689|0.4942|0.6321|0.5|0.5974|0.5382|0.5257|||||||||||||5.750|0.429409||||||||||||', 'A|intron_variant&non_coding_transcript_variant|MODIFIER|LOC124900384|124900384|Transcript|XR_001737579.3|lncRNA||5/5|XR_001737579.3:n.1968-12197C>T|||||||rs367730352||-1||EntrezGene||YES||||||||||||RefSeq||G|G|||||||||||||||||||||0.5479|0.4621|0.5467|0.5007|0.5689|0.4942|0.6321|0.5|0.5974|0.5382|0.5257|||||||||||||5.750|0.429409||||||||||||']}\n",
            "FORMAT: GT:AB:AD:AF:DP:F1R2:F2R1:SB\n",
            "Sample 200852: 0/1\n",
            "\n",
            "\n",
            "CHROM: chr1\n",
            "POS: 183629\n",
            "ID: rs71267774\n",
            "REF: G\n",
            "ALT: [A]\n",
            "QUAL: None\n",
            "FILTER: []\n",
            "INFO: {'ACMG_class': ['Benign'], 'ACMG_coding_impact': ['non%40coding'], 'ACMG_rules': ['BA1%2CBP4'], 'ACMG_score': [0.193], 'AMP_matches': ['E%2CV'], 'AMP_rules': ['Freq_II%2CType_IV%2CPred_IV'], 'AMP_tier': ['Tier%40IV'], 'AMP_total_samples': [0], 'AS_FilterStatus': 'SITE', 'AS_SB_TABLE': '50%2C53|19%2C14', 'DP': 139, 'ECNT': 4, 'GERMQ': 93, 'Gene': ['DDX11L17'], 'MBQ': [29], 'MFRL': [366], 'MMQ': [46], 'MPOS': [29], 'POPAF': [7.3], 'TLOD': [68.5], 'function': ['intronic'], 'gnomadGenomesAC': [14902], 'gnomadGenomesAN': [111796], 'gnomadGenomesEthnic_AC_Hom': [1], 'gnomadGenomes_AC_Hom': [14902], 'gnomadGenomes_AF': [0.133296], 'gnomadGenomes_AF_ethnic': [0.220891], 'CSQ': ['A|downstream_gene_variant|MODIFIER|MIR6859-2|ENSG00000273874|Transcript|ENST00000612080.1|miRNA||||||||||rs71267774|4262|-1||HGNC|HGNC:49954|YES||||||||||||Ensembl||G|G|||||||||||||||||||||0.1333|0.03798|0.1565|0.1196|0.1272|0.01827|0.2103|0.1154|0.2209|0.1247|0.1071|||||||||||||0.798|-0.107991||||||||||||', 'A|downstream_gene_variant|MODIFIER|WASH9P|ENSG00000279457|Transcript|ENST00000623083.4|unprocessed_pseudogene||||||||||rs71267774|1588|-1||HGNC|HGNC:53981|YES||||||||||||Ensembl||G|G|||||||||||||||||||||0.1333|0.03798|0.1565|0.1196|0.1272|0.01827|0.2103|0.1154|0.2209|0.1247|0.1071|||||||||||||0.798|-0.107991||||||||||||', 'A|intron_variant&non_coding_transcript_variant|MODIFIER|DDX11L17|ENSG00000279928|Transcript|ENST00000624431.2|unprocessed_pseudogene||3/4|ENST00000624431.2:n.214+58G>A|||||||rs71267774||1||HGNC|HGNC:55080|YES||||||||||||Ensembl||G|G|||||||||||||||||||||0.1333|0.03798|0.1565|0.1196|0.1272|0.01827|0.2103|0.1154|0.2209|0.1247|0.1071|||||||||||||0.798|-0.107991||||||||||||', 'A|intron_variant&non_coding_transcript_variant|MODIFIER|DDX11L17|102725121|Transcript|NR_148357.1|transcribed_pseudogene||2/2|NR_148357.1:n.469-111G>A|||||||rs71267774||1||EntrezGene|HGNC:55080|YES||||||||||||RefSeq||G|G|||||||||||||||||||||0.1333|0.03798|0.1565|0.1196|0.1272|0.01827|0.2103|0.1154|0.2209|0.1247|0.1071|||||||||||||0.798|-0.107991||||||||||||', 'A|downstream_gene_variant|MODIFIER|LOC100996442|100996442|Transcript|XR_007065318.1|lncRNA||||||||||rs71267774|1287|-1||EntrezGene||YES||||||||||||RefSeq||G|G|||||||||||||||||||||0.1333|0.03798|0.1565|0.1196|0.1272|0.01827|0.2103|0.1154|0.2209|0.1247|0.1071|||||||||||||0.798|-0.107991||||||||||||', 'A|upstream_gene_variant|MODIFIER|LOC124903814|124903814|Transcript|XR_007065332.1|lncRNA||||||||||rs71267774|3524|-1||EntrezGene||YES||||||||||||RefSeq||G|G|||||||||||||||||||||0.1333|0.03798|0.1565|0.1196|0.1272|0.01827|0.2103|0.1154|0.2209|0.1247|0.1071|||||||||||||0.798|-0.107991||||||||||||', 'A|upstream_gene_variant|MODIFIER|LOC124903814|124903814|Transcript|XR_007065333.1|lncRNA||||||||||rs71267774|3524|-1||EntrezGene||||||||||||||RefSeq||G|G|||||||||||||||||||||0.1333|0.03798|0.1565|0.1196|0.1272|0.01827|0.2103|0.1154|0.2209|0.1247|0.1071|||||||||||||0.798|-0.107991||||||||||||', 'A|upstream_gene_variant|MODIFIER|LOC124903814|124903814|Transcript|XR_007065334.1|lncRNA||||||||||rs71267774|3524|-1||EntrezGene||||||||||||||RefSeq||G|G|||||||||||||||||||||0.1333|0.03798|0.1565|0.1196|0.1272|0.01827|0.2103|0.1154|0.2209|0.1247|0.1071|||||||||||||0.798|-0.107991||||||||||||']}\n",
            "FORMAT: GT:AB:AD:AF:DP:F1R2:F2R1:SB\n",
            "Sample 200852: 0/1\n",
            "\n",
            "\n",
            "CHROM: chr1\n",
            "POS: 184267\n",
            "ID: rs1396210256\n",
            "REF: C\n",
            "ALT: [T]\n",
            "QUAL: None\n",
            "FILTER: []\n",
            "INFO: {'ACMG_class': ['Likely%40Benign'], 'ACMG_coding_impact': ['non%40coding'], 'ACMG_rules': ['BS1%2CBP4'], 'ACMG_score': [0.193], 'AMP_matches': ['E%2CV'], 'AMP_rules': ['Freq_II%2CType_IV%2CPred_IV'], 'AMP_tier': ['Tier%40IV'], 'AMP_total_samples': [0], 'AS_FilterStatus': 'SITE', 'AS_SB_TABLE': '32%2C33|3%2C4', 'DP': 78, 'ECNT': 1, 'GERMQ': 93, 'Gene': ['DDX11L17'], 'MBQ': [28], 'MFRL': [237], 'MMQ': [39], 'MPOS': [13], 'POPAF': [7.3], 'TLOD': [11.4], 'function': ['non-coding%40exon', \"3'flank\"], 'gnomadGenomesAC': [887], 'gnomadGenomesAN': [144780], 'gnomadGenomesEthnic_AC_Hom': [0], 'gnomadGenomes_AC_Hom': [887], 'gnomadGenomes_AF': [0.00612654], 'gnomadGenomes_AF_ethnic': [0.00631038], 'CSQ': ['T|downstream_gene_variant|MODIFIER|MIR6859-2|ENSG00000273874|Transcript|ENST00000612080.1|miRNA||||||||||rs1396210256|3624|-1||HGNC|HGNC:49954|YES||||||||||||Ensembl||C|C|||||||||||||||||||||0.006127|0.002936|0.01439|0.007364|0.009034|0.0006326|0.01197|0.01316|0.00631|0.006972|0.01678|||||||||||||1.100|-0.040819||||||||||||', 'T|downstream_gene_variant|MODIFIER|WASH9P|ENSG00000279457|Transcript|ENST00000623083.4|unprocessed_pseudogene||||||||||rs1396210256|950|-1||HGNC|HGNC:53981|YES||||||||||||Ensembl||C|C|||||||||||||||||||||0.006127|0.002936|0.01439|0.007364|0.009034|0.0006326|0.01197|0.01316|0.00631|0.006972|0.01678|||||||||||||1.100|-0.040819||||||||||||', 'T|downstream_gene_variant|MODIFIER|DDX11L17|ENSG00000279928|Transcript|ENST00000624431.2|unprocessed_pseudogene||||||||||rs1396210256|93|1||HGNC|HGNC:55080|YES||||||||||||Ensembl||C|C|||||||||||||||||||||0.006127|0.002936|0.01439|0.007364|0.009034|0.0006326|0.01197|0.01316|0.00631|0.006972|0.01678|||||||||||||1.100|-0.040819||||||||||||', 'T|non_coding_transcript_exon_variant|MODIFIER|DDX11L17|102725121|Transcript|NR_148357.1|transcribed_pseudogene|3/3||NR_148357.1:n.996C>T||996|||||rs1396210256||1||EntrezGene|HGNC:55080|YES||||||||||||RefSeq||C|C|||||||||||||||||||||0.006127|0.002936|0.01439|0.007364|0.009034|0.0006326|0.01197|0.01316|0.00631|0.006972|0.01678|||||||||||||1.100|-0.040819||||||||||||', 'T|downstream_gene_variant|MODIFIER|LOC100996442|100996442|Transcript|XR_007065318.1|lncRNA||||||||||rs1396210256|649|-1||EntrezGene||YES||||||||||||RefSeq||C|C|||||||||||||||||||||0.006127|0.002936|0.01439|0.007364|0.009034|0.0006326|0.01197|0.01316|0.00631|0.006972|0.01678|||||||||||||1.100|-0.040819||||||||||||', 'T|upstream_gene_variant|MODIFIER|LOC124903814|124903814|Transcript|XR_007065332.1|lncRNA||||||||||rs1396210256|4162|-1||EntrezGene||YES||||||||||||RefSeq||C|C|||||||||||||||||||||0.006127|0.002936|0.01439|0.007364|0.009034|0.0006326|0.01197|0.01316|0.00631|0.006972|0.01678|||||||||||||1.100|-0.040819||||||||||||', 'T|upstream_gene_variant|MODIFIER|LOC124903814|124903814|Transcript|XR_007065333.1|lncRNA||||||||||rs1396210256|4162|-1||EntrezGene||||||||||||||RefSeq||C|C|||||||||||||||||||||0.006127|0.002936|0.01439|0.007364|0.009034|0.0006326|0.01197|0.01316|0.00631|0.006972|0.01678|||||||||||||1.100|-0.040819||||||||||||', 'T|upstream_gene_variant|MODIFIER|LOC124903814|124903814|Transcript|XR_007065334.1|lncRNA||||||||||rs1396210256|4162|-1||EntrezGene||||||||||||||RefSeq||C|C|||||||||||||||||||||0.006127|0.002936|0.01439|0.007364|0.009034|0.0006326|0.01197|0.01316|0.00631|0.006972|0.01678|||||||||||||1.100|-0.040819||||||||||||']}\n",
            "FORMAT: GT:AB:AD:AF:DP:F1R2:F2R1:SB\n",
            "Sample 200852: 0/1\n",
            "\n",
            "\n",
            "CHROM: chr1\n",
            "POS: 184413\n",
            "ID: rs1441656517\n",
            "REF: C\n",
            "ALT: [A]\n",
            "QUAL: None\n",
            "FILTER: []\n",
            "INFO: {'ACMG_class': ['Uncertain%40Significance'], 'ACMG_coding_impact': ['non%40coding'], 'ACMG_rules': ['BP4'], 'ACMG_score': [0.193], 'AMP_matches': ['V'], 'AMP_rules': ['Freq_II%2CType_IV%2CPred_IV'], 'AMP_tier': ['Tier%40IV'], 'AMP_total_samples': [0], 'AS_FilterStatus': 'SITE', 'AS_SB_TABLE': '68%2C49|12%2C10', 'DP': 153, 'ECNT': 1, 'GERMQ': 93, 'Gene': ['DDX11L17'], 'MBQ': [28], 'MFRL': [228], 'MMQ': [37], 'MPOS': [23], 'POPAF': [7.3], 'TLOD': [42.4], 'function': ['non-coding%40exon', \"3'flank\"], 'gnomadGenomesAC': [33879], 'gnomadGenomesAN': [114958], 'gnomadGenomesEthnic_AC_Hom': [18], 'gnomadGenomes_AC_Hom': [33879], 'gnomadGenomes_AF': [0.294708], 'gnomadGenomes_AF_ethnic': [0.335956], 'CSQ': ['A|downstream_gene_variant|MODIFIER|MIR6859-2|ENSG00000273874|Transcript|ENST00000612080.1|miRNA||||||||||rs1441656517|3478|-1||HGNC|HGNC:49954|YES||||||||||||Ensembl||C|C|||||||||||||||||||||0.2947|0.2|0.4231|0.364|0.3479|0.3555|0.2805|0.3261|0.336|0.3249|0.2691|||||||||||||1.513|0.026980||||||||||||', 'A|downstream_gene_variant|MODIFIER|WASH9P|ENSG00000279457|Transcript|ENST00000623083.4|unprocessed_pseudogene||||||||||rs1441656517|804|-1||HGNC|HGNC:53981|YES||||||||||||Ensembl||C|C|||||||||||||||||||||0.2947|0.2|0.4231|0.364|0.3479|0.3555|0.2805|0.3261|0.336|0.3249|0.2691|||||||||||||1.513|0.026980||||||||||||', 'A|downstream_gene_variant|MODIFIER|DDX11L17|ENSG00000279928|Transcript|ENST00000624431.2|unprocessed_pseudogene||||||||||rs1441656517|239|1||HGNC|HGNC:55080|YES||||||||||||Ensembl||C|C|||||||||||||||||||||0.2947|0.2|0.4231|0.364|0.3479|0.3555|0.2805|0.3261|0.336|0.3249|0.2691|||||||||||||1.513|0.026980||||||||||||', 'A|non_coding_transcript_exon_variant|MODIFIER|DDX11L17|102725121|Transcript|NR_148357.1|transcribed_pseudogene|3/3||NR_148357.1:n.1142C>A||1142|||||rs1441656517||1||EntrezGene|HGNC:55080|YES||||||||||||RefSeq||C|C|||||||||||||||||||||0.2947|0.2|0.4231|0.364|0.3479|0.3555|0.2805|0.3261|0.336|0.3249|0.2691|||||||||||||1.513|0.026980||||||||||||', 'A|downstream_gene_variant|MODIFIER|LOC100996442|100996442|Transcript|XR_007065318.1|lncRNA||||||||||rs1441656517|503|-1||EntrezGene||YES||||||||||||RefSeq||C|C|||||||||||||||||||||0.2947|0.2|0.4231|0.364|0.3479|0.3555|0.2805|0.3261|0.336|0.3249|0.2691|||||||||||||1.513|0.026980||||||||||||', 'A|upstream_gene_variant|MODIFIER|LOC124903814|124903814|Transcript|XR_007065332.1|lncRNA||||||||||rs1441656517|4308|-1||EntrezGene||YES||||||||||||RefSeq||C|C|||||||||||||||||||||0.2947|0.2|0.4231|0.364|0.3479|0.3555|0.2805|0.3261|0.336|0.3249|0.2691|||||||||||||1.513|0.026980||||||||||||', 'A|upstream_gene_variant|MODIFIER|LOC124903814|124903814|Transcript|XR_007065333.1|lncRNA||||||||||rs1441656517|4308|-1||EntrezGene||||||||||||||RefSeq||C|C|||||||||||||||||||||0.2947|0.2|0.4231|0.364|0.3479|0.3555|0.2805|0.3261|0.336|0.3249|0.2691|||||||||||||1.513|0.026980||||||||||||', 'A|upstream_gene_variant|MODIFIER|LOC124903814|124903814|Transcript|XR_007065334.1|lncRNA||||||||||rs1441656517|4308|-1||EntrezGene||||||||||||||RefSeq||C|C|||||||||||||||||||||0.2947|0.2|0.4231|0.364|0.3479|0.3555|0.2805|0.3261|0.336|0.3249|0.2691|||||||||||||1.513|0.026980||||||||||||']}\n",
            "FORMAT: GT:AB:AD:AF:DP:F1R2:F2R1:SB\n",
            "Sample 200852: 0/1\n",
            "\n",
            "\n",
            "CHROM: chr1\n",
            "POS: 186338\n",
            "ID: rs1490434665\n",
            "REF: T\n",
            "ALT: [G]\n",
            "QUAL: None\n",
            "FILTER: []\n",
            "INFO: {'ACMG_class': ['Uncertain%40Significance'], 'ACMG_coding_impact': ['non%40coding'], 'ACMG_rules': ['BP4'], 'ACMG_score': [0.193], 'AMP_matches': ['V'], 'AMP_rules': ['Freq_II%2CType_IV%2CPred_IV'], 'AMP_tier': ['Tier%40IV'], 'AMP_total_samples': [0], 'AS_FilterStatus': 'SITE', 'AS_SB_TABLE': '32%2C17|6%2C0', 'DP': 60, 'ECNT': 3, 'GERMQ': 93, 'Gene': ['WASH9P'], 'MBQ': [30], 'MFRL': [398], 'MMQ': [38], 'MPOS': [26], 'POPAF': [7.3], 'TLOD': [10.6], 'function': ['non-coding%40exon'], 'gnomadGenomesAC': [12286], 'gnomadGenomesAN': [90730], 'gnomadGenomesEthnic_AC_Hom': [8], 'gnomadGenomes_AC_Hom': [12286], 'gnomadGenomes_AF': [0.135413], 'gnomadGenomes_AF_ethnic': [0.114409], 'CSQ': ['G|downstream_gene_variant|MODIFIER|MIR6859-2|ENSG00000273874|Transcript|ENST00000612080.1|miRNA||||||||||rs1490434665|1553|-1||HGNC|HGNC:49954|YES||||||||||||Ensembl||T|T|||||||||||||||||||||0.1354|0.1788|0.1337|0.1068|0.08754|0.1299|0.2021|0.08|0.1144|0.1153|0.0768|||||||||||||18.84|1.944793||||||||||||', 'G|non_coding_transcript_exon_variant|MODIFIER|WASH9P|ENSG00000279457|Transcript|ENST00000623083.4|unprocessed_pseudogene|8/10||ENST00000623083.4:n.1173A>C||1173|||||rs1490434665||-1||HGNC|HGNC:53981|YES||||||||||||Ensembl||T|T|||||||||||||||||||||0.1354|0.1788|0.1337|0.1068|0.08754|0.1299|0.2021|0.08|0.1144|0.1153|0.0768|||||||||||||18.84|1.944793||||||||||||', 'G|downstream_gene_variant|MODIFIER|DDX11L17|ENSG00000279928|Transcript|ENST00000624431.2|unprocessed_pseudogene||||||||||rs1490434665|2164|1||HGNC|HGNC:55080|YES||||||||||||Ensembl||T|T|||||||||||||||||||||0.1354|0.1788|0.1337|0.1068|0.08754|0.1299|0.2021|0.08|0.1144|0.1153|0.0768|||||||||||||18.84|1.944793||||||||||||', 'G|downstream_gene_variant|MODIFIER|DDX11L17|102725121|Transcript|NR_148357.1|transcribed_pseudogene||||||||||rs1490434665|1460|1||EntrezGene|HGNC:55080|YES||||||||||||RefSeq||T|T|||||||||||||||||||||0.1354|0.1788|0.1337|0.1068|0.08754|0.1299|0.2021|0.08|0.1144|0.1153|0.0768|||||||||||||18.84|1.944793||||||||||||', 'G|non_coding_transcript_exon_variant|MODIFIER|LOC100996442|100996442|Transcript|XR_007065318.1|lncRNA|9/11||XR_007065318.1:n.1242A>C||1242|||||rs1490434665||-1||EntrezGene||YES||||||||||||RefSeq||T|T|||||||||||||||||||||0.1354|0.1788|0.1337|0.1068|0.08754|0.1299|0.2021|0.08|0.1144|0.1153|0.0768|||||||||||||18.84|1.944793||||||||||||', 'G|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00000344295|CTCF_binding_site||||||||||rs1490434665||||||||||||||||||||||||||||||||||||||||||0.1354|0.1788|0.1337|0.1068|0.08754|0.1299|0.2021|0.08|0.1144|0.1153|0.0768|||||||||||||18.84|1.944793||||||||||||', 'G|TF_binding_site_variant|MODIFIER|||MotifFeature|ENSM00000225283|||||||||||rs1490434665||1||||||||||||||||||||||||||||||||||||||||0.1354|0.1788|0.1337|0.1068|0.08754|0.1299|0.2021|0.08|0.1144|0.1153|0.0768||||||ENSPFM0042|14|Y|0.035|CTCF|||18.84|1.944793||||||||||||']}\n",
            "FORMAT: GT:AB:AD:AF:DP:F1R2:F2R1:SB\n",
            "Sample 200852: 0/1\n",
            "\n",
            "\n",
            "CHROM: chr1\n",
            "POS: 186835\n",
            "ID: rs1364805962\n",
            "REF: A\n",
            "ALT: [G]\n",
            "QUAL: None\n",
            "FILTER: []\n",
            "INFO: {'ACMG_class': ['Uncertain%40Significance'], 'ACMG_coding_impact': ['non%40coding'], 'ACMG_rules': ['BP4'], 'ACMG_score': [-0.047], 'AMP_matches': [''], 'AMP_rules': ['Type_IV%2CPred_IV'], 'AMP_tier': ['Tier%40IV'], 'AMP_total_samples': [0], 'AS_FilterStatus': 'SITE', 'AS_SB_TABLE': '6%2C2|4%2C2', 'DP': 15, 'ECNT': 1, 'GERMQ': 32, 'Gene': ['WASH9P'], 'MBQ': [29], 'MFRL': [332], 'MMQ': [36], 'MPOS': [5], 'POPAF': [7.3], 'TLOD': [13.9], 'function': ['intronic'], 'gnomadGenomesAC': [97257], 'gnomadGenomesAN': [150682], 'gnomadGenomesEthnic_AC_Hom': [4494], 'gnomadGenomes_AC_Hom': [97257], 'gnomadGenomes_AF': [0.645445], 'gnomadGenomes_AF_ethnic': [0.566261], 'CSQ': ['G|downstream_gene_variant|MODIFIER|MIR6859-2|ENSG00000273874|Transcript|ENST00000612080.1|miRNA||||||||||rs1364805962|1056|-1||HGNC|HGNC:49954|YES||||||||||||Ensembl||A|A|||||||||||||||||||||0.6454|0.8165|0.5659|0.6155|0.5712|0.7173|0.5622|0.6019|0.5663|0.6252|0.5847|||||||||||||4.041|0.278704||||||||||||', 'G|intron_variant&non_coding_transcript_variant|MODIFIER|WASH9P|ENSG00000279457|Transcript|ENST00000623083.4|unprocessed_pseudogene||7/9|ENST00000623083.4:n.1041+294T>C|||||||rs1364805962||-1||HGNC|HGNC:53981|YES||||||||||||Ensembl||A|A|||||||||||||||||||||0.6454|0.8165|0.5659|0.6155|0.5712|0.7173|0.5622|0.6019|0.5663|0.6252|0.5847|||||||||||||4.041|0.278704||||||||||||', 'G|downstream_gene_variant|MODIFIER|DDX11L17|ENSG00000279928|Transcript|ENST00000624431.2|unprocessed_pseudogene||||||||||rs1364805962|2661|1||HGNC|HGNC:55080|YES||||||||||||Ensembl||A|A|||||||||||||||||||||0.6454|0.8165|0.5659|0.6155|0.5712|0.7173|0.5622|0.6019|0.5663|0.6252|0.5847|||||||||||||4.041|0.278704||||||||||||', 'G|downstream_gene_variant|MODIFIER|DDX11L17|102725121|Transcript|NR_148357.1|transcribed_pseudogene||||||||||rs1364805962|1957|1||EntrezGene|HGNC:55080|YES||||||||||||RefSeq||A|A|||||||||||||||||||||0.6454|0.8165|0.5659|0.6155|0.5712|0.7173|0.5622|0.6019|0.5663|0.6252|0.5847|||||||||||||4.041|0.278704||||||||||||', 'G|intron_variant&non_coding_transcript_variant|MODIFIER|LOC100996442|100996442|Transcript|XR_007065318.1|lncRNA||8/10|XR_007065318.1:n.1110+294T>C|||||||rs1364805962||-1||EntrezGene||YES||||||||||||RefSeq||A|A|||||||||||||||||||||0.6454|0.8165|0.5659|0.6155|0.5712|0.7173|0.5622|0.6019|0.5663|0.6252|0.5847|||||||||||||4.041|0.278704||||||||||||', 'G|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00000344295|CTCF_binding_site||||||||||rs1364805962||||||||||||||||||||||||||||||||||||||||||0.6454|0.8165|0.5659|0.6155|0.5712|0.7173|0.5622|0.6019|0.5663|0.6252|0.5847|||||||||||||4.041|0.278704||||||||||||']}\n",
            "FORMAT: GT:AB:AD:AF:DP:F1R2:F2R1:SB\n",
            "Sample 200852: 0/1\n",
            "\n",
            "\n",
            "CHROM: chr1\n",
            "POS: 187102\n",
            "ID: rs1221392627\n",
            "REF: C\n",
            "ALT: [G]\n",
            "QUAL: None\n",
            "FILTER: []\n",
            "INFO: {'ACMG_class': ['Uncertain%40Significance'], 'ACMG_coding_impact': ['non%40coding'], 'ACMG_rules': ['BP4'], 'ACMG_score': [0.193], 'AMP_matches': ['V'], 'AMP_rules': ['Freq_II%2CType_IV%2CPred_IV'], 'AMP_tier': ['Tier%40IV'], 'AMP_total_samples': [0], 'AS_FilterStatus': 'SITE', 'AS_SB_TABLE': '63%2C48|19%2C14', 'DP': 149, 'ECNT': 4, 'GERMQ': 93, 'Gene': ['WASH9P'], 'MBQ': [30], 'MFRL': [251], 'MMQ': [40], 'MPOS': [15], 'POPAF': [7.3], 'TLOD': [70.9], 'function': ['intronic'], 'gnomadGenomesAC': [54488], 'gnomadGenomesAN': [119036], 'gnomadGenomesEthnic_AC_Hom': [4], 'gnomadGenomes_AC_Hom': [54488], 'gnomadGenomes_AF': [0.457744], 'gnomadGenomes_AF_ethnic': [0.445951], 'CSQ': ['G|downstream_gene_variant|MODIFIER|MIR6859-2|ENSG00000273874|Transcript|ENST00000612080.1|miRNA||||||||||rs1221392627|789|-1||HGNC|HGNC:49954|YES||||||||||||Ensembl||C|C|||||||||||||||||||||0.4577|0.4973|0.4637|0.4473|0.4448|0.3969|0.4094|0.4113|0.446|0.4559|0.4219|||||||||||||0.531|-0.192081||||||||||||', 'G|intron_variant&non_coding_transcript_variant|MODIFIER|WASH9P|ENSG00000279457|Transcript|ENST00000623083.4|unprocessed_pseudogene||7/9|ENST00000623083.4:n.1041+27G>C|||||||rs1221392627||-1||HGNC|HGNC:53981|YES||||||||||||Ensembl||C|C|||||||||||||||||||||0.4577|0.4973|0.4637|0.4473|0.4448|0.3969|0.4094|0.4113|0.446|0.4559|0.4219|||||||||||||0.531|-0.192081||||||||||||', 'G|downstream_gene_variant|MODIFIER|DDX11L17|ENSG00000279928|Transcript|ENST00000624431.2|unprocessed_pseudogene||||||||||rs1221392627|2928|1||HGNC|HGNC:55080|YES||||||||||||Ensembl||C|C|||||||||||||||||||||0.4577|0.4973|0.4637|0.4473|0.4448|0.3969|0.4094|0.4113|0.446|0.4559|0.4219|||||||||||||0.531|-0.192081||||||||||||', 'G|downstream_gene_variant|MODIFIER|DDX11L17|102725121|Transcript|NR_148357.1|transcribed_pseudogene||||||||||rs1221392627|2224|1||EntrezGene|HGNC:55080|YES||||||||||||RefSeq||C|C|||||||||||||||||||||0.4577|0.4973|0.4637|0.4473|0.4448|0.3969|0.4094|0.4113|0.446|0.4559|0.4219|||||||||||||0.531|-0.192081||||||||||||', 'G|intron_variant&non_coding_transcript_variant|MODIFIER|LOC100996442|100996442|Transcript|XR_007065318.1|lncRNA||8/10|XR_007065318.1:n.1110+27G>C|||||||rs1221392627||-1||EntrezGene||YES||||||||||||RefSeq||C|C|||||||||||||||||||||0.4577|0.4973|0.4637|0.4473|0.4448|0.3969|0.4094|0.4113|0.446|0.4559|0.4219|||||||||||||0.531|-0.192081||||||||||||', 'G|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00000344295|CTCF_binding_site||||||||||rs1221392627||||||||||||||||||||||||||||||||||||||||||0.4577|0.4973|0.4637|0.4473|0.4448|0.3969|0.4094|0.4113|0.446|0.4559|0.4219|||||||||||||0.531|-0.192081||||||||||||']}\n",
            "FORMAT: GT:AB:AD:AF:DP:F1R2:F2R1:SB\n",
            "Sample 200852: 0/1\n",
            "\n",
            "\n"
          ]
        }
      ],
      "source": [
        "import vcf\n",
        "\n",
        "vcf_file = \"data/EE_015.vcf.gz\"\n",
        "\n",
        "with open(vcf_file, 'rb') as vcf_file_binary:\n",
        "    vcf_reader = vcf.Reader(vcf_file_binary)\n",
        "\n",
        "    i=0\n",
        "    for record in vcf_reader:\n",
        "        i+=1\n",
        "        if i>10:\n",
        "            break\n",
        "        print(\"CHROM:\", record.CHROM)\n",
        "        print(\"POS:\", record.POS)\n",
        "        print(\"ID:\", record.ID)\n",
        "        print(\"REF:\", record.REF)\n",
        "        print(\"ALT:\", record.ALT)\n",
        "        print(\"QUAL:\", record.QUAL)\n",
        "        print(\"FILTER:\", record.FILTER)\n",
        "        print(\"INFO:\", record.INFO)\n",
        "        print(\"FORMAT:\", record.FORMAT)\n",
        "\n",
        "        for sample in record.samples:\n",
        "            print(f\"Sample {sample.sample}: {sample['GT']}\")\n",
        "\n",
        "        print(\"\\n\")"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "9M1uFHLCqTJ3"
      },
      "source": [
        "Here is the explanation for the above fields(courtesy of ChatGPT):\n",
        "* CHROM: The chromosome or sequence name where the variant is located. It is a required field and is typically specified in the VCF record.\n",
        "\n",
        "* POS: The position of the variant on the chromosome. It is also a required field and is specified in the VCF record.\n",
        "\n",
        "* ID: An optional identifier for the variant. If not provided, it can be left blank or set to a dot (\".\").\n",
        "\n",
        "* REF: The reference allele(s) at the specified position. It is a required field.\n",
        "\n",
        "* ALT: The alternate allele(s) at the specified position. It is a required field.\n",
        "\n",
        "* QUAL: The Phred-scaled quality score. It represents the confidence in the variant call and is typically specified even if the value is missing or set to a default value.\n",
        "\n",
        "* FILTER: The filter status for the variant, indicating whether it passed certain quality control filters. It is typically set to either \"PASS\" or a specific filter name if applied."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "ocqiEZ4nrDyx"
      },
      "source": [
        "Below code is responsible for reading the meta data and available \"custom\" fields in the file to know what we can work with."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "LJ5FEbEme783",
        "outputId": "40fc973f-33de-4972-c96c-7e6f5e0f3109"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Metadata: fileformat\n",
            "Metadata: reference\n",
            "Metadata: SentieonCommandLine.TNfilter\n",
            "Metadata: SentieonCommandLine.TNhaplotyper2\n",
            "Metadata: tumor_sample\n",
            "Metadata: VEP\n",
            "Metadata: EVE_CLASS\n",
            "Metadata: EVE_SCORE\n",
            "Metadata: CADD_PHRED\n",
            "Metadata: CADD_RAW\n",
            "Metadata: SpliceAI_pred_DP_AG\n",
            "Metadata: SpliceAI_pred_DP_AL\n",
            "Metadata: SpliceAI_pred_DP_DG\n",
            "Metadata: SpliceAI_pred_DP_DL\n",
            "Metadata: SpliceAI_pred_DS_AG\n",
            "Metadata: SpliceAI_pred_DS_AL\n",
            "Metadata: SpliceAI_pred_DS_DG\n",
            "Metadata: SpliceAI_pred_DS_DL\n",
            "Metadata: SpliceAI_pred_SYMBOL\n",
            "Metadata: LOEUF\n",
            "Metadata: PHENOTYPES\n",
            "Metadata: NMD\n",
            "Metadata: VEP-command-line\n",
            "Field Names: odict_keys(['ACMG_class', 'ACMG_coding_impact', 'ACMG_gene', 'ACMG_rules', 'ACMG_score', 'ACMG_transcript', 'AMP_matches', 'AMP_rules', 'AMP_score', 'AMP_tier', 'AMP_total_samples', 'AS_FilterStatus', 'AS_SB_TABLE', 'CGDinheritance', 'ClinVarClass', 'ClinVarDisease', 'DANN_score', 'DP', 'ECNT', 'GERMQ', 'Gene', 'MBQ', 'MFRL', 'MMQ', 'MPOS', 'MutationTaster_pred', 'MutationTaster_score', 'POPAF', 'RPA', 'RU', 'SIFT_score', 'STR', 'STRQ', 'TLOD', 'coding_impact', 'cosmicFathMMPrediction', 'cosmicFathMMScore', 'function', 'gnomadExomesEthnic_AC_Hom', 'gnomadExomes_AC', 'gnomadExomes_AC_Hemi', 'gnomadExomes_AC_Hom', 'gnomadExomes_AF', 'gnomadExomes_AF_ethnic', 'gnomadExomes_AN', 'gnomadGenomesAC', 'gnomadGenomesAN', 'gnomadGenomesEthnic_AC_Hom', 'gnomadGenomes_AC_Hemi', 'gnomadGenomes_AC_Hom', 'gnomadGenomes_AF', 'gnomadGenomes_AF_ethnic', 'hgvs', 'CSQ'])\n"
          ]
        }
      ],
      "source": [
        "import vcf\n",
        "\n",
        "with open(vcf_file, 'rb') as vcf_file_binary:\n",
        "    vcf_reader = vcf.Reader(vcf_file_binary)\n",
        "\n",
        "    for meta_info in vcf_reader.metadata:\n",
        "        print(\"Metadata:\", meta_info)\n",
        "\n",
        "    column_headers = vcf_reader.infos.keys()\n",
        "    print(\"Field Names:\", column_headers)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "T2tzxySbrccG"
      },
      "source": [
        "Lastly, let's put it all together and read both \"built-in\" and \"custom\" fields."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ZZuThfk8huhc",
        "outputId": "59b7ecfc-7dca-44e5-be9f-8bf39723a6cb"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Metadata: fileformat\n",
            "Metadata: reference\n",
            "Metadata: SentieonCommandLine.TNfilter\n",
            "Metadata: SentieonCommandLine.TNhaplotyper2\n",
            "Metadata: tumor_sample\n",
            "Metadata: VEP\n",
            "Metadata: EVE_CLASS\n",
            "Metadata: EVE_SCORE\n",
            "Metadata: CADD_PHRED\n",
            "Metadata: CADD_RAW\n",
            "Metadata: SpliceAI_pred_DP_AG\n",
            "Metadata: SpliceAI_pred_DP_AL\n",
            "Metadata: SpliceAI_pred_DP_DG\n",
            "Metadata: SpliceAI_pred_DP_DL\n",
            "Metadata: SpliceAI_pred_DS_AG\n",
            "Metadata: SpliceAI_pred_DS_AL\n",
            "Metadata: SpliceAI_pred_DS_DG\n",
            "Metadata: SpliceAI_pred_DS_DL\n",
            "Metadata: SpliceAI_pred_SYMBOL\n",
            "Metadata: LOEUF\n",
            "Metadata: PHENOTYPES\n",
            "Metadata: NMD\n",
            "Metadata: VEP-command-line\n",
            "Field Names: odict_keys(['ACMG_class', 'ACMG_coding_impact', 'ACMG_gene', 'ACMG_rules', 'ACMG_score', 'ACMG_transcript', 'AMP_matches', 'AMP_rules', 'AMP_score', 'AMP_tier', 'AMP_total_samples', 'AS_FilterStatus', 'AS_SB_TABLE', 'CGDinheritance', 'ClinVarClass', 'ClinVarDisease', 'DANN_score', 'DP', 'ECNT', 'GERMQ', 'Gene', 'MBQ', 'MFRL', 'MMQ', 'MPOS', 'MutationTaster_pred', 'MutationTaster_score', 'POPAF', 'RPA', 'RU', 'SIFT_score', 'STR', 'STRQ', 'TLOD', 'coding_impact', 'cosmicFathMMPrediction', 'cosmicFathMMScore', 'function', 'gnomadExomesEthnic_AC_Hom', 'gnomadExomes_AC', 'gnomadExomes_AC_Hemi', 'gnomadExomes_AC_Hom', 'gnomadExomes_AF', 'gnomadExomes_AF_ethnic', 'gnomadExomes_AN', 'gnomadGenomesAC', 'gnomadGenomesAN', 'gnomadGenomesEthnic_AC_Hom', 'gnomadGenomes_AC_Hemi', 'gnomadGenomes_AC_Hom', 'gnomadGenomes_AF', 'gnomadGenomes_AF_ethnic', 'hgvs', 'CSQ'])\n",
            "CHROM: chr1\n",
            "POS: 15820\n",
            "ID: rs2691315\n",
            "REF: G\n",
            "ALT: [T]\n",
            "QUAL: None\n",
            "FILTER: []\n",
            "ACMG_class: ['Uncertain%40Significance']\n",
            "ACMG_coding_impact: ['non%40coding']\n",
            "ACMG_rules: ['BP4']\n",
            "ACMG_score: [0.193]\n",
            "AMP_matches: ['V']\n",
            "AMP_rules: ['Freq_II%2CType_IV%2CPred_IV']\n",
            "AMP_tier: ['Tier%40IV']\n",
            "AMP_total_samples: [0]\n",
            "AS_FilterStatus: SITE\n",
            "AS_SB_TABLE: 12%2C23|7%2C2\n",
            "DP: 45\n",
            "ECNT: 1\n",
            "GERMQ: 55\n",
            "Gene: ['WASH7P']\n",
            "MBQ: [29]\n",
            "MFRL: [219]\n",
            "MMQ: [34]\n",
            "MPOS: [30]\n",
            "POPAF: [7.3]\n",
            "TLOD: [10.9]\n",
            "function: ['non-coding%40exon']\n",
            "gnomadGenomesAC: [18059]\n",
            "gnomadGenomesAN: [82218]\n",
            "gnomadGenomesEthnic_AC_Hom: [233]\n",
            "gnomadGenomes_AC_Hom: [18059]\n",
            "gnomadGenomes_AF: [0.219648]\n",
            "gnomadGenomes_AF_ethnic: [0.0938074]\n",
            "CSQ: ['T|downstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000450305.2|transcribed_unprocessed_pseudogene||||||||||rs2691315|2150|1||HGNC|HGNC:37102|YES||||||||||||Ensembl||G|G||||||0.4105|0.4849|0.2939|0.6052|0.2714|0.3354||||||||||0.2196|0.4324|0.1146|0.1638|0.1285|0.06176|0.06299|0.2151|0.09381|0.2235|0.1329|||||||||||||16.02|1.543896||||||||||||', 'T|downstream_gene_variant|MODIFIER|DDX11L2|ENSG00000290825|Transcript|ENST00000456328.2|lncRNA||||||||||rs2691315|1411|1||EntrezGene||YES|||1|||||||||Ensembl||G|G||||||0.4105|0.4849|0.2939|0.6052|0.2714|0.3354||||||||||0.2196|0.4324|0.1146|0.1638|0.1285|0.06176|0.06299|0.2151|0.09381|0.2235|0.1329|||||||||||||16.02|1.543896||||||||||||', 'T|non_coding_transcript_exon_variant|MODIFIER|WASH7P|ENSG00000227232|Transcript|ENST00000488147.1|unprocessed_pseudogene|9/11||ENST00000488147.1:n.1195C>A||1195|||||rs2691315||-1||HGNC|HGNC:38034|YES||||||||||||Ensembl||G|G||||||0.4105|0.4849|0.2939|0.6052|0.2714|0.3354||||||||||0.2196|0.4324|0.1146|0.1638|0.1285|0.06176|0.06299|0.2151|0.09381|0.2235|0.1329|||||||||||||16.02|1.543896||||||||||||', 'T|downstream_gene_variant|MODIFIER|MIR6859-1|ENSG00000278267|Transcript|ENST00000619216.1|miRNA||||||||||rs2691315|1549|-1||HGNC|HGNC:50039|YES||||||||||||Ensembl||G|G||||||0.4105|0.4849|0.2939|0.6052|0.2714|0.3354||||||||||0.2196|0.4324|0.1146|0.1638|0.1285|0.06176|0.06299|0.2151|0.09381|0.2235|0.1329|||||||||||||16.02|1.543896||||||||||||', 'T|non_coding_transcript_exon_variant|MODIFIER|WASH7P|653635|Transcript|NR_024540.1|transcribed_pseudogene|9/11||NR_024540.1:n.1208C>A||1208|||||rs2691315||-1||EntrezGene|HGNC:38034|YES||||||||||||RefSeq||G|G|OK|||||0.4105|0.4849|0.2939|0.6052|0.2714|0.3354||||||||||0.2196|0.4324|0.1146|0.1638|0.1285|0.06176|0.06299|0.2151|0.09381|0.2235|0.1329|||||||||||||16.02|1.543896||||||||||||', 'T|downstream_gene_variant|MODIFIER|DDX11L1|100287102|Transcript|NR_046018.2|transcribed_pseudogene||||||||||rs2691315|1411|1||EntrezGene|HGNC:37102|YES||||||||||||RefSeq||G|G||||||0.4105|0.4849|0.2939|0.6052|0.2714|0.3354||||||||||0.2196|0.4324|0.1146|0.1638|0.1285|0.06176|0.06299|0.2151|0.09381|0.2235|0.1329|||||||||||||16.02|1.543896||||||||||||', 'T|downstream_gene_variant|MODIFIER|MIR6859-1|102466751|Transcript|NR_106918.1|miRNA||||||||||rs2691315|1549|-1||EntrezGene|HGNC:50039|YES||||||||||||RefSeq||G|G||||||0.4105|0.4849|0.2939|0.6052|0.2714|0.3354||||||||||0.2196|0.4324|0.1146|0.1638|0.1285|0.06176|0.06299|0.2151|0.09381|0.2235|0.1329|||||||||||||16.02|1.543896||||||||||||', 'T|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00000344266|CTCF_binding_site||||||||||rs2691315|||||||||||||||||||||||||||0.4105|0.4849|0.2939|0.6052|0.2714|0.3354||||||||||0.2196|0.4324|0.1146|0.1638|0.1285|0.06176|0.06299|0.2151|0.09381|0.2235|0.1329|||||||||||||16.02|1.543896||||||||||||', 'T|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00001164745|enhancer||||||||||rs2691315|||||||||||||||||||||||||||0.4105|0.4849|0.2939|0.6052|0.2714|0.3354||||||||||0.2196|0.4324|0.1146|0.1638|0.1285|0.06176|0.06299|0.2151|0.09381|0.2235|0.1329|||||||||||||16.02|1.543896||||||||||||', 'T|TF_binding_site_variant|MODIFIER|||MotifFeature|ENSM00000173493|||||||||||rs2691315||1|||||||||||||||||||||||||0.4105|0.4849|0.2939|0.6052|0.2714|0.3354||||||||||0.2196|0.4324|0.1146|0.1638|0.1285|0.06176|0.06299|0.2151|0.09381|0.2235|0.1329||||||ENSPFM0042|17|N|-0.005|CTCF|||16.02|1.543896||||||||||||']\n",
            "Sample 200852: 0/1\n",
            "\n",
            "\n",
            "CHROM: chr1\n",
            "POS: 17385\n",
            "ID: rs201535981\n",
            "REF: G\n",
            "ALT: [A]\n",
            "QUAL: None\n",
            "FILTER: []\n",
            "ACMG_class: ['Uncertain%40Significance']\n",
            "ACMG_coding_impact: ['non%40coding']\n",
            "ACMG_rules: ['BP4']\n",
            "ACMG_score: [-0.047]\n",
            "AMP_matches: ['']\n",
            "AMP_rules: ['Type_IV%2CPred_IV']\n",
            "AMP_tier: ['Tier%40IV']\n",
            "AMP_total_samples: [0]\n",
            "AS_FilterStatus: SITE\n",
            "AS_SB_TABLE: 37%2C9|17%2C6\n",
            "DP: 70\n",
            "ECNT: 1\n",
            "GERMQ: 37\n",
            "Gene: ['MIR6859-1', 'WASH7P']\n",
            "MBQ: [29]\n",
            "MFRL: [293]\n",
            "MMQ: [40]\n",
            "MPOS: [19]\n",
            "POPAF: [7.3]\n",
            "TLOD: [45.3]\n",
            "function: ['non-coding%40exon', 'intronic']\n",
            "gnomadExomesEthnic_AC_Hom: [2]\n",
            "gnomadExomes_AC: [9331]\n",
            "gnomadExomes_AC_Hom: [9331]\n",
            "gnomadExomes_AF: [0.173001]\n",
            "gnomadExomes_AF_ethnic: [0.199722]\n",
            "gnomadExomes_AN: [53936]\n",
            "gnomadGenomesAC: [21384]\n",
            "gnomadGenomesAN: [111582]\n",
            "gnomadGenomesEthnic_AC_Hom: [1]\n",
            "gnomadGenomes_AC_Hom: [21384]\n",
            "gnomadGenomes_AF: [0.191644]\n",
            "gnomadGenomes_AF_ethnic: [0.213724]\n",
            "CSQ: ['A|downstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000450305.2|transcribed_unprocessed_pseudogene||||||||||rs201535981|3715|1||HGNC|HGNC:37102|YES||||||||||||Ensembl||G|G||||||||||||0.173|0.1443|0.2215|0.2532|0.02807|0.207|0.1997|0.1992|0.1629|0.1916|0.1547|0.2446|0.2082|0.261|0.03913|0.2582|0.2822|0.2137|0.2152|0.1667|||||||||||||7.934|0.644040||||||||||||', 'A|downstream_gene_variant|MODIFIER|DDX11L2|ENSG00000290825|Transcript|ENST00000456328.2|lncRNA||||||||||rs201535981|2976|1||EntrezGene||YES|||1|||||||||Ensembl||G|G||||||||||||0.173|0.1443|0.2215|0.2532|0.02807|0.207|0.1997|0.1992|0.1629|0.1916|0.1547|0.2446|0.2082|0.261|0.03913|0.2582|0.2822|0.2137|0.2152|0.1667|||||||||||||7.934|0.644040||||||||||||', 'A|splice_polypyrimidine_tract_variant&intron_variant&non_coding_transcript_variant|LOW|WASH7P|ENSG00000227232|Transcript|ENST00000488147.1|unprocessed_pseudogene||5/10|ENST00000488147.1:n.575-17C>T|||||||rs201535981||-1||HGNC|HGNC:38034|YES||||||||||||Ensembl||G|G||||||||||||0.173|0.1443|0.2215|0.2532|0.02807|0.207|0.1997|0.1992|0.1629|0.1916|0.1547|0.2446|0.2082|0.261|0.03913|0.2582|0.2822|0.2137|0.2152|0.1667|||||||||||||7.934|0.644040||||||||||||', 'A|mature_miRNA_variant|MODIFIER|MIR6859-1|ENSG00000278267|Transcript|ENST00000619216.1|miRNA|1/1||ENST00000619216.1:n.52C>T||52|||||rs201535981||-1||HGNC|HGNC:50039|YES||||||||||||Ensembl||G|G||||||||||||0.173|0.1443|0.2215|0.2532|0.02807|0.207|0.1997|0.1992|0.1629|0.1916|0.1547|0.2446|0.2082|0.261|0.03913|0.2582|0.2822|0.2137|0.2152|0.1667|||||||||||||7.934|0.644040||||||||||||', 'A|splice_polypyrimidine_tract_variant&intron_variant&non_coding_transcript_variant|LOW|WASH7P|653635|Transcript|NR_024540.1|transcribed_pseudogene||5/10|NR_024540.1:n.588-17C>T|||||||rs201535981||-1||EntrezGene|HGNC:38034|YES||||||||||||RefSeq||G|G|OK|||||||||||0.173|0.1443|0.2215|0.2532|0.02807|0.207|0.1997|0.1992|0.1629|0.1916|0.1547|0.2446|0.2082|0.261|0.03913|0.2582|0.2822|0.2137|0.2152|0.1667|||||||||||||7.934|0.644040||||||||||||', 'A|downstream_gene_variant|MODIFIER|DDX11L1|100287102|Transcript|NR_046018.2|transcribed_pseudogene||||||||||rs201535981|2976|1||EntrezGene|HGNC:37102|YES||||||||||||RefSeq||G|G||||||||||||0.173|0.1443|0.2215|0.2532|0.02807|0.207|0.1997|0.1992|0.1629|0.1916|0.1547|0.2446|0.2082|0.261|0.03913|0.2582|0.2822|0.2137|0.2152|0.1667|||||||||||||7.934|0.644040||||||||||||', 'A|non_coding_transcript_exon_variant|MODIFIER|MIR6859-1|102466751|Transcript|NR_106918.1|miRNA|1/1||NR_106918.1:n.52C>T||52|||||rs201535981||-1||EntrezGene|HGNC:50039|YES||||||||||||RefSeq||G|G||||||||||||0.173|0.1443|0.2215|0.2532|0.02807|0.207|0.1997|0.1992|0.1629|0.1916|0.1547|0.2446|0.2082|0.261|0.03913|0.2582|0.2822|0.2137|0.2152|0.1667|||||||||||||7.934|0.644040||||||||||||']\n",
            "Sample 200852: 0/1\n",
            "\n",
            "\n",
            "CHROM: chr1\n",
            "POS: 17697\n",
            "ID: rs71260069\n",
            "REF: G\n",
            "ALT: [C]\n",
            "QUAL: None\n",
            "FILTER: []\n",
            "ACMG_class: ['Benign']\n",
            "ACMG_coding_impact: ['non%40coding']\n",
            "ACMG_rules: ['BA1%2CBP4']\n",
            "ACMG_score: [0.193]\n",
            "AMP_matches: ['E%2CV']\n",
            "AMP_rules: ['Freq_II%2CType_IV%2CPred_IV']\n",
            "AMP_tier: ['Tier%40IV']\n",
            "AMP_total_samples: [0]\n",
            "AS_FilterStatus: SITE\n",
            "AS_SB_TABLE: 0%2C39|0%2C8\n",
            "DP: 49\n",
            "ECNT: 1\n",
            "GERMQ: 67\n",
            "Gene: ['MIR6859-1', 'WASH7P']\n",
            "MBQ: [29]\n",
            "MFRL: [331]\n",
            "MMQ: [40]\n",
            "MPOS: [17]\n",
            "POPAF: [7.3]\n",
            "TLOD: [13.2]\n",
            "function: [\"5'flank\", 'non-coding%40exon']\n",
            "gnomadGenomesAC: [17539]\n",
            "gnomadGenomesAN: [116050]\n",
            "gnomadGenomesEthnic_AC_Hom: [2]\n",
            "gnomadGenomes_AC_Hom: [17539]\n",
            "gnomadGenomes_AF: [0.151133]\n",
            "gnomadGenomes_AF_ethnic: [0.178196]\n",
            "CSQ: ['C|downstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000450305.2|transcribed_unprocessed_pseudogene||||||||||rs71260069|4027|1||HGNC|HGNC:37102|YES||||||||||||Ensembl||G|G|||||||||||||||||||||0.1511|0.1238|0.0654|0.1197|0.1641|0.02216|0.2241|0.196|0.1782|0.1404|0.1298|||||||||||||12.16|1.043489||||||||||||', 'C|downstream_gene_variant|MODIFIER|DDX11L2|ENSG00000290825|Transcript|ENST00000456328.2|lncRNA||||||||||rs71260069|3288|1||EntrezGene||YES|||1|||||||||Ensembl||G|G|||||||||||||||||||||0.1511|0.1238|0.0654|0.1197|0.1641|0.02216|0.2241|0.196|0.1782|0.1404|0.1298|||||||||||||12.16|1.043489||||||||||||', 'C|non_coding_transcript_exon_variant|MODIFIER|WASH7P|ENSG00000227232|Transcript|ENST00000488147.1|unprocessed_pseudogene|5/11||ENST00000488147.1:n.483C>G||483|||||rs71260069||-1||HGNC|HGNC:38034|YES||||||||||||Ensembl||G|G|||||||||||||||||||||0.1511|0.1238|0.0654|0.1197|0.1641|0.02216|0.2241|0.196|0.1782|0.1404|0.1298|||||||||||||12.16|1.043489||||||||||||', 'C|upstream_gene_variant|MODIFIER|MIR6859-1|ENSG00000278267|Transcript|ENST00000619216.1|miRNA||||||||||rs71260069|261|-1||HGNC|HGNC:50039|YES||||||||||||Ensembl||G|G|||||||||||||||||||||0.1511|0.1238|0.0654|0.1197|0.1641|0.02216|0.2241|0.196|0.1782|0.1404|0.1298|||||||||||||12.16|1.043489||||||||||||', 'C|non_coding_transcript_exon_variant|MODIFIER|WASH7P|653635|Transcript|NR_024540.1|transcribed_pseudogene|5/11||NR_024540.1:n.496C>G||496|||||rs71260069||-1||EntrezGene|HGNC:38034|YES||||||||||||RefSeq||G|G|OK||||||||||||||||||||0.1511|0.1238|0.0654|0.1197|0.1641|0.02216|0.2241|0.196|0.1782|0.1404|0.1298|||||||||||||12.16|1.043489||||||||||||', 'C|downstream_gene_variant|MODIFIER|DDX11L1|100287102|Transcript|NR_046018.2|transcribed_pseudogene||||||||||rs71260069|3288|1||EntrezGene|HGNC:37102|YES||||||||||||RefSeq||G|G|||||||||||||||||||||0.1511|0.1238|0.0654|0.1197|0.1641|0.02216|0.2241|0.196|0.1782|0.1404|0.1298|||||||||||||12.16|1.043489||||||||||||', 'C|upstream_gene_variant|MODIFIER|MIR6859-1|102466751|Transcript|NR_106918.1|miRNA||||||||||rs71260069|261|-1||EntrezGene|HGNC:50039|YES||||||||||||RefSeq||G|G|||||||||||||||||||||0.1511|0.1238|0.0654|0.1197|0.1641|0.02216|0.2241|0.196|0.1782|0.1404|0.1298|||||||||||||12.16|1.043489||||||||||||']\n",
            "Sample 200852: 0/1\n",
            "\n",
            "\n",
            "CHROM: chr1\n",
            "POS: 133129\n",
            "ID: rs367730352\n",
            "REF: G\n",
            "ALT: [A]\n",
            "QUAL: None\n",
            "FILTER: []\n",
            "ACMG_class: ['Uncertain%40Significance']\n",
            "ACMG_coding_impact: ['non%40coding']\n",
            "ACMG_rules: ['BP4']\n",
            "ACMG_score: [-0.047]\n",
            "AMP_matches: ['']\n",
            "AMP_rules: ['Type_IV%2CPred_IV']\n",
            "AMP_tier: ['Tier%40IV']\n",
            "AMP_total_samples: [0]\n",
            "AS_FilterStatus: SITE\n",
            "AS_SB_TABLE: 0%2C0|18%2C2\n",
            "DP: 22\n",
            "ECNT: 1\n",
            "GERMQ: 92\n",
            "Gene: ['CICP27', 'ENSG00000238009']\n",
            "MBQ: [27]\n",
            "MFRL: [348]\n",
            "MMQ: [33]\n",
            "MPOS: [20]\n",
            "POPAF: [7.3]\n",
            "TLOD: [57.1]\n",
            "function: ['non-coding%40exon', 'intronic']\n",
            "gnomadGenomesAC: [72154]\n",
            "gnomadGenomesAN: [131698]\n",
            "gnomadGenomesEthnic_AC_Hom: [7468]\n",
            "gnomadGenomes_AC_Hom: [72154]\n",
            "gnomadGenomes_AF: [0.547875]\n",
            "gnomadGenomes_AF_ethnic: [0.597389]\n",
            "CSQ: ['A|non_coding_transcript_exon_variant|MODIFIER|CICP27|ENSG00000233750|Transcript|ENST00000442987.3|processed_pseudogene|1/1||ENST00000442987.3:n.2105G>A||2105|||||rs367730352||1||HGNC|HGNC:48835|YES||||||||||||Ensembl||G|G|||||||||||||||||||||0.5479|0.4621|0.5467|0.5007|0.5689|0.4942|0.6321|0.5|0.5974|0.5382|0.5257|||||||||||||5.750|0.429409||||||||||||', 'A|intron_variant&non_coding_transcript_variant|MODIFIER||ENSG00000238009|Transcript|ENST00000453576.2|lncRNA||1/1|ENST00000453576.2:n.193+245C>T|||||||rs367730352||-1|||||||2|||||||||Ensembl||G|G|||||||||||||||||||||0.5479|0.4621|0.5467|0.5007|0.5689|0.4942|0.6321|0.5|0.5974|0.5382|0.5257|||||||||||||5.750|0.429409||||||||||||', 'A|upstream_gene_variant|MODIFIER||ENSG00000238009|Transcript|ENST00000471248.1|lncRNA||||||||||rs367730352|3956|-1|||||||5|||||||||Ensembl||G|G|||||||||||||||||||||0.5479|0.4621|0.5467|0.5007|0.5689|0.4942|0.6321|0.5|0.5974|0.5382|0.5257|||||||||||||5.750|0.429409||||||||||||', 'A|upstream_gene_variant|MODIFIER||ENSG00000238009|Transcript|ENST00000477740.5|lncRNA||||||||||rs367730352|3912|-1||||YES|||5|||||||||Ensembl||G|G|||||||||||||||||||||0.5479|0.4621|0.5467|0.5007|0.5689|0.4942|0.6321|0.5|0.5974|0.5382|0.5257|||||||||||||5.750|0.429409||||||||||||', 'A|downstream_gene_variant|MODIFIER||ENSG00000268903|Transcript|ENST00000494149.2|processed_pseudogene||||||||||rs367730352|2012|-1||||YES||||||||||||Ensembl||G|G|||||||||||||||||||||0.5479|0.4621|0.5467|0.5007|0.5689|0.4942|0.6321|0.5|0.5974|0.5382|0.5257|||||||||||||5.750|0.429409||||||||||||', 'A|downstream_gene_variant|MODIFIER||ENSG00000269981|Transcript|ENST00000595919.1|processed_pseudogene||||||||||rs367730352|4553|-1||||YES||||||||||||Ensembl||G|G|||||||||||||||||||||0.5479|0.4621|0.5467|0.5007|0.5689|0.4942|0.6321|0.5|0.5974|0.5382|0.5257|||||||||||||5.750|0.429409||||||||||||', 'A|intron_variant&non_coding_transcript_variant|MODIFIER||ENSG00000238009|Transcript|ENST00000610542.1|lncRNA||1/3|ENST00000610542.1:n.350+245C>T|||||||rs367730352||-1|||||||5|||||||||Ensembl||G|G|||||||||||||||||||||0.5479|0.4621|0.5467|0.5007|0.5689|0.4942|0.6321|0.5|0.5974|0.5382|0.5257|||||||||||||5.750|0.429409||||||||||||', 'A|downstream_gene_variant|MODIFIER|LOC729737|729737|Transcript|NR_039983.2|lncRNA||||||||||rs367730352|1644|-1||EntrezGene||YES||||||||||||RefSeq||G|G|||||||||||||||||||||0.5479|0.4621|0.5467|0.5007|0.5689|0.4942|0.6321|0.5|0.5974|0.5382|0.5257|||||||||||||5.750|0.429409||||||||||||', 'A|intron_variant&non_coding_transcript_variant|MODIFIER|LOC124900384|124900384|Transcript|XR_001737579.3|lncRNA||5/5|XR_001737579.3:n.1968-12197C>T|||||||rs367730352||-1||EntrezGene||YES||||||||||||RefSeq||G|G|||||||||||||||||||||0.5479|0.4621|0.5467|0.5007|0.5689|0.4942|0.6321|0.5|0.5974|0.5382|0.5257|||||||||||||5.750|0.429409||||||||||||']\n",
            "Sample 200852: 0/1\n",
            "\n",
            "\n",
            "CHROM: chr1\n",
            "POS: 183629\n",
            "ID: rs71267774\n",
            "REF: G\n",
            "ALT: [A]\n",
            "QUAL: None\n",
            "FILTER: []\n",
            "ACMG_class: ['Benign']\n",
            "ACMG_coding_impact: ['non%40coding']\n",
            "ACMG_rules: ['BA1%2CBP4']\n",
            "ACMG_score: [0.193]\n",
            "AMP_matches: ['E%2CV']\n",
            "AMP_rules: ['Freq_II%2CType_IV%2CPred_IV']\n",
            "AMP_tier: ['Tier%40IV']\n",
            "AMP_total_samples: [0]\n",
            "AS_FilterStatus: SITE\n",
            "AS_SB_TABLE: 50%2C53|19%2C14\n",
            "DP: 139\n",
            "ECNT: 4\n",
            "GERMQ: 93\n",
            "Gene: ['DDX11L17']\n",
            "MBQ: [29]\n",
            "MFRL: [366]\n",
            "MMQ: [46]\n",
            "MPOS: [29]\n",
            "POPAF: [7.3]\n",
            "TLOD: [68.5]\n",
            "function: ['intronic']\n",
            "gnomadGenomesAC: [14902]\n",
            "gnomadGenomesAN: [111796]\n",
            "gnomadGenomesEthnic_AC_Hom: [1]\n",
            "gnomadGenomes_AC_Hom: [14902]\n",
            "gnomadGenomes_AF: [0.133296]\n",
            "gnomadGenomes_AF_ethnic: [0.220891]\n",
            "CSQ: ['A|downstream_gene_variant|MODIFIER|MIR6859-2|ENSG00000273874|Transcript|ENST00000612080.1|miRNA||||||||||rs71267774|4262|-1||HGNC|HGNC:49954|YES||||||||||||Ensembl||G|G|||||||||||||||||||||0.1333|0.03798|0.1565|0.1196|0.1272|0.01827|0.2103|0.1154|0.2209|0.1247|0.1071|||||||||||||0.798|-0.107991||||||||||||', 'A|downstream_gene_variant|MODIFIER|WASH9P|ENSG00000279457|Transcript|ENST00000623083.4|unprocessed_pseudogene||||||||||rs71267774|1588|-1||HGNC|HGNC:53981|YES||||||||||||Ensembl||G|G|||||||||||||||||||||0.1333|0.03798|0.1565|0.1196|0.1272|0.01827|0.2103|0.1154|0.2209|0.1247|0.1071|||||||||||||0.798|-0.107991||||||||||||', 'A|intron_variant&non_coding_transcript_variant|MODIFIER|DDX11L17|ENSG00000279928|Transcript|ENST00000624431.2|unprocessed_pseudogene||3/4|ENST00000624431.2:n.214+58G>A|||||||rs71267774||1||HGNC|HGNC:55080|YES||||||||||||Ensembl||G|G|||||||||||||||||||||0.1333|0.03798|0.1565|0.1196|0.1272|0.01827|0.2103|0.1154|0.2209|0.1247|0.1071|||||||||||||0.798|-0.107991||||||||||||', 'A|intron_variant&non_coding_transcript_variant|MODIFIER|DDX11L17|102725121|Transcript|NR_148357.1|transcribed_pseudogene||2/2|NR_148357.1:n.469-111G>A|||||||rs71267774||1||EntrezGene|HGNC:55080|YES||||||||||||RefSeq||G|G|||||||||||||||||||||0.1333|0.03798|0.1565|0.1196|0.1272|0.01827|0.2103|0.1154|0.2209|0.1247|0.1071|||||||||||||0.798|-0.107991||||||||||||', 'A|downstream_gene_variant|MODIFIER|LOC100996442|100996442|Transcript|XR_007065318.1|lncRNA||||||||||rs71267774|1287|-1||EntrezGene||YES||||||||||||RefSeq||G|G|||||||||||||||||||||0.1333|0.03798|0.1565|0.1196|0.1272|0.01827|0.2103|0.1154|0.2209|0.1247|0.1071|||||||||||||0.798|-0.107991||||||||||||', 'A|upstream_gene_variant|MODIFIER|LOC124903814|124903814|Transcript|XR_007065332.1|lncRNA||||||||||rs71267774|3524|-1||EntrezGene||YES||||||||||||RefSeq||G|G|||||||||||||||||||||0.1333|0.03798|0.1565|0.1196|0.1272|0.01827|0.2103|0.1154|0.2209|0.1247|0.1071|||||||||||||0.798|-0.107991||||||||||||', 'A|upstream_gene_variant|MODIFIER|LOC124903814|124903814|Transcript|XR_007065333.1|lncRNA||||||||||rs71267774|3524|-1||EntrezGene||||||||||||||RefSeq||G|G|||||||||||||||||||||0.1333|0.03798|0.1565|0.1196|0.1272|0.01827|0.2103|0.1154|0.2209|0.1247|0.1071|||||||||||||0.798|-0.107991||||||||||||', 'A|upstream_gene_variant|MODIFIER|LOC124903814|124903814|Transcript|XR_007065334.1|lncRNA||||||||||rs71267774|3524|-1||EntrezGene||||||||||||||RefSeq||G|G|||||||||||||||||||||0.1333|0.03798|0.1565|0.1196|0.1272|0.01827|0.2103|0.1154|0.2209|0.1247|0.1071|||||||||||||0.798|-0.107991||||||||||||']\n",
            "Sample 200852: 0/1\n",
            "\n",
            "\n",
            "CHROM: chr1\n",
            "POS: 184267\n",
            "ID: rs1396210256\n",
            "REF: C\n",
            "ALT: [T]\n",
            "QUAL: None\n",
            "FILTER: []\n",
            "ACMG_class: ['Likely%40Benign']\n",
            "ACMG_coding_impact: ['non%40coding']\n",
            "ACMG_rules: ['BS1%2CBP4']\n",
            "ACMG_score: [0.193]\n",
            "AMP_matches: ['E%2CV']\n",
            "AMP_rules: ['Freq_II%2CType_IV%2CPred_IV']\n",
            "AMP_tier: ['Tier%40IV']\n",
            "AMP_total_samples: [0]\n",
            "AS_FilterStatus: SITE\n",
            "AS_SB_TABLE: 32%2C33|3%2C4\n",
            "DP: 78\n",
            "ECNT: 1\n",
            "GERMQ: 93\n",
            "Gene: ['DDX11L17']\n",
            "MBQ: [28]\n",
            "MFRL: [237]\n",
            "MMQ: [39]\n",
            "MPOS: [13]\n",
            "POPAF: [7.3]\n",
            "TLOD: [11.4]\n",
            "function: ['non-coding%40exon', \"3'flank\"]\n",
            "gnomadGenomesAC: [887]\n",
            "gnomadGenomesAN: [144780]\n",
            "gnomadGenomesEthnic_AC_Hom: [0]\n",
            "gnomadGenomes_AC_Hom: [887]\n",
            "gnomadGenomes_AF: [0.00612654]\n",
            "gnomadGenomes_AF_ethnic: [0.00631038]\n",
            "CSQ: ['T|downstream_gene_variant|MODIFIER|MIR6859-2|ENSG00000273874|Transcript|ENST00000612080.1|miRNA||||||||||rs1396210256|3624|-1||HGNC|HGNC:49954|YES||||||||||||Ensembl||C|C|||||||||||||||||||||0.006127|0.002936|0.01439|0.007364|0.009034|0.0006326|0.01197|0.01316|0.00631|0.006972|0.01678|||||||||||||1.100|-0.040819||||||||||||', 'T|downstream_gene_variant|MODIFIER|WASH9P|ENSG00000279457|Transcript|ENST00000623083.4|unprocessed_pseudogene||||||||||rs1396210256|950|-1||HGNC|HGNC:53981|YES||||||||||||Ensembl||C|C|||||||||||||||||||||0.006127|0.002936|0.01439|0.007364|0.009034|0.0006326|0.01197|0.01316|0.00631|0.006972|0.01678|||||||||||||1.100|-0.040819||||||||||||', 'T|downstream_gene_variant|MODIFIER|DDX11L17|ENSG00000279928|Transcript|ENST00000624431.2|unprocessed_pseudogene||||||||||rs1396210256|93|1||HGNC|HGNC:55080|YES||||||||||||Ensembl||C|C|||||||||||||||||||||0.006127|0.002936|0.01439|0.007364|0.009034|0.0006326|0.01197|0.01316|0.00631|0.006972|0.01678|||||||||||||1.100|-0.040819||||||||||||', 'T|non_coding_transcript_exon_variant|MODIFIER|DDX11L17|102725121|Transcript|NR_148357.1|transcribed_pseudogene|3/3||NR_148357.1:n.996C>T||996|||||rs1396210256||1||EntrezGene|HGNC:55080|YES||||||||||||RefSeq||C|C|||||||||||||||||||||0.006127|0.002936|0.01439|0.007364|0.009034|0.0006326|0.01197|0.01316|0.00631|0.006972|0.01678|||||||||||||1.100|-0.040819||||||||||||', 'T|downstream_gene_variant|MODIFIER|LOC100996442|100996442|Transcript|XR_007065318.1|lncRNA||||||||||rs1396210256|649|-1||EntrezGene||YES||||||||||||RefSeq||C|C|||||||||||||||||||||0.006127|0.002936|0.01439|0.007364|0.009034|0.0006326|0.01197|0.01316|0.00631|0.006972|0.01678|||||||||||||1.100|-0.040819||||||||||||', 'T|upstream_gene_variant|MODIFIER|LOC124903814|124903814|Transcript|XR_007065332.1|lncRNA||||||||||rs1396210256|4162|-1||EntrezGene||YES||||||||||||RefSeq||C|C|||||||||||||||||||||0.006127|0.002936|0.01439|0.007364|0.009034|0.0006326|0.01197|0.01316|0.00631|0.006972|0.01678|||||||||||||1.100|-0.040819||||||||||||', 'T|upstream_gene_variant|MODIFIER|LOC124903814|124903814|Transcript|XR_007065333.1|lncRNA||||||||||rs1396210256|4162|-1||EntrezGene||||||||||||||RefSeq||C|C|||||||||||||||||||||0.006127|0.002936|0.01439|0.007364|0.009034|0.0006326|0.01197|0.01316|0.00631|0.006972|0.01678|||||||||||||1.100|-0.040819||||||||||||', 'T|upstream_gene_variant|MODIFIER|LOC124903814|124903814|Transcript|XR_007065334.1|lncRNA||||||||||rs1396210256|4162|-1||EntrezGene||||||||||||||RefSeq||C|C|||||||||||||||||||||0.006127|0.002936|0.01439|0.007364|0.009034|0.0006326|0.01197|0.01316|0.00631|0.006972|0.01678|||||||||||||1.100|-0.040819||||||||||||']\n",
            "Sample 200852: 0/1\n",
            "\n",
            "\n",
            "CHROM: chr1\n",
            "POS: 184413\n",
            "ID: rs1441656517\n",
            "REF: C\n",
            "ALT: [A]\n",
            "QUAL: None\n",
            "FILTER: []\n",
            "ACMG_class: ['Uncertain%40Significance']\n",
            "ACMG_coding_impact: ['non%40coding']\n",
            "ACMG_rules: ['BP4']\n",
            "ACMG_score: [0.193]\n",
            "AMP_matches: ['V']\n",
            "AMP_rules: ['Freq_II%2CType_IV%2CPred_IV']\n",
            "AMP_tier: ['Tier%40IV']\n",
            "AMP_total_samples: [0]\n",
            "AS_FilterStatus: SITE\n",
            "AS_SB_TABLE: 68%2C49|12%2C10\n",
            "DP: 153\n",
            "ECNT: 1\n",
            "GERMQ: 93\n",
            "Gene: ['DDX11L17']\n",
            "MBQ: [28]\n",
            "MFRL: [228]\n",
            "MMQ: [37]\n",
            "MPOS: [23]\n",
            "POPAF: [7.3]\n",
            "TLOD: [42.4]\n",
            "function: ['non-coding%40exon', \"3'flank\"]\n",
            "gnomadGenomesAC: [33879]\n",
            "gnomadGenomesAN: [114958]\n",
            "gnomadGenomesEthnic_AC_Hom: [18]\n",
            "gnomadGenomes_AC_Hom: [33879]\n",
            "gnomadGenomes_AF: [0.294708]\n",
            "gnomadGenomes_AF_ethnic: [0.335956]\n",
            "CSQ: ['A|downstream_gene_variant|MODIFIER|MIR6859-2|ENSG00000273874|Transcript|ENST00000612080.1|miRNA||||||||||rs1441656517|3478|-1||HGNC|HGNC:49954|YES||||||||||||Ensembl||C|C|||||||||||||||||||||0.2947|0.2|0.4231|0.364|0.3479|0.3555|0.2805|0.3261|0.336|0.3249|0.2691|||||||||||||1.513|0.026980||||||||||||', 'A|downstream_gene_variant|MODIFIER|WASH9P|ENSG00000279457|Transcript|ENST00000623083.4|unprocessed_pseudogene||||||||||rs1441656517|804|-1||HGNC|HGNC:53981|YES||||||||||||Ensembl||C|C|||||||||||||||||||||0.2947|0.2|0.4231|0.364|0.3479|0.3555|0.2805|0.3261|0.336|0.3249|0.2691|||||||||||||1.513|0.026980||||||||||||', 'A|downstream_gene_variant|MODIFIER|DDX11L17|ENSG00000279928|Transcript|ENST00000624431.2|unprocessed_pseudogene||||||||||rs1441656517|239|1||HGNC|HGNC:55080|YES||||||||||||Ensembl||C|C|||||||||||||||||||||0.2947|0.2|0.4231|0.364|0.3479|0.3555|0.2805|0.3261|0.336|0.3249|0.2691|||||||||||||1.513|0.026980||||||||||||', 'A|non_coding_transcript_exon_variant|MODIFIER|DDX11L17|102725121|Transcript|NR_148357.1|transcribed_pseudogene|3/3||NR_148357.1:n.1142C>A||1142|||||rs1441656517||1||EntrezGene|HGNC:55080|YES||||||||||||RefSeq||C|C|||||||||||||||||||||0.2947|0.2|0.4231|0.364|0.3479|0.3555|0.2805|0.3261|0.336|0.3249|0.2691|||||||||||||1.513|0.026980||||||||||||', 'A|downstream_gene_variant|MODIFIER|LOC100996442|100996442|Transcript|XR_007065318.1|lncRNA||||||||||rs1441656517|503|-1||EntrezGene||YES||||||||||||RefSeq||C|C|||||||||||||||||||||0.2947|0.2|0.4231|0.364|0.3479|0.3555|0.2805|0.3261|0.336|0.3249|0.2691|||||||||||||1.513|0.026980||||||||||||', 'A|upstream_gene_variant|MODIFIER|LOC124903814|124903814|Transcript|XR_007065332.1|lncRNA||||||||||rs1441656517|4308|-1||EntrezGene||YES||||||||||||RefSeq||C|C|||||||||||||||||||||0.2947|0.2|0.4231|0.364|0.3479|0.3555|0.2805|0.3261|0.336|0.3249|0.2691|||||||||||||1.513|0.026980||||||||||||', 'A|upstream_gene_variant|MODIFIER|LOC124903814|124903814|Transcript|XR_007065333.1|lncRNA||||||||||rs1441656517|4308|-1||EntrezGene||||||||||||||RefSeq||C|C|||||||||||||||||||||0.2947|0.2|0.4231|0.364|0.3479|0.3555|0.2805|0.3261|0.336|0.3249|0.2691|||||||||||||1.513|0.026980||||||||||||', 'A|upstream_gene_variant|MODIFIER|LOC124903814|124903814|Transcript|XR_007065334.1|lncRNA||||||||||rs1441656517|4308|-1||EntrezGene||||||||||||||RefSeq||C|C|||||||||||||||||||||0.2947|0.2|0.4231|0.364|0.3479|0.3555|0.2805|0.3261|0.336|0.3249|0.2691|||||||||||||1.513|0.026980||||||||||||']\n",
            "Sample 200852: 0/1\n",
            "\n",
            "\n",
            "CHROM: chr1\n",
            "POS: 186338\n",
            "ID: rs1490434665\n",
            "REF: T\n",
            "ALT: [G]\n",
            "QUAL: None\n",
            "FILTER: []\n",
            "ACMG_class: ['Uncertain%40Significance']\n",
            "ACMG_coding_impact: ['non%40coding']\n",
            "ACMG_rules: ['BP4']\n",
            "ACMG_score: [0.193]\n",
            "AMP_matches: ['V']\n",
            "AMP_rules: ['Freq_II%2CType_IV%2CPred_IV']\n",
            "AMP_tier: ['Tier%40IV']\n",
            "AMP_total_samples: [0]\n",
            "AS_FilterStatus: SITE\n",
            "AS_SB_TABLE: 32%2C17|6%2C0\n",
            "DP: 60\n",
            "ECNT: 3\n",
            "GERMQ: 93\n",
            "Gene: ['WASH9P']\n",
            "MBQ: [30]\n",
            "MFRL: [398]\n",
            "MMQ: [38]\n",
            "MPOS: [26]\n",
            "POPAF: [7.3]\n",
            "TLOD: [10.6]\n",
            "function: ['non-coding%40exon']\n",
            "gnomadGenomesAC: [12286]\n",
            "gnomadGenomesAN: [90730]\n",
            "gnomadGenomesEthnic_AC_Hom: [8]\n",
            "gnomadGenomes_AC_Hom: [12286]\n",
            "gnomadGenomes_AF: [0.135413]\n",
            "gnomadGenomes_AF_ethnic: [0.114409]\n",
            "CSQ: ['G|downstream_gene_variant|MODIFIER|MIR6859-2|ENSG00000273874|Transcript|ENST00000612080.1|miRNA||||||||||rs1490434665|1553|-1||HGNC|HGNC:49954|YES||||||||||||Ensembl||T|T|||||||||||||||||||||0.1354|0.1788|0.1337|0.1068|0.08754|0.1299|0.2021|0.08|0.1144|0.1153|0.0768|||||||||||||18.84|1.944793||||||||||||', 'G|non_coding_transcript_exon_variant|MODIFIER|WASH9P|ENSG00000279457|Transcript|ENST00000623083.4|unprocessed_pseudogene|8/10||ENST00000623083.4:n.1173A>C||1173|||||rs1490434665||-1||HGNC|HGNC:53981|YES||||||||||||Ensembl||T|T|||||||||||||||||||||0.1354|0.1788|0.1337|0.1068|0.08754|0.1299|0.2021|0.08|0.1144|0.1153|0.0768|||||||||||||18.84|1.944793||||||||||||', 'G|downstream_gene_variant|MODIFIER|DDX11L17|ENSG00000279928|Transcript|ENST00000624431.2|unprocessed_pseudogene||||||||||rs1490434665|2164|1||HGNC|HGNC:55080|YES||||||||||||Ensembl||T|T|||||||||||||||||||||0.1354|0.1788|0.1337|0.1068|0.08754|0.1299|0.2021|0.08|0.1144|0.1153|0.0768|||||||||||||18.84|1.944793||||||||||||', 'G|downstream_gene_variant|MODIFIER|DDX11L17|102725121|Transcript|NR_148357.1|transcribed_pseudogene||||||||||rs1490434665|1460|1||EntrezGene|HGNC:55080|YES||||||||||||RefSeq||T|T|||||||||||||||||||||0.1354|0.1788|0.1337|0.1068|0.08754|0.1299|0.2021|0.08|0.1144|0.1153|0.0768|||||||||||||18.84|1.944793||||||||||||', 'G|non_coding_transcript_exon_variant|MODIFIER|LOC100996442|100996442|Transcript|XR_007065318.1|lncRNA|9/11||XR_007065318.1:n.1242A>C||1242|||||rs1490434665||-1||EntrezGene||YES||||||||||||RefSeq||T|T|||||||||||||||||||||0.1354|0.1788|0.1337|0.1068|0.08754|0.1299|0.2021|0.08|0.1144|0.1153|0.0768|||||||||||||18.84|1.944793||||||||||||', 'G|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00000344295|CTCF_binding_site||||||||||rs1490434665||||||||||||||||||||||||||||||||||||||||||0.1354|0.1788|0.1337|0.1068|0.08754|0.1299|0.2021|0.08|0.1144|0.1153|0.0768|||||||||||||18.84|1.944793||||||||||||', 'G|TF_binding_site_variant|MODIFIER|||MotifFeature|ENSM00000225283|||||||||||rs1490434665||1||||||||||||||||||||||||||||||||||||||||0.1354|0.1788|0.1337|0.1068|0.08754|0.1299|0.2021|0.08|0.1144|0.1153|0.0768||||||ENSPFM0042|14|Y|0.035|CTCF|||18.84|1.944793||||||||||||']\n",
            "Sample 200852: 0/1\n",
            "\n",
            "\n",
            "CHROM: chr1\n",
            "POS: 186835\n",
            "ID: rs1364805962\n",
            "REF: A\n",
            "ALT: [G]\n",
            "QUAL: None\n",
            "FILTER: []\n",
            "ACMG_class: ['Uncertain%40Significance']\n",
            "ACMG_coding_impact: ['non%40coding']\n",
            "ACMG_rules: ['BP4']\n",
            "ACMG_score: [-0.047]\n",
            "AMP_matches: ['']\n",
            "AMP_rules: ['Type_IV%2CPred_IV']\n",
            "AMP_tier: ['Tier%40IV']\n",
            "AMP_total_samples: [0]\n",
            "AS_FilterStatus: SITE\n",
            "AS_SB_TABLE: 6%2C2|4%2C2\n",
            "DP: 15\n",
            "ECNT: 1\n",
            "GERMQ: 32\n",
            "Gene: ['WASH9P']\n",
            "MBQ: [29]\n",
            "MFRL: [332]\n",
            "MMQ: [36]\n",
            "MPOS: [5]\n",
            "POPAF: [7.3]\n",
            "TLOD: [13.9]\n",
            "function: ['intronic']\n",
            "gnomadGenomesAC: [97257]\n",
            "gnomadGenomesAN: [150682]\n",
            "gnomadGenomesEthnic_AC_Hom: [4494]\n",
            "gnomadGenomes_AC_Hom: [97257]\n",
            "gnomadGenomes_AF: [0.645445]\n",
            "gnomadGenomes_AF_ethnic: [0.566261]\n",
            "CSQ: ['G|downstream_gene_variant|MODIFIER|MIR6859-2|ENSG00000273874|Transcript|ENST00000612080.1|miRNA||||||||||rs1364805962|1056|-1||HGNC|HGNC:49954|YES||||||||||||Ensembl||A|A|||||||||||||||||||||0.6454|0.8165|0.5659|0.6155|0.5712|0.7173|0.5622|0.6019|0.5663|0.6252|0.5847|||||||||||||4.041|0.278704||||||||||||', 'G|intron_variant&non_coding_transcript_variant|MODIFIER|WASH9P|ENSG00000279457|Transcript|ENST00000623083.4|unprocessed_pseudogene||7/9|ENST00000623083.4:n.1041+294T>C|||||||rs1364805962||-1||HGNC|HGNC:53981|YES||||||||||||Ensembl||A|A|||||||||||||||||||||0.6454|0.8165|0.5659|0.6155|0.5712|0.7173|0.5622|0.6019|0.5663|0.6252|0.5847|||||||||||||4.041|0.278704||||||||||||', 'G|downstream_gene_variant|MODIFIER|DDX11L17|ENSG00000279928|Transcript|ENST00000624431.2|unprocessed_pseudogene||||||||||rs1364805962|2661|1||HGNC|HGNC:55080|YES||||||||||||Ensembl||A|A|||||||||||||||||||||0.6454|0.8165|0.5659|0.6155|0.5712|0.7173|0.5622|0.6019|0.5663|0.6252|0.5847|||||||||||||4.041|0.278704||||||||||||', 'G|downstream_gene_variant|MODIFIER|DDX11L17|102725121|Transcript|NR_148357.1|transcribed_pseudogene||||||||||rs1364805962|1957|1||EntrezGene|HGNC:55080|YES||||||||||||RefSeq||A|A|||||||||||||||||||||0.6454|0.8165|0.5659|0.6155|0.5712|0.7173|0.5622|0.6019|0.5663|0.6252|0.5847|||||||||||||4.041|0.278704||||||||||||', 'G|intron_variant&non_coding_transcript_variant|MODIFIER|LOC100996442|100996442|Transcript|XR_007065318.1|lncRNA||8/10|XR_007065318.1:n.1110+294T>C|||||||rs1364805962||-1||EntrezGene||YES||||||||||||RefSeq||A|A|||||||||||||||||||||0.6454|0.8165|0.5659|0.6155|0.5712|0.7173|0.5622|0.6019|0.5663|0.6252|0.5847|||||||||||||4.041|0.278704||||||||||||', 'G|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00000344295|CTCF_binding_site||||||||||rs1364805962||||||||||||||||||||||||||||||||||||||||||0.6454|0.8165|0.5659|0.6155|0.5712|0.7173|0.5622|0.6019|0.5663|0.6252|0.5847|||||||||||||4.041|0.278704||||||||||||']\n",
            "Sample 200852: 0/1\n",
            "\n",
            "\n",
            "CHROM: chr1\n",
            "POS: 187102\n",
            "ID: rs1221392627\n",
            "REF: C\n",
            "ALT: [G]\n",
            "QUAL: None\n",
            "FILTER: []\n",
            "ACMG_class: ['Uncertain%40Significance']\n",
            "ACMG_coding_impact: ['non%40coding']\n",
            "ACMG_rules: ['BP4']\n",
            "ACMG_score: [0.193]\n",
            "AMP_matches: ['V']\n",
            "AMP_rules: ['Freq_II%2CType_IV%2CPred_IV']\n",
            "AMP_tier: ['Tier%40IV']\n",
            "AMP_total_samples: [0]\n",
            "AS_FilterStatus: SITE\n",
            "AS_SB_TABLE: 63%2C48|19%2C14\n",
            "DP: 149\n",
            "ECNT: 4\n",
            "GERMQ: 93\n",
            "Gene: ['WASH9P']\n",
            "MBQ: [30]\n",
            "MFRL: [251]\n",
            "MMQ: [40]\n",
            "MPOS: [15]\n",
            "POPAF: [7.3]\n",
            "TLOD: [70.9]\n",
            "function: ['intronic']\n",
            "gnomadGenomesAC: [54488]\n",
            "gnomadGenomesAN: [119036]\n",
            "gnomadGenomesEthnic_AC_Hom: [4]\n",
            "gnomadGenomes_AC_Hom: [54488]\n",
            "gnomadGenomes_AF: [0.457744]\n",
            "gnomadGenomes_AF_ethnic: [0.445951]\n",
            "CSQ: ['G|downstream_gene_variant|MODIFIER|MIR6859-2|ENSG00000273874|Transcript|ENST00000612080.1|miRNA||||||||||rs1221392627|789|-1||HGNC|HGNC:49954|YES||||||||||||Ensembl||C|C|||||||||||||||||||||0.4577|0.4973|0.4637|0.4473|0.4448|0.3969|0.4094|0.4113|0.446|0.4559|0.4219|||||||||||||0.531|-0.192081||||||||||||', 'G|intron_variant&non_coding_transcript_variant|MODIFIER|WASH9P|ENSG00000279457|Transcript|ENST00000623083.4|unprocessed_pseudogene||7/9|ENST00000623083.4:n.1041+27G>C|||||||rs1221392627||-1||HGNC|HGNC:53981|YES||||||||||||Ensembl||C|C|||||||||||||||||||||0.4577|0.4973|0.4637|0.4473|0.4448|0.3969|0.4094|0.4113|0.446|0.4559|0.4219|||||||||||||0.531|-0.192081||||||||||||', 'G|downstream_gene_variant|MODIFIER|DDX11L17|ENSG00000279928|Transcript|ENST00000624431.2|unprocessed_pseudogene||||||||||rs1221392627|2928|1||HGNC|HGNC:55080|YES||||||||||||Ensembl||C|C|||||||||||||||||||||0.4577|0.4973|0.4637|0.4473|0.4448|0.3969|0.4094|0.4113|0.446|0.4559|0.4219|||||||||||||0.531|-0.192081||||||||||||', 'G|downstream_gene_variant|MODIFIER|DDX11L17|102725121|Transcript|NR_148357.1|transcribed_pseudogene||||||||||rs1221392627|2224|1||EntrezGene|HGNC:55080|YES||||||||||||RefSeq||C|C|||||||||||||||||||||0.4577|0.4973|0.4637|0.4473|0.4448|0.3969|0.4094|0.4113|0.446|0.4559|0.4219|||||||||||||0.531|-0.192081||||||||||||', 'G|intron_variant&non_coding_transcript_variant|MODIFIER|LOC100996442|100996442|Transcript|XR_007065318.1|lncRNA||8/10|XR_007065318.1:n.1110+27G>C|||||||rs1221392627||-1||EntrezGene||YES||||||||||||RefSeq||C|C|||||||||||||||||||||0.4577|0.4973|0.4637|0.4473|0.4448|0.3969|0.4094|0.4113|0.446|0.4559|0.4219|||||||||||||0.531|-0.192081||||||||||||', 'G|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00000344295|CTCF_binding_site||||||||||rs1221392627||||||||||||||||||||||||||||||||||||||||||0.4577|0.4973|0.4637|0.4473|0.4448|0.3969|0.4094|0.4113|0.446|0.4559|0.4219|||||||||||||0.531|-0.192081||||||||||||']\n",
            "Sample 200852: 0/1\n",
            "\n",
            "\n"
          ]
        }
      ],
      "source": [
        "i=0\n",
        "with open(vcf_file, 'rb') as vcf_file_binary:\n",
        "    vcf_reader = vcf.Reader(vcf_file_binary)\n",
        "\n",
        "    for meta_info in vcf_reader.metadata:\n",
        "        print(\"Metadata:\", meta_info)\n",
        "\n",
        "    column_headers = vcf_reader.infos.keys()\n",
        "    print(\"Field Names:\", column_headers)\n",
        "\n",
        "    for record in vcf_reader:\n",
        "        i+=1\n",
        "        if i>10:\n",
        "            break\n",
        "        print(\"CHROM:\", record.CHROM)\n",
        "        print(\"POS:\", record.POS)\n",
        "        print(\"ID:\", record.ID)\n",
        "        print(\"REF:\", record.REF)\n",
        "        print(\"ALT:\", record.ALT)\n",
        "        print(\"QUAL:\", record.QUAL)\n",
        "        print(\"FILTER:\", record.FILTER)\n",
        "\n",
        "        for field in column_headers:\n",
        "            info_value = record.INFO.get(field)\n",
        "            if info_value is not None:\n",
        "                print(f\"{field}:\", info_value)\n",
        "\n",
        "        for sample in record.samples:\n",
        "            print(f\"Sample {sample.sample}: {sample['GT']}\")\n",
        "\n",
        "        print(\"\\n\")"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "#### Reading CSQ description "
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 31,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "['Allele', 'Consequence', 'IMPACT', 'SYMBOL', 'Gene', 'Feature_type', 'Feature', 'BIOTYPE', 'EXON', 'INTRON', 'HGVSc', 'HGVSp', 'cDNA_position', 'CDS_position', 'Protein_position', 'Amino_acids', 'Codons', 'Existing_variation', 'DISTANCE', 'STRAND', 'FLAGS', 'SYMBOL_SOURCE', 'HGNC_ID', 'CANONICAL', 'MANE_SELECT', 'MANE_PLUS_CLINICAL', 'TSL', 'APPRIS', 'CCDS', 'ENSP', 'SWISSPROT', 'TREMBL', 'UNIPARC', 'UNIPROT_ISOFORM', 'REFSEQ_MATCH', 'SOURCE', 'REFSEQ_OFFSET', 'GIVEN_REF', 'USED_REF', 'BAM_EDIT', 'SIFT', 'PolyPhen', 'DOMAINS', 'HGVS_OFFSET', 'AF', 'AFR_AF', 'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF', 'gnomADe_AF', 'gnomADe_AFR_AF', 'gnomADe_AMR_AF', 'gnomADe_ASJ_AF', 'gnomADe_EAS_AF', 'gnomADe_FIN_AF', 'gnomADe_NFE_AF', 'gnomADe_OTH_AF', 'gnomADe_SAS_AF', 'gnomADg_AF', 'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'CLIN_SIG', 'SOMATIC', 'PHENO', 'PUBMED', 'VAR_SYNONYMS', 'MOTIF_NAME', 'MOTIF_POS', 'HIGH_INF_POS', 'MOTIF_SCORE_CHANGE', 'TRANSCRIPTION_FACTORS', 'EVE_CLASS', 'EVE_SCORE', 'CADD_PHRED', 'CADD_RAW', 'SpliceAI_pred_DP_AG', 'SpliceAI_pred_DP_AL', 'SpliceAI_pred_DP_DG', 'SpliceAI_pred_DP_DL', 'SpliceAI_pred_DS_AG', 'SpliceAI_pred_DS_AL', 'SpliceAI_pred_DS_DG', 'SpliceAI_pred_DS_DL', 'SpliceAI_pred_SYMBOL', 'LOEUF', 'PHENOTYPES', 'NMD']\n",
            "['ACMG_class', 'ACMG_coding_impact', 'ACMG_gene', 'ACMG_rules', 'ACMG_score', 'ACMG_transcript', 'AMP_matches', 'AMP_rules', 'AMP_score', 'AMP_tier', 'AMP_total_samples', 'AS_FilterStatus', 'AS_SB_TABLE', 'CGDinheritance', 'ClinVarClass', 'ClinVarDisease', 'DANN_score', 'DP', 'ECNT', 'GERMQ', 'Gene', 'MBQ', 'MFRL', 'MMQ', 'MPOS', 'MutationTaster_pred', 'MutationTaster_score', 'POPAF', 'RPA', 'RU', 'SIFT_score', 'STR', 'STRQ', 'TLOD', 'coding_impact', 'cosmicFathMMPrediction', 'cosmicFathMMScore', 'function', 'gnomadExomesEthnic_AC_Hom', 'gnomadExomes_AC', 'gnomadExomes_AC_Hemi', 'gnomadExomes_AC_Hom', 'gnomadExomes_AF', 'gnomadExomes_AF_ethnic', 'gnomadExomes_AN', 'gnomadGenomesAC', 'gnomadGenomesAN', 'gnomadGenomesEthnic_AC_Hom', 'gnomadGenomes_AC_Hemi', 'gnomadGenomes_AC_Hom', 'gnomadGenomes_AF', 'gnomadGenomes_AF_ethnic', 'hgvs', 'CSQ']\n"
          ]
        }
      ],
      "source": [
        "import vcf\n",
        "\n",
        "with open(\"data/EE_sample/EE_sample.vcf\", 'r') as vcf_file:\n",
        "    vcf_reader = vcf.Reader(vcf_file)\n",
        "\n",
        "    # for key, val in vcf_reader.infos.items():\n",
        "    #     print(f\"KEY: {key:30} DESCR: {val}\")\n",
        "\n",
        "    print(vcf_reader.infos[\"CSQ\"].desc[50:].split(\"|\"))\n",
        "    print(list(vcf_reader.infos.keys()))"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 35,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/plain": [
              "{'samples': array(['200852'], dtype=object),\n",
              " 'calldata/GT': array([[[0, 1]],\n",
              " \n",
              "        [[0, 1]],\n",
              " \n",
              "        [[0, 1]],\n",
              " \n",
              "        [[0, 1]],\n",
              " \n",
              "        [[0, 1]],\n",
              " \n",
              "        [[0, 1]]], dtype=int8),\n",
              " 'variants/ALT': array([['T', '', ''],\n",
              "        ['A', '', ''],\n",
              "        ['C', '', ''],\n",
              "        ['A', '', ''],\n",
              "        ['A', '', ''],\n",
              "        ['T', '', '']], dtype=object),\n",
              " 'variants/CHROM': array(['chr1', 'chr1', 'chr1', 'chr1', 'chr1', 'chr1'], dtype=object),\n",
              " 'variants/FILTER_PASS': array([ True,  True,  True,  True,  True,  True]),\n",
              " 'variants/ID': array(['rs2691315', 'rs201535981', 'rs71260069', 'rs367730352',\n",
              "        'rs71267774', 'rs1396210256'], dtype=object),\n",
              " 'variants/POS': array([ 15820,  17385,  17697, 133129, 183629, 184267]),\n",
              " 'variants/QUAL': array([nan, nan, nan, nan, nan, nan], dtype=float32),\n",
              " 'variants/REF': array(['G', 'G', 'G', 'G', 'G', 'C'], dtype=object)}"
            ]
          },
          "execution_count": 35,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "import allel\n",
        "data = allel.read_vcf(\"data/EE_sample/EE_sample.vcf\")\n",
        "data"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 41,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>CHROM</th>\n",
              "      <th>POS</th>\n",
              "      <th>ID</th>\n",
              "      <th>REF</th>\n",
              "      <th>ALT_1</th>\n",
              "      <th>ALT_2</th>\n",
              "      <th>ALT_3</th>\n",
              "      <th>QUAL</th>\n",
              "      <th>FILTER_PASS</th>\n",
              "      <th>FILTER_SB</th>\n",
              "      <th>...</th>\n",
              "      <th>gnomadGenomes_AC_Hemi_3</th>\n",
              "      <th>gnomadGenomes_AC_Hom_1</th>\n",
              "      <th>gnomadGenomes_AC_Hom_2</th>\n",
              "      <th>gnomadGenomes_AC_Hom_3</th>\n",
              "      <th>gnomadGenomes_AF_1</th>\n",
              "      <th>gnomadGenomes_AF_2</th>\n",
              "      <th>gnomadGenomes_AF_3</th>\n",
              "      <th>gnomadGenomes_AF_ethnic</th>\n",
              "      <th>hgvs</th>\n",
              "      <th>CSQ</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>chr1</td>\n",
              "      <td>15820</td>\n",
              "      <td>rs2691315</td>\n",
              "      <td>G</td>\n",
              "      <td>T</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>True</td>\n",
              "      <td>False</td>\n",
              "      <td>...</td>\n",
              "      <td>-1</td>\n",
              "      <td>18059</td>\n",
              "      <td>-1</td>\n",
              "      <td>-1</td>\n",
              "      <td>0.219648</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.093807</td>\n",
              "      <td>NaN</td>\n",
              "      <td>T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>chr1</td>\n",
              "      <td>17385</td>\n",
              "      <td>rs201535981</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>True</td>\n",
              "      <td>False</td>\n",
              "      <td>...</td>\n",
              "      <td>-1</td>\n",
              "      <td>21384</td>\n",
              "      <td>-1</td>\n",
              "      <td>-1</td>\n",
              "      <td>0.191644</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.213724</td>\n",
              "      <td>NaN</td>\n",
              "      <td>A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>chr1</td>\n",
              "      <td>17697</td>\n",
              "      <td>rs71260069</td>\n",
              "      <td>G</td>\n",
              "      <td>C</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>True</td>\n",
              "      <td>False</td>\n",
              "      <td>...</td>\n",
              "      <td>-1</td>\n",
              "      <td>17539</td>\n",
              "      <td>-1</td>\n",
              "      <td>-1</td>\n",
              "      <td>0.151133</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.178196</td>\n",
              "      <td>NaN</td>\n",
              "      <td>C|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>chr1</td>\n",
              "      <td>133129</td>\n",
              "      <td>rs367730352</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>True</td>\n",
              "      <td>False</td>\n",
              "      <td>...</td>\n",
              "      <td>-1</td>\n",
              "      <td>72154</td>\n",
              "      <td>-1</td>\n",
              "      <td>-1</td>\n",
              "      <td>0.547875</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.597389</td>\n",
              "      <td>NaN</td>\n",
              "      <td>A|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>chr1</td>\n",
              "      <td>183629</td>\n",
              "      <td>rs71267774</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>True</td>\n",
              "      <td>False</td>\n",
              "      <td>...</td>\n",
              "      <td>-1</td>\n",
              "      <td>14902</td>\n",
              "      <td>-1</td>\n",
              "      <td>-1</td>\n",
              "      <td>0.133296</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.220891</td>\n",
              "      <td>NaN</td>\n",
              "      <td>A|downstream_gene_variant|MODIFIER|MIR6859-2|E...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>chr1</td>\n",
              "      <td>184267</td>\n",
              "      <td>rs1396210256</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>True</td>\n",
              "      <td>False</td>\n",
              "      <td>...</td>\n",
              "      <td>-1</td>\n",
              "      <td>887</td>\n",
              "      <td>-1</td>\n",
              "      <td>-1</td>\n",
              "      <td>0.006127</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.006310</td>\n",
              "      <td>NaN</td>\n",
              "      <td>T|downstream_gene_variant|MODIFIER|MIR6859-2|E...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>6 rows  113 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "  CHROM     POS            ID REF ALT_1  ALT_2  ALT_3  QUAL  FILTER_PASS  \\\n",
              "0  chr1   15820     rs2691315   G     T    NaN    NaN   NaN         True   \n",
              "1  chr1   17385   rs201535981   G     A    NaN    NaN   NaN         True   \n",
              "2  chr1   17697    rs71260069   G     C    NaN    NaN   NaN         True   \n",
              "3  chr1  133129   rs367730352   G     A    NaN    NaN   NaN         True   \n",
              "4  chr1  183629    rs71267774   G     A    NaN    NaN   NaN         True   \n",
              "5  chr1  184267  rs1396210256   C     T    NaN    NaN   NaN         True   \n",
              "\n",
              "   FILTER_SB  ...  gnomadGenomes_AC_Hemi_3  gnomadGenomes_AC_Hom_1  \\\n",
              "0      False  ...                       -1                   18059   \n",
              "1      False  ...                       -1                   21384   \n",
              "2      False  ...                       -1                   17539   \n",
              "3      False  ...                       -1                   72154   \n",
              "4      False  ...                       -1                   14902   \n",
              "5      False  ...                       -1                     887   \n",
              "\n",
              "   gnomadGenomes_AC_Hom_2  gnomadGenomes_AC_Hom_3  gnomadGenomes_AF_1  \\\n",
              "0                      -1                      -1            0.219648   \n",
              "1                      -1                      -1            0.191644   \n",
              "2                      -1                      -1            0.151133   \n",
              "3                      -1                      -1            0.547875   \n",
              "4                      -1                      -1            0.133296   \n",
              "5                      -1                      -1            0.006127   \n",
              "\n",
              "   gnomadGenomes_AF_2  gnomadGenomes_AF_3 gnomadGenomes_AF_ethnic hgvs  \\\n",
              "0                 NaN                 NaN                0.093807  NaN   \n",
              "1                 NaN                 NaN                0.213724  NaN   \n",
              "2                 NaN                 NaN                0.178196  NaN   \n",
              "3                 NaN                 NaN                0.597389  NaN   \n",
              "4                 NaN                 NaN                0.220891  NaN   \n",
              "5                 NaN                 NaN                0.006310  NaN   \n",
              "\n",
              "                                                 CSQ  \n",
              "0  T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...  \n",
              "1  A|downstream_gene_variant|MODIFIER|DDX11L1|ENS...  \n",
              "2  C|downstream_gene_variant|MODIFIER|DDX11L1|ENS...  \n",
              "3  A|non_coding_transcript_exon_variant|MODIFIER|...  \n",
              "4  A|downstream_gene_variant|MODIFIER|MIR6859-2|E...  \n",
              "5  T|downstream_gene_variant|MODIFIER|MIR6859-2|E...  \n",
              "\n",
              "[6 rows x 113 columns]"
            ]
          },
          "execution_count": 41,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "df = allel.vcf_to_dataframe(\"data/EE_sample/EE_sample.vcf\", [\"CHROM\", \"POS\", \"ID\", \"REF\", \"ALT\", \"QUAL\", \"FILTER\", \"INFO\"])\n",
        "df"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 40,
      "metadata": {},
      "outputs": [],
      "source": [
        "allel.vcf_to_csv(\"data/EE_sample/EE_sample.vcf\", \"data/EE_sample/EE_sample_allel.csv\")"
      ]
    }
  ],
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.11.6"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}
