{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_genotype = pd.read_csv(\"../data/EE_015/EE_015_genotype.csv.gz\", sep=\";\", compression=\"gzip\")\n",
    "df_default = pd.read_csv(\"../data/EE_015/EE_015_default.csv.gz\", sep=\";\", compression=\"gzip\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['ACMG_class', 'ACMG_coding_impact', 'ACMG_gene', 'ACMG_rules',\n",
       "       'ACMG_transcript', 'AMP_matches', 'AMP_rules', 'AMP_tier',\n",
       "       'AS_FilterStatus', 'AS_SB_TABLE', 'CGDinheritance', 'Gene',\n",
       "       'MutationTaster_pred', 'MutationTaster_score', 'RU', 'SIFT_score',\n",
       "       'coding_impact', 'cosmicFathMMPrediction', 'function', 'hgvs'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_genotype.select_dtypes(object).columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([  0,   2,   1,   3,   6, 108, 112, 109,   5, 113, 110], dtype=int64)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_genotype[\"ClinVarClass\"].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.concat([df_default.drop(\"Unnamed: 0\", axis=1), df_genotype.drop(\"Unnamed: 0\", axis=1)], axis=1, ignore_index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "object_columns = df.select_dtypes(object).columns\n",
    "\n",
    "# i = 3\n",
    "# for col in object_columns[i:i+1]:\n",
    "#     # print(f\"Column {col}: {df[col].unique()}\")\n",
    "#     print(col)\n",
    "#     unique = df[col].unique()\n",
    "#     print(len(unique))\n",
    "#     # print([x for x in unique if len(x.split(\";\"))>1])\n",
    "\n",
    "#     for unique_val in unique:\n",
    "#         print(unique_val)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df[[c for c in df.columns if c not in df.select_dtypes(object).columns]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['PASS', 'clustered_events', 'haplotype']"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# sklearn.preprocessing.OneHotEncoder\n",
    "\n",
    "filters = [\"PASS\", \"SB\", \"UM\", \"clustered_events\", \"fragment\", \"germline\", \"haplotype\", \"multiallelic\", \"slippage\"]\n",
    "filter_cols = [\"FILTER_\" + f for f in filters]\n",
    "df[\"FILTER\"].astype(str).str.split(\";\").iloc[-6]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'0',\n",
       " \"3'flank\",\n",
       " \"3'utr\",\n",
       " \"5'flank\",\n",
       " \"5'utr\",\n",
       " 'NMD',\n",
       " 'coding',\n",
       " 'intronic',\n",
       " 'non-coding%40exon',\n",
       " 'splicing',\n",
       " 'splicing-ACMG'}"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "functions = [\"0\", \"NMD\", \"3'utr\", \"5'utr\", \"3'flank\", \"5'flank\", \"coding\", \"non-coding%40exon\", \"intronic\", \"splicing\", \"splicing-ACMG\"]\n",
    "function_cols = [\"function_\" + f for f in filters]\n",
    "\n",
    "uniq = set([])\n",
    "for x in df[\"function\"].astype(str).str.split(\",\"):\n",
    "    for y in x:\n",
    "        if y not in uniq:\n",
    "            uniq.add(y)\n",
    "\n",
    "uniq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['#CHROM', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'ACMG_class',\n",
       "       'ACMG_coding_impact', 'ACMG_gene', 'ACMG_rules', 'ACMG_transcript',\n",
       "       'AMP_matches', 'AMP_rules', 'AMP_tier', 'AS_FilterStatus',\n",
       "       'AS_SB_TABLE', 'CGDinheritance', 'Gene', 'MutationTaster_pred',\n",
       "       'MutationTaster_score', 'RU', 'SIFT_score', 'coding_impact',\n",
       "       'cosmicFathMMPrediction', 'function', 'hgvs'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "object_columns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Object columns:\n",
    "- #CHROM - no arrays\n",
    "- ID - id\n",
    "- REF, ALT - no arrays\n",
    "- QUAL - binary 1 or .\n",
    "- FILTER - [\"PASS\", \"SB\", \"UM\", \"clustered_events\", \"fragment\", \"germline\", \"haplotype\", \"multiallelic\", \"slippage\"]\n",
    "- ACMG_class - no arrays\n",
    "- ACMG_coding_impact - no arrays\n",
    "- ACMG_gene - no arrays\n",
    "- ACMG_rules - https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4544753/ I will leave this data unseparated for now \n",
    "- ACMG_transcript - no arrays\n",
    "- AMP_matches - no arrays\n",
    "- AMP_rules - https://varsome.com/about/resources/somatic-implementation/ I will leave this data unseparated for now\n",
    "- AMP_tier - no arrays\n",
    "- AS_FilterStatus - https://docs.varsome.com/en/vcf-attributes-explained I will leave this data unseparated for now\n",
    "- AS_SB_TABLE - https://docs.varsome.com/en/vcf-attributes-explained I will leave this data unseparated for now\n",
    "- CGDinheritance - I couldn't find good info, looks separable, left for now\n",
    "- Gene - gene symbols - might be separable, maybe not worth is if a lot of unique values, check relation to ACMG (target) class\n",
    "- MutationTaster_pred - I don't know which of the symbols represent which type and why there is a number of them, %3B is ';', left unseparated for now. https://www.mutationtaster.org/info/documentation.html. MutationTaster predicts an alteration as one of four possible types:\n",
    "    - disease causing - i.e. probably deleterious\n",
    "    - disease causing automatic - i.e. known to be deleterious, see section dbSNP / TGP / ClinVar / HGMD for details\n",
    "    - polymorphism - i.e. probably harmless\n",
    "    - polymorphism automatic - i.e. known to be harmless, see section dbSNP / TGP / ClinVar / HGMD for details\n",
    "- MutationTaster_score - respective score (certainty?)\n",
    "- RU - no arrays\n",
    "- SIFT_score - https://ionreporter.thermofisher.com/ionreporter/help/GUID-2097F236-C8A2-4E67-862D-0FB5875979AC.html, separable but I don't know how\n",
    "- coding_impact - https://www.ebi.ac.uk/training/online/courses/human-genetic-variation-introduction/what-is-genetic-variation/what-effect-do-variants-in-coding-regions-have/ seems like there are 3 important classes, couldn't find info on the other, leaving unseparated for now\n",
    "- cosmicFathMMPrediction - no arrays - also target\n",
    "- function - https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2872880/ - [\"0\", \"NMD\", \"3'utr\", \"5'utr\", \"3'flank\", \"5'flank\", \"coding\", \"non-coding%40exon\", \"intronic\", \"splicing\", \"splicing-ACMG\"]\n",
    "- hgvs - https://varnomen.hgvs.org/recommendations/general/ - again I don't know how to separate this well"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Analyzing additional columns in EE_069"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\barte\\AppData\\Local\\Temp\\ipykernel_32944\\4088336807.py:7: DtypeWarning: Columns (12) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df2 = pd.read_csv(\"data/EE_069/EE_069_genotype.csv.gz\", sep=\";\", compression=\"gzip\").drop(\"Unnamed: 0\", axis=1)\n"
     ]
    }
   ],
   "source": [
    "df1 = pd.read_csv(\"data/EE_050/EE_050_default.csv.gz\", sep=\";\", compression=\"gzip\").drop(\"Unnamed: 0\", axis=1)\n",
    "df2 = pd.read_csv(\"data/EE_050/EE_050_genotype.csv.gz\", sep=\";\", compression=\"gzip\").drop(\"Unnamed: 0\", axis=1)\n",
    "\n",
    "df_50 = pd.concat([df1, df2], ignore_index=False, axis=1)\n",
    "\n",
    "df1 = pd.read_csv(\"data/EE_069/EE_069_default.csv.gz\", sep=\";\", compression=\"gzip\").drop(\"Unnamed: 0\", axis=1)\n",
    "df2 = pd.read_csv(\"data/EE_069/EE_069_genotype.csv.gz\", sep=\";\", compression=\"gzip\").drop(\"Unnamed: 0\", axis=1)\n",
    "\n",
    "df_69 = pd.concat([df1, df2], ignore_index=False, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'AMP_matches',\n",
       " 'AMP_rules',\n",
       " 'AMP_score',\n",
       " 'AMP_tier',\n",
       " 'AMP_total_samples',\n",
       " 'AS_FilterStatus',\n",
       " 'AS_SB_TABLE',\n",
       " 'ECNT',\n",
       " 'GERMQ',\n",
       " 'MBQ',\n",
       " 'MFRL',\n",
       " 'MMQ',\n",
       " 'MPOS',\n",
       " 'POPAF',\n",
       " 'RPA',\n",
       " 'RU',\n",
       " 'STR',\n",
       " 'STRQ',\n",
       " 'TLOD',\n",
       " 'cosmicFathMMPrediction',\n",
       " 'cosmicFathMMScore'}"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "diff_cols_50 = set(df_50.columns).difference(set(df_69.columns))\n",
    "diff_cols_50"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'AC',\n",
       " 'AF',\n",
       " 'AN',\n",
       " 'BaseQRankSum',\n",
       " 'ClippingRankSum',\n",
       " 'ExcessHet',\n",
       " 'FS',\n",
       " 'MLEAC',\n",
       " 'MLEAF',\n",
       " 'MQ',\n",
       " 'MQRankSum',\n",
       " 'QD',\n",
       " 'ReadPosRankSum',\n",
       " 'SOR'}"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "diff_cols_69 = set(df_69.columns).difference(set(df_50.columns))\n",
    "diff_cols_69"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df_69[list(diff_cols_69)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['BaseQRankSum', 'MQRankSum', 'ClippingRankSum', 'ReadPosRankSum'], dtype='object')"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "object_columns = df_69[list(diff_cols_69)].select_dtypes(object).columns\n",
    "\n",
    "object_columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ClippingRankSum\n",
      "2\n",
      "0\n",
      ".\n"
     ]
    }
   ],
   "source": [
    "i = 2\n",
    "for col in object_columns[i:i+1]:\n",
    "    # print(f\"Column {col}: {df[col].unique()}\")\n",
    "    print(col)\n",
    "    unique = df_69[col].unique()\n",
    "    print(len(unique))\n",
    "    # print([x for x in unique if len(x.split(\";\"))>1])\n",
    "\n",
    "    for unique_val in unique:\n",
    "        print(unique_val)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- ReadPosRankSum - no arrays looks like float + nan\n",
    "- ClippingRankSum - no arrays\n",
    "- MQRankSum - no arrays - float and nan\n",
    "- BaseQRankSum - same"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Analyzing allel and uncsv read difference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "df1 = pd.read_csv(\"data/EE_015/EE_015_default.csv.gz\", sep=\";\", compression=\"gzip\").drop(\"Unnamed: 0\", axis=1)\n",
    "df2 = pd.read_csv(\"data/EE_015/EE_015_genotype.csv.gz\", sep=\";\", compression=\"gzip\").drop(\"Unnamed: 0\", axis=1)\n",
    "\n",
    "df_15 = pd.concat([df1, df2], axis=1, ignore_index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df_15"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "df1 = pd.read_csv(\"data/EE_050/EE_050_default.csv.gz\", sep=\";\", compression=\"gzip\").drop(\"Unnamed: 0\", axis=1)\n",
    "df2 = pd.read_csv(\"data/EE_050/EE_050_genotype.csv.gz\", sep=\";\", compression=\"gzip\").drop(\"Unnamed: 0\", axis=1)\n",
    "\n",
    "df_50 = pd.concat([df1, df2], axis=1, ignore_index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df_50"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\barte\\AppData\\Local\\Temp\\ipykernel_32944\\3002690269.py:2: DtypeWarning: Columns (12) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df2 = pd.read_csv(\"data/EE_069/EE_069_genotype.csv.gz\", sep=\";\", compression=\"gzip\").drop(\"Unnamed: 0\", axis=1)\n"
     ]
    }
   ],
   "source": [
    "df1 = pd.read_csv(\"data/EE_069/EE_069_default.csv.gz\", sep=\";\", compression=\"gzip\").drop(\"Unnamed: 0\", axis=1)\n",
    "df2 = pd.read_csv(\"data/EE_069/EE_069_genotype.csv.gz\", sep=\";\", compression=\"gzip\").drop(\"Unnamed: 0\", axis=1)\n",
    "\n",
    "df_69 = pd.concat([df1, df2], axis=1, ignore_index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>AC</th>\n",
       "      <th>ACMG_class</th>\n",
       "      <th>ACMG_coding_impact</th>\n",
       "      <th>...</th>\n",
       "      <th>gnomadExomes_AF_ethnic</th>\n",
       "      <th>gnomadExomes_AN</th>\n",
       "      <th>gnomadGenomesAC</th>\n",
       "      <th>gnomadGenomesAN</th>\n",
       "      <th>gnomadGenomesEthnic_AC_Hom</th>\n",
       "      <th>gnomadGenomes_AC_Hemi</th>\n",
       "      <th>gnomadGenomes_AC_Hom</th>\n",
       "      <th>gnomadGenomes_AF</th>\n",
       "      <th>gnomadGenomes_AF_ethnic</th>\n",
       "      <th>hgvs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14653</td>\n",
       "      <td>rs62635297</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>331.77</td>\n",
       "      <td>PASS</td>\n",
       "      <td>1</td>\n",
       "      <td>Uncertain%40Significance</td>\n",
       "      <td>non%40coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>31329</td>\n",
       "      <td>79992</td>\n",
       "      <td>420</td>\n",
       "      <td>0</td>\n",
       "      <td>31329</td>\n",
       "      <td>0.391652</td>\n",
       "      <td>0.447103</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17594</td>\n",
       "      <td>rs377698370</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>109.77</td>\n",
       "      <td>PASS</td>\n",
       "      <td>1</td>\n",
       "      <td>Benign</td>\n",
       "      <td>non%40coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>3276</td>\n",
       "      <td>132414</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3276</td>\n",
       "      <td>0.024741</td>\n",
       "      <td>0.007104</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17614</td>\n",
       "      <td>rs201057270</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>258.77</td>\n",
       "      <td>PASS</td>\n",
       "      <td>1</td>\n",
       "      <td>Benign</td>\n",
       "      <td>non%40coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>6270</td>\n",
       "      <td>126212</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6270</td>\n",
       "      <td>0.049678</td>\n",
       "      <td>0.056625</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17697</td>\n",
       "      <td>rs71260069</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>224.77</td>\n",
       "      <td>PASS</td>\n",
       "      <td>1</td>\n",
       "      <td>Benign</td>\n",
       "      <td>non%40coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>17539</td>\n",
       "      <td>116050</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>17539</td>\n",
       "      <td>0.151133</td>\n",
       "      <td>0.178196</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17722</td>\n",
       "      <td>rs376731495</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>313.77</td>\n",
       "      <td>PASS</td>\n",
       "      <td>1</td>\n",
       "      <td>Uncertain%40Significance</td>\n",
       "      <td>non%40coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>2836</td>\n",
       "      <td>123488</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2836</td>\n",
       "      <td>0.022966</td>\n",
       "      <td>0.027542</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140169</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56856571</td>\n",
       "      <td>rs9988391</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>293.77</td>\n",
       "      <td>PASS</td>\n",
       "      <td>1</td>\n",
       "      <td>Uncertain%40Significance</td>\n",
       "      <td>non%40coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140170</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56856581</td>\n",
       "      <td>rs9320083</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>750.77</td>\n",
       "      <td>PASS</td>\n",
       "      <td>1</td>\n",
       "      <td>Uncertain%40Significance</td>\n",
       "      <td>non%40coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>135</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.014815</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140171</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56856587</td>\n",
       "      <td>rs7067511</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>1164.77</td>\n",
       "      <td>PASS</td>\n",
       "      <td>1</td>\n",
       "      <td>Uncertain%40Significance</td>\n",
       "      <td>non%40coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>123</td>\n",
       "      <td>123</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>123</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140172</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56856668</td>\n",
       "      <td>rs5013140</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>467.77</td>\n",
       "      <td>PASS</td>\n",
       "      <td>1</td>\n",
       "      <td>Uncertain%40Significance</td>\n",
       "      <td>non%40coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>101</td>\n",
       "      <td>103</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>101</td>\n",
       "      <td>0.980583</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140173</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56856670</td>\n",
       "      <td>rs5013141</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>467.77</td>\n",
       "      <td>PASS</td>\n",
       "      <td>1</td>\n",
       "      <td>Uncertain%40Significance</td>\n",
       "      <td>non%40coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>100</td>\n",
       "      <td>102</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>100</td>\n",
       "      <td>0.980392</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>140174 rows × 53 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       #CHROM       POS           ID REF ALT     QUAL FILTER  AC  \\\n",
       "0        chr1     14653   rs62635297   C   T   331.77   PASS   1   \n",
       "1        chr1     17594  rs377698370   C   T   109.77   PASS   1   \n",
       "2        chr1     17614  rs201057270   G   A   258.77   PASS   1   \n",
       "3        chr1     17697   rs71260069   G   C   224.77   PASS   1   \n",
       "4        chr1     17722  rs376731495   A   G   313.77   PASS   1   \n",
       "...       ...       ...          ...  ..  ..      ...    ...  ..   \n",
       "140169   chrY  56856571    rs9988391   G   A   293.77   PASS   1   \n",
       "140170   chrY  56856581    rs9320083   G   A   750.77   PASS   1   \n",
       "140171   chrY  56856587    rs7067511   A   C  1164.77   PASS   1   \n",
       "140172   chrY  56856668    rs5013140   A   G   467.77   PASS   1   \n",
       "140173   chrY  56856670    rs5013141   G   T   467.77   PASS   1   \n",
       "\n",
       "                      ACMG_class ACMG_coding_impact  ...  \\\n",
       "0       Uncertain%40Significance       non%40coding  ...   \n",
       "1                         Benign       non%40coding  ...   \n",
       "2                         Benign       non%40coding  ...   \n",
       "3                         Benign       non%40coding  ...   \n",
       "4       Uncertain%40Significance       non%40coding  ...   \n",
       "...                          ...                ...  ...   \n",
       "140169  Uncertain%40Significance       non%40coding  ...   \n",
       "140170  Uncertain%40Significance       non%40coding  ...   \n",
       "140171  Uncertain%40Significance       non%40coding  ...   \n",
       "140172  Uncertain%40Significance       non%40coding  ...   \n",
       "140173  Uncertain%40Significance       non%40coding  ...   \n",
       "\n",
       "       gnomadExomes_AF_ethnic gnomadExomes_AN  gnomadGenomesAC  \\\n",
       "0                         0.0               0            31329   \n",
       "1                         0.0               0             3276   \n",
       "2                         0.0               0             6270   \n",
       "3                         0.0               0            17539   \n",
       "4                         0.0               0             2836   \n",
       "...                       ...             ...              ...   \n",
       "140169                    0.0               0                0   \n",
       "140170                    0.0               0                2   \n",
       "140171                    0.0               0              123   \n",
       "140172                    0.0               0              101   \n",
       "140173                    0.0               0              100   \n",
       "\n",
       "       gnomadGenomesAN  gnomadGenomesEthnic_AC_Hom  gnomadGenomes_AC_Hemi  \\\n",
       "0                79992                         420                      0   \n",
       "1               132414                           0                      0   \n",
       "2               126212                           0                      0   \n",
       "3               116050                           2                      0   \n",
       "4               123488                           0                      0   \n",
       "...                ...                         ...                    ...   \n",
       "140169               0                           0                      0   \n",
       "140170             135                           0                      0   \n",
       "140171             123                           0                      0   \n",
       "140172             103                           0                      0   \n",
       "140173             102                           0                      0   \n",
       "\n",
       "       gnomadGenomes_AC_Hom gnomadGenomes_AF gnomadGenomes_AF_ethnic  hgvs  \n",
       "0                     31329         0.391652                0.447103     0  \n",
       "1                      3276         0.024741                0.007104     0  \n",
       "2                      6270         0.049678                0.056625     0  \n",
       "3                     17539         0.151133                0.178196     0  \n",
       "4                      2836         0.022966                0.027542     0  \n",
       "...                     ...              ...                     ...   ...  \n",
       "140169                    0         0.000000                0.000000     0  \n",
       "140170                    2         0.014815                0.000000     0  \n",
       "140171                  123         1.000000                1.000000     0  \n",
       "140172                  101         0.980583                1.000000     0  \n",
       "140173                  100         0.980392                1.000000     0  \n",
       "\n",
       "[140174 rows x 53 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# df_69"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_csv = pd.concat([df_15, df_50, df_69], axis=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>ACMG_class</th>\n",
       "      <th>ACMG_coding_impact</th>\n",
       "      <th>ACMG_gene</th>\n",
       "      <th>...</th>\n",
       "      <th>ClippingRankSum</th>\n",
       "      <th>ExcessHet</th>\n",
       "      <th>FS</th>\n",
       "      <th>MLEAC</th>\n",
       "      <th>MLEAF</th>\n",
       "      <th>MQ</th>\n",
       "      <th>MQRankSum</th>\n",
       "      <th>QD</th>\n",
       "      <th>ReadPosRankSum</th>\n",
       "      <th>SOR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>15820</td>\n",
       "      <td>rs2691315</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>Uncertain%40Significance</td>\n",
       "      <td>non%40coding</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17385</td>\n",
       "      <td>rs201535981</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>Uncertain%40Significance</td>\n",
       "      <td>non%40coding</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17697</td>\n",
       "      <td>rs71260069</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>Benign</td>\n",
       "      <td>non%40coding</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>133129</td>\n",
       "      <td>rs367730352</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>Uncertain%40Significance</td>\n",
       "      <td>non%40coding</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>183629</td>\n",
       "      <td>rs71267774</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>Benign</td>\n",
       "      <td>non%40coding</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140169</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56856571</td>\n",
       "      <td>rs9988391</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>293.77</td>\n",
       "      <td>PASS</td>\n",
       "      <td>Uncertain%40Significance</td>\n",
       "      <td>non%40coding</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>3.0103</td>\n",
       "      <td>8.315</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>41.25</td>\n",
       "      <td>-0.431</td>\n",
       "      <td>5.76</td>\n",
       "      <td>-1.02</td>\n",
       "      <td>0.419</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140170</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56856581</td>\n",
       "      <td>rs9320083</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>750.77</td>\n",
       "      <td>PASS</td>\n",
       "      <td>Uncertain%40Significance</td>\n",
       "      <td>non%40coding</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>3.0103</td>\n",
       "      <td>2.660</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>41.23</td>\n",
       "      <td>-2.005</td>\n",
       "      <td>16.68</td>\n",
       "      <td>-0.205</td>\n",
       "      <td>0.361</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140171</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56856587</td>\n",
       "      <td>rs7067511</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>1164.77</td>\n",
       "      <td>PASS</td>\n",
       "      <td>Uncertain%40Significance</td>\n",
       "      <td>non%40coding</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>3.0103</td>\n",
       "      <td>0.000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>40.19</td>\n",
       "      <td>-3.699</td>\n",
       "      <td>31.48</td>\n",
       "      <td>-1.691</td>\n",
       "      <td>0.530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140172</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56856668</td>\n",
       "      <td>rs5013140</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>467.77</td>\n",
       "      <td>PASS</td>\n",
       "      <td>Uncertain%40Significance</td>\n",
       "      <td>non%40coding</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>3.0103</td>\n",
       "      <td>0.000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>44.02</td>\n",
       "      <td>-1.952</td>\n",
       "      <td>33.41</td>\n",
       "      <td>0.183</td>\n",
       "      <td>2.985</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140173</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56856670</td>\n",
       "      <td>rs5013141</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>467.77</td>\n",
       "      <td>PASS</td>\n",
       "      <td>Uncertain%40Significance</td>\n",
       "      <td>non%40coding</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>3.0103</td>\n",
       "      <td>0.000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>44.02</td>\n",
       "      <td>-1.952</td>\n",
       "      <td>33.41</td>\n",
       "      <td>0.549</td>\n",
       "      <td>2.985</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>444444 rows × 74 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       #CHROM       POS           ID REF ALT     QUAL FILTER  \\\n",
       "0        chr1     15820    rs2691315   G   T        .   PASS   \n",
       "1        chr1     17385  rs201535981   G   A        .   PASS   \n",
       "2        chr1     17697   rs71260069   G   C        .   PASS   \n",
       "3        chr1    133129  rs367730352   G   A        .   PASS   \n",
       "4        chr1    183629   rs71267774   G   A        .   PASS   \n",
       "...       ...       ...          ...  ..  ..      ...    ...   \n",
       "140169   chrY  56856571    rs9988391   G   A   293.77   PASS   \n",
       "140170   chrY  56856581    rs9320083   G   A   750.77   PASS   \n",
       "140171   chrY  56856587    rs7067511   A   C  1164.77   PASS   \n",
       "140172   chrY  56856668    rs5013140   A   G   467.77   PASS   \n",
       "140173   chrY  56856670    rs5013141   G   T   467.77   PASS   \n",
       "\n",
       "                      ACMG_class ACMG_coding_impact ACMG_gene  ...  \\\n",
       "0       Uncertain%40Significance       non%40coding         0  ...   \n",
       "1       Uncertain%40Significance       non%40coding         0  ...   \n",
       "2                         Benign       non%40coding         0  ...   \n",
       "3       Uncertain%40Significance       non%40coding         0  ...   \n",
       "4                         Benign       non%40coding         0  ...   \n",
       "...                          ...                ...       ...  ...   \n",
       "140169  Uncertain%40Significance       non%40coding         0  ...   \n",
       "140170  Uncertain%40Significance       non%40coding         0  ...   \n",
       "140171  Uncertain%40Significance       non%40coding         0  ...   \n",
       "140172  Uncertain%40Significance       non%40coding         0  ...   \n",
       "140173  Uncertain%40Significance       non%40coding         0  ...   \n",
       "\n",
       "       ClippingRankSum  ExcessHet     FS MLEAC MLEAF     MQ MQRankSum     QD  \\\n",
       "0                  NaN        NaN    NaN   NaN   NaN    NaN       NaN    NaN   \n",
       "1                  NaN        NaN    NaN   NaN   NaN    NaN       NaN    NaN   \n",
       "2                  NaN        NaN    NaN   NaN   NaN    NaN       NaN    NaN   \n",
       "3                  NaN        NaN    NaN   NaN   NaN    NaN       NaN    NaN   \n",
       "4                  NaN        NaN    NaN   NaN   NaN    NaN       NaN    NaN   \n",
       "...                ...        ...    ...   ...   ...    ...       ...    ...   \n",
       "140169               0     3.0103  8.315   1.0   0.5  41.25    -0.431   5.76   \n",
       "140170               0     3.0103  2.660   1.0   0.5  41.23    -2.005  16.68   \n",
       "140171               0     3.0103  0.000   1.0   0.5  40.19    -3.699  31.48   \n",
       "140172               0     3.0103  0.000   1.0   0.5  44.02    -1.952  33.41   \n",
       "140173               0     3.0103  0.000   1.0   0.5  44.02    -1.952  33.41   \n",
       "\n",
       "       ReadPosRankSum    SOR  \n",
       "0                 NaN    NaN  \n",
       "1                 NaN    NaN  \n",
       "2                 NaN    NaN  \n",
       "3                 NaN    NaN  \n",
       "4                 NaN    NaN  \n",
       "...               ...    ...  \n",
       "140169          -1.02  0.419  \n",
       "140170         -0.205  0.361  \n",
       "140171         -1.691  0.530  \n",
       "140172          0.183  2.985  \n",
       "140173          0.549  2.985  \n",
       "\n",
       "[444444 rows x 74 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\Users\\barte\\anaconda3\\envs\\pathogen\\Lib\\site-packages\\allel\\io\\vcf_read.py:1240: UserWarning: '#CHROM' INFO header not found\n",
      "  warnings.warn('%r INFO header not found' % name)\n",
      "c:\\Users\\barte\\anaconda3\\envs\\pathogen\\Lib\\site-packages\\allel\\io\\vcf_read.py:1454: UserWarning: no type for field 'variants/#CHROM', assuming object\n",
      "  warnings.warn('no type for field %r, assuming %s' % (f, normed_types[f]))\n",
      "c:\\Users\\barte\\anaconda3\\envs\\pathogen\\Lib\\site-packages\\allel\\io\\vcf_read.py:1564: UserWarning: no number for field 'variants/#CHROM', assuming 1\n",
      "  warnings.warn('no number for field %r, assuming 1' % f)\n"
     ]
    }
   ],
   "source": [
    "import allel\n",
    "\n",
    "df_allel_15 = allel.vcf_to_dataframe(\"data/EE_015/EE_015.vcf.gz\", [\"#CHROM\", \"POS\", \"ID\", \"REF\", \"ALT\", \"QUAL\", \"FILTER\", \"INFO\"])\n",
    "df_allel_50 = allel.vcf_to_dataframe(\"data/EE_050/EE_050.vcf.gz\", [\"#CHROM\", \"POS\", \"ID\", \"REF\", \"ALT\", \"QUAL\", \"FILTER\", \"INFO\"])\n",
    "df_allel_69 = allel.vcf_to_dataframe(\"data/EE_069/EE_069.vcf.gz\", [\"#CHROM\", \"POS\", \"ID\", \"REF\", \"ALT\", \"QUAL\", \"FILTER\", \"INFO\"])\n",
    "\n",
    "df_allel = pd.concat([df_allel_15, df_allel_50, df_allel_69], axis=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_allel.drop(\"CSQ\", inplace=True, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT_1</th>\n",
       "      <th>ALT_2</th>\n",
       "      <th>ALT_3</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER_PASS</th>\n",
       "      <th>FILTER_SB</th>\n",
       "      <th>...</th>\n",
       "      <th>MLEAC_2</th>\n",
       "      <th>MLEAC_3</th>\n",
       "      <th>MLEAF_1</th>\n",
       "      <th>MLEAF_2</th>\n",
       "      <th>MLEAF_3</th>\n",
       "      <th>MQ</th>\n",
       "      <th>MQRankSum</th>\n",
       "      <th>QD</th>\n",
       "      <th>ReadPosRankSum</th>\n",
       "      <th>SOR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>15820</td>\n",
       "      <td>rs2691315</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>17385</td>\n",
       "      <td>rs201535981</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>17697</td>\n",
       "      <td>rs71260069</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>133129</td>\n",
       "      <td>rs367730352</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>183629</td>\n",
       "      <td>rs71267774</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9097</th>\n",
       "      <td>NaN</td>\n",
       "      <td>56856571</td>\n",
       "      <td>rs9988391</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>293.769989</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>41.250000</td>\n",
       "      <td>-0.431</td>\n",
       "      <td>5.76</td>\n",
       "      <td>-1.020</td>\n",
       "      <td>0.419</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9098</th>\n",
       "      <td>NaN</td>\n",
       "      <td>56856581</td>\n",
       "      <td>rs9320083</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>750.770020</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>41.230000</td>\n",
       "      <td>-2.005</td>\n",
       "      <td>16.68</td>\n",
       "      <td>-0.205</td>\n",
       "      <td>0.361</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9099</th>\n",
       "      <td>NaN</td>\n",
       "      <td>56856587</td>\n",
       "      <td>rs7067511</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1164.770020</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>40.189999</td>\n",
       "      <td>-3.699</td>\n",
       "      <td>31.48</td>\n",
       "      <td>-1.691</td>\n",
       "      <td>0.530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9100</th>\n",
       "      <td>NaN</td>\n",
       "      <td>56856668</td>\n",
       "      <td>rs5013140</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>467.769989</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>44.020000</td>\n",
       "      <td>-1.952</td>\n",
       "      <td>33.41</td>\n",
       "      <td>0.183</td>\n",
       "      <td>2.985</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9101</th>\n",
       "      <td>NaN</td>\n",
       "      <td>56856670</td>\n",
       "      <td>rs5013141</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>467.769989</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>44.020000</td>\n",
       "      <td>-1.952</td>\n",
       "      <td>33.41</td>\n",
       "      <td>0.549</td>\n",
       "      <td>2.985</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>444444 rows × 139 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #CHROM       POS           ID REF ALT_1  ALT_2  ALT_3         QUAL  \\\n",
       "0        NaN     15820    rs2691315   G     T    NaN    NaN          NaN   \n",
       "1        NaN     17385  rs201535981   G     A    NaN    NaN          NaN   \n",
       "2        NaN     17697   rs71260069   G     C    NaN    NaN          NaN   \n",
       "3        NaN    133129  rs367730352   G     A    NaN    NaN          NaN   \n",
       "4        NaN    183629   rs71267774   G     A    NaN    NaN          NaN   \n",
       "...      ...       ...          ...  ..   ...    ...    ...          ...   \n",
       "9097     NaN  56856571    rs9988391   G     A    NaN    NaN   293.769989   \n",
       "9098     NaN  56856581    rs9320083   G     A    NaN    NaN   750.770020   \n",
       "9099     NaN  56856587    rs7067511   A     C    NaN    NaN  1164.770020   \n",
       "9100     NaN  56856668    rs5013140   A     G    NaN    NaN   467.769989   \n",
       "9101     NaN  56856670    rs5013141   G     T    NaN    NaN   467.769989   \n",
       "\n",
       "      FILTER_PASS  FILTER_SB  ...  MLEAC_2 MLEAC_3 MLEAF_1 MLEAF_2 MLEAF_3  \\\n",
       "0            True      False  ...      NaN     NaN     NaN     NaN     NaN   \n",
       "1            True      False  ...      NaN     NaN     NaN     NaN     NaN   \n",
       "2            True      False  ...      NaN     NaN     NaN     NaN     NaN   \n",
       "3            True      False  ...      NaN     NaN     NaN     NaN     NaN   \n",
       "4            True      False  ...      NaN     NaN     NaN     NaN     NaN   \n",
       "...           ...        ...  ...      ...     ...     ...     ...     ...   \n",
       "9097         True      False  ...     -1.0    -1.0     0.5     NaN     NaN   \n",
       "9098         True      False  ...     -1.0    -1.0     0.5     NaN     NaN   \n",
       "9099         True      False  ...     -1.0    -1.0     0.5     NaN     NaN   \n",
       "9100         True      False  ...     -1.0    -1.0     0.5     NaN     NaN   \n",
       "9101         True      False  ...     -1.0    -1.0     0.5     NaN     NaN   \n",
       "\n",
       "             MQ MQRankSum     QD ReadPosRankSum    SOR  \n",
       "0           NaN       NaN    NaN            NaN    NaN  \n",
       "1           NaN       NaN    NaN            NaN    NaN  \n",
       "2           NaN       NaN    NaN            NaN    NaN  \n",
       "3           NaN       NaN    NaN            NaN    NaN  \n",
       "4           NaN       NaN    NaN            NaN    NaN  \n",
       "...         ...       ...    ...            ...    ...  \n",
       "9097  41.250000    -0.431   5.76         -1.020  0.419  \n",
       "9098  41.230000    -2.005  16.68         -0.205  0.361  \n",
       "9099  40.189999    -3.699  31.48         -1.691  0.530  \n",
       "9100  44.020000    -1.952  33.41          0.183  2.985  \n",
       "9101  44.020000    -1.952  33.41          0.549  2.985  \n",
       "\n",
       "[444444 rows x 139 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_allel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "set()\n",
      "{'STRQ', 'ClippingRankSum', 'GERMQ', 'MBQ', 'POPAF', 'ReadPosRankSum', 'AMP_total_samples', 'ExcessHet', 'MLEAF', 'AMP_tier', 'MQRankSum', 'RPA', 'MPOS', 'TLOD', 'QD', 'STR', 'cosmicFathMMPrediction', 'AMP_score', 'AN', 'SOR', 'MQ', 'AS_FilterStatus', 'ECNT', 'AMP_rules', 'BaseQRankSum', 'AS_SB_TABLE', 'AF', 'FS', 'MLEAC', 'MMQ', 'cosmicFathMMScore', 'AMP_matches', 'MFRL', 'RU', 'AC'}\n",
      "{'STRQ', 'ClippingRankSum', 'GERMQ', 'MBQ', 'POPAF', 'ReadPosRankSum', 'AMP_total_samples', 'ExcessHet', 'MLEAF', 'AMP_tier', 'MQRankSum', 'RPA', 'MPOS', 'TLOD', 'QD', 'STR', 'cosmicFathMMPrediction', 'AMP_score', 'AN', 'SOR', 'MQ', 'AS_FilterStatus', 'ECNT', 'AMP_rules', 'BaseQRankSum', 'AS_SB_TABLE', 'AF', 'FS', 'MLEAC', 'MMQ', 'cosmicFathMMScore', 'AMP_matches', 'MFRL', 'RU', 'AC'}\n"
     ]
    }
   ],
   "source": [
    "print(set(df_15.columns).symmetric_difference(set(df_50.columns)))\n",
    "print(set(df_15.columns).symmetric_difference(set(df_69.columns)))\n",
    "print(set(df_50.columns).symmetric_difference(set(df_69.columns)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'AC_1',\n",
       " 'AC_2',\n",
       " 'AC_3',\n",
       " 'AF_1',\n",
       " 'AF_2',\n",
       " 'AF_3',\n",
       " 'ALT_1',\n",
       " 'ALT_2',\n",
       " 'ALT_3',\n",
       " 'FILTER_FAIL',\n",
       " 'FILTER_PASS',\n",
       " 'FILTER_SB',\n",
       " 'FILTER_UM',\n",
       " 'FILTER_base_qual',\n",
       " 'FILTER_clustered_events',\n",
       " 'FILTER_fragment',\n",
       " 'FILTER_germline',\n",
       " 'FILTER_haplotype',\n",
       " 'FILTER_map_qual',\n",
       " 'FILTER_multiallelic',\n",
       " 'FILTER_slippage',\n",
       " 'FILTER_strand_bias',\n",
       " 'FILTER_weak_evidence',\n",
       " 'MBQ_1',\n",
       " 'MBQ_2',\n",
       " 'MBQ_3',\n",
       " 'MBQ_4',\n",
       " 'MFRL_1',\n",
       " 'MFRL_2',\n",
       " 'MFRL_3',\n",
       " 'MFRL_4',\n",
       " 'MLEAC_1',\n",
       " 'MLEAC_2',\n",
       " 'MLEAC_3',\n",
       " 'MLEAF_1',\n",
       " 'MLEAF_2',\n",
       " 'MLEAF_3',\n",
       " 'MMQ_1',\n",
       " 'MMQ_2',\n",
       " 'MMQ_3',\n",
       " 'MMQ_4',\n",
       " 'MPOS_1',\n",
       " 'MPOS_2',\n",
       " 'MPOS_3',\n",
       " 'POPAF_1',\n",
       " 'POPAF_2',\n",
       " 'POPAF_3',\n",
       " 'RPA_1',\n",
       " 'RPA_2',\n",
       " 'RPA_3',\n",
       " 'RPA_4',\n",
       " 'TLOD_1',\n",
       " 'TLOD_2',\n",
       " 'TLOD_3',\n",
       " 'gnomadExomesEthnic_AC_Hom_1',\n",
       " 'gnomadExomesEthnic_AC_Hom_2',\n",
       " 'gnomadExomesEthnic_AC_Hom_3',\n",
       " 'gnomadExomes_AC_1',\n",
       " 'gnomadExomes_AC_2',\n",
       " 'gnomadExomes_AC_3',\n",
       " 'gnomadExomes_AC_Hemi_1',\n",
       " 'gnomadExomes_AC_Hemi_2',\n",
       " 'gnomadExomes_AC_Hemi_3',\n",
       " 'gnomadExomes_AC_Hom_1',\n",
       " 'gnomadExomes_AC_Hom_2',\n",
       " 'gnomadExomes_AC_Hom_3',\n",
       " 'gnomadExomes_AF_1',\n",
       " 'gnomadExomes_AF_2',\n",
       " 'gnomadExomes_AF_3',\n",
       " 'gnomadExomes_AN_1',\n",
       " 'gnomadExomes_AN_2',\n",
       " 'gnomadExomes_AN_3',\n",
       " 'gnomadGenomesAC_1',\n",
       " 'gnomadGenomesAC_2',\n",
       " 'gnomadGenomesAC_3',\n",
       " 'gnomadGenomesAN_1',\n",
       " 'gnomadGenomesAN_2',\n",
       " 'gnomadGenomesAN_3',\n",
       " 'gnomadGenomesEthnic_AC_Hom_1',\n",
       " 'gnomadGenomesEthnic_AC_Hom_2',\n",
       " 'gnomadGenomesEthnic_AC_Hom_3',\n",
       " 'gnomadGenomes_AC_Hemi_1',\n",
       " 'gnomadGenomes_AC_Hemi_2',\n",
       " 'gnomadGenomes_AC_Hemi_3',\n",
       " 'gnomadGenomes_AC_Hom_1',\n",
       " 'gnomadGenomes_AC_Hom_2',\n",
       " 'gnomadGenomes_AC_Hom_3',\n",
       " 'gnomadGenomes_AF_1',\n",
       " 'gnomadGenomes_AF_2',\n",
       " 'gnomadGenomes_AF_3'}"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "allel_csv_difference = set(df_allel.columns).difference(set(df_csv.columns))\n",
    "allel_csv_difference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([-1., nan])"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_allel[\"MBQ_3\"].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([nan,  1.,  2.])"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_csv[\"AC\"].unique()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Columns split (seemingly unnecessarily) into 3/4 columns in allel:\n",
    "- AC, MBQ, FILTER, MLEAC, MLEAF, MFRL, MPOS, POPAF, RPA, TLOD\n",
    "- gnomadExomesEthnic_AC_xxx, gnomadExomesEthnic_AF_xxx, NOT gnomadGenomes_AF_ethnic "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "pathogen",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
