{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "from utils import get_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\Users\\barte\\Desktop\\Studies\\V semester\\pathogenicity-assessment\\unvcf_tests\\utils.py:43: DtypeWarning: Columns (12) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df2 = pd.read_csv(folder_path / f\"{sample}_{f_type}.csv.gz\", sep=\";\", compression=\"gzip\").drop(\"Unnamed: 0\", axis=1)\n"
     ]
    }
   ],
   "source": [
    "df_015 = get_data(\"../data/EE_015/\", [\"default\", \"genotype\"])\n",
    "df_050 = get_data(\"../data/EE_050/\", [\"default\", \"genotype\"])\n",
    "df_069 = get_data(\"../data/EE_069/\", [\"default\", \"genotype\"])\n",
    "\n",
    "unvcf_df = pd.concat([df_015, df_050, df_069], ignore_index=True, axis=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>ACMG_class</th>\n",
       "      <th>ACMG_coding_impact</th>\n",
       "      <th>ACMG_gene</th>\n",
       "      <th>...</th>\n",
       "      <th>ClippingRankSum</th>\n",
       "      <th>ExcessHet</th>\n",
       "      <th>FS</th>\n",
       "      <th>MLEAC</th>\n",
       "      <th>MLEAF</th>\n",
       "      <th>MQ</th>\n",
       "      <th>MQRankSum</th>\n",
       "      <th>QD</th>\n",
       "      <th>ReadPosRankSum</th>\n",
       "      <th>SOR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>15820</td>\n",
       "      <td>rs2691315</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>Uncertain%40Significance</td>\n",
       "      <td>non%40coding</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 74 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  #CHROM    POS         ID REF ALT QUAL FILTER                ACMG_class  \\\n",
       "0   chr1  15820  rs2691315   G   T    .   PASS  Uncertain%40Significance   \n",
       "\n",
       "  ACMG_coding_impact ACMG_gene  ... ClippingRankSum  ExcessHet  FS MLEAC  \\\n",
       "0       non%40coding         0  ...             NaN        NaN NaN   NaN   \n",
       "\n",
       "  MLEAF  MQ MQRankSum  QD ReadPosRankSum SOR  \n",
       "0   NaN NaN       NaN NaN            NaN NaN  \n",
       "\n",
       "[1 rows x 74 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unvcf_df.head(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dtype: float64:\n",
      " - ACMG_score\n",
      " - AMP_score\n",
      " - AMP_total_samples\n",
      " - ECNT\n",
      " - GERMQ\n",
      " - MBQ\n",
      " - MFRL\n",
      " - MMQ\n",
      " - MPOS\n",
      " - POPAF\n",
      " - RPA\n",
      " - STR\n",
      " - STRQ\n",
      " - TLOD\n",
      " - cosmicFathMMScore\n",
      " - gnomadExomes_AF\n",
      " - gnomadExomes_AF_ethnic\n",
      " - gnomadGenomes_AF\n",
      " - gnomadGenomes_AF_ethnic\n",
      " - AC\n",
      " - AF\n",
      " - AN\n",
      " - ExcessHet\n",
      " - FS\n",
      " - MLEAC\n",
      " - MLEAF\n",
      " - MQ\n",
      " - QD\n",
      " - SOR\n",
      "Dtype: object:\n",
      " - #CHROM\n",
      " - ID\n",
      " - REF\n",
      " - ALT\n",
      " - QUAL\n",
      " - FILTER\n",
      " - ACMG_class\n",
      " - ACMG_coding_impact\n",
      " - ACMG_gene\n",
      " - ACMG_rules\n",
      " - ACMG_transcript\n",
      " - AMP_matches\n",
      " - AMP_rules\n",
      " - AMP_tier\n",
      " - AS_FilterStatus\n",
      " - AS_SB_TABLE\n",
      " - CGDinheritance\n",
      " - ClinVarClass\n",
      " - Gene\n",
      " - MutationTaster_pred\n",
      " - MutationTaster_score\n",
      " - RU\n",
      " - SIFT_score\n",
      " - coding_impact\n",
      " - cosmicFathMMPrediction\n",
      " - function\n",
      " - hgvs\n",
      " - BaseQRankSum\n",
      " - ClippingRankSum\n",
      " - MQRankSum\n",
      " - ReadPosRankSum\n",
      "Dtype: int64:\n",
      " - POS\n",
      " - ClinVarDisease\n",
      " - DANN_score\n",
      " - DP\n",
      " - gnomadExomesEthnic_AC_Hom\n",
      " - gnomadExomes_AC\n",
      " - gnomadExomes_AC_Hemi\n",
      " - gnomadExomes_AC_Hom\n",
      " - gnomadExomes_AN\n",
      " - gnomadGenomesAC\n",
      " - gnomadGenomesAN\n",
      " - gnomadGenomesEthnic_AC_Hom\n",
      " - gnomadGenomes_AC_Hemi\n",
      " - gnomadGenomes_AC_Hom\n"
     ]
    }
   ],
   "source": [
    "dt = unvcf_df.dtypes.to_numpy()\n",
    "cols = unvcf_df.columns\n",
    "dt_cols = [(d, col) for d, col in zip(dt, cols)]\n",
    "for d in set(dt):\n",
    "    print(f\"Dtype: {d}:\")\n",
    "    for _d, col in dt_cols:\n",
    "        if d == _d:\n",
    "            print(f\" - {col}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['#CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'ACMG_class',\n",
       "       'ACMG_coding_impact', 'ACMG_gene', 'ACMG_rules', 'ACMG_score',\n",
       "       'ACMG_transcript', 'AMP_matches', 'AMP_rules', 'AMP_score', 'AMP_tier',\n",
       "       'AMP_total_samples', 'AS_FilterStatus', 'AS_SB_TABLE', 'CGDinheritance',\n",
       "       'ClinVarClass', 'ClinVarDisease', 'DANN_score', 'DP', 'ECNT', 'GERMQ',\n",
       "       'Gene', 'MBQ', 'MFRL', 'MMQ', 'MPOS', 'MutationTaster_pred',\n",
       "       'MutationTaster_score', 'POPAF', 'RPA', 'RU', 'SIFT_score', 'STR',\n",
       "       'STRQ', 'TLOD', 'coding_impact', 'cosmicFathMMPrediction',\n",
       "       'cosmicFathMMScore', 'function', 'gnomadExomesEthnic_AC_Hom',\n",
       "       'gnomadExomes_AC', 'gnomadExomes_AC_Hemi', 'gnomadExomes_AC_Hom',\n",
       "       'gnomadExomes_AF', 'gnomadExomes_AF_ethnic', 'gnomadExomes_AN',\n",
       "       'gnomadGenomesAC', 'gnomadGenomesAN', 'gnomadGenomesEthnic_AC_Hom',\n",
       "       'gnomadGenomes_AC_Hemi', 'gnomadGenomes_AC_Hom', 'gnomadGenomes_AF',\n",
       "       'gnomadGenomes_AF_ethnic', 'hgvs', 'AC', 'AF', 'AN', 'BaseQRankSum',\n",
       "       'ClippingRankSum', 'ExcessHet', 'FS', 'MLEAC', 'MLEAF', 'MQ',\n",
       "       'MQRankSum', 'QD', 'ReadPosRankSum', 'SOR'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unvcf_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all(df_050.columns == df_015.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'#CHROM',\n",
       " 'ACMG_class',\n",
       " 'ACMG_coding_impact',\n",
       " 'ACMG_gene',\n",
       " 'ACMG_rules',\n",
       " 'ACMG_score',\n",
       " 'ACMG_transcript',\n",
       " 'ALT',\n",
       " 'CGDinheritance',\n",
       " 'ClinVarClass',\n",
       " 'ClinVarDisease',\n",
       " 'DANN_score',\n",
       " 'DP',\n",
       " 'FILTER',\n",
       " 'Gene',\n",
       " 'ID',\n",
       " 'MutationTaster_pred',\n",
       " 'MutationTaster_score',\n",
       " 'POS',\n",
       " 'QUAL',\n",
       " 'REF',\n",
       " 'SIFT_score',\n",
       " 'coding_impact',\n",
       " 'function',\n",
       " 'gnomadExomesEthnic_AC_Hom',\n",
       " 'gnomadExomes_AC',\n",
       " 'gnomadExomes_AC_Hemi',\n",
       " 'gnomadExomes_AC_Hom',\n",
       " 'gnomadExomes_AF',\n",
       " 'gnomadExomes_AF_ethnic',\n",
       " 'gnomadExomes_AN',\n",
       " 'gnomadGenomesAC',\n",
       " 'gnomadGenomesAN',\n",
       " 'gnomadGenomesEthnic_AC_Hom',\n",
       " 'gnomadGenomes_AC_Hemi',\n",
       " 'gnomadGenomes_AC_Hom',\n",
       " 'gnomadGenomes_AF',\n",
       " 'gnomadGenomes_AF_ethnic',\n",
       " 'hgvs'}"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(df_050.columns).intersection(set(df_069.columns))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'AMP_matches',\n",
       " 'AMP_rules',\n",
       " 'AMP_score',\n",
       " 'AMP_tier',\n",
       " 'AMP_total_samples',\n",
       " 'AS_FilterStatus',\n",
       " 'AS_SB_TABLE',\n",
       " 'ECNT',\n",
       " 'GERMQ',\n",
       " 'MBQ',\n",
       " 'MFRL',\n",
       " 'MMQ',\n",
       " 'MPOS',\n",
       " 'POPAF',\n",
       " 'RPA',\n",
       " 'RU',\n",
       " 'STR',\n",
       " 'STRQ',\n",
       " 'TLOD',\n",
       " 'cosmicFathMMPrediction',\n",
       " 'cosmicFathMMScore'}"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(df_050.columns).difference(set(df_069.columns))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'AC',\n",
       " 'AF',\n",
       " 'AN',\n",
       " 'BaseQRankSum',\n",
       " 'ClippingRankSum',\n",
       " 'ExcessHet',\n",
       " 'FS',\n",
       " 'MLEAC',\n",
       " 'MLEAF',\n",
       " 'MQ',\n",
       " 'MQRankSum',\n",
       " 'QD',\n",
       " 'ReadPosRankSum',\n",
       " 'SOR'}"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(df_069.columns).difference(set(df_050.columns))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\Users\\barte\\Desktop\\Studies\\V semester\\pathogenicity-assessment\\unvcf_tests\\utils.py:43: DtypeWarning: Columns (26,40,74,75,76,78,80) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df2 = pd.read_csv(folder_path / f\"{sample}_{f_type}.csv.gz\", sep=\";\", compression=\"gzip\").drop(\"Unnamed: 0\", axis=1)\n",
      "c:\\Users\\barte\\Desktop\\Studies\\V semester\\pathogenicity-assessment\\unvcf_tests\\utils.py:43: DtypeWarning: Columns (26,40,74,75,76,78,80,94) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df2 = pd.read_csv(folder_path / f\"{sample}_{f_type}.csv.gz\", sep=\";\", compression=\"gzip\").drop(\"Unnamed: 0\", axis=1)\n",
      "c:\\Users\\barte\\Desktop\\Studies\\V semester\\pathogenicity-assessment\\unvcf_tests\\utils.py:43: DtypeWarning: Columns (26,40,74,75,76,78,80,96) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df2 = pd.read_csv(folder_path / f\"{sample}_{f_type}.csv.gz\", sep=\";\", compression=\"gzip\").drop(\"Unnamed: 0\", axis=1)\n"
     ]
    }
   ],
   "source": [
    "df_015_csq = get_data(\"../data/EE_015/\", [\"default\", \"csq\"])\n",
    "df_050_csq = get_data(\"../data/EE_050/\", [\"default\", \"csq\"])\n",
    "df_069_csq = get_data(\"../data/EE_069/\", [\"default\", \"csq\"])\n",
    "\n",
    "unvcf_df_csq = pd.concat([df_015_csq, df_050_csq, df_069_csq], ignore_index=True, axis=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>...</th>\n",
       "      <th>SpliceAI_pred_DP_DG</th>\n",
       "      <th>SpliceAI_pred_DP_DL</th>\n",
       "      <th>SpliceAI_pred_DS_AG</th>\n",
       "      <th>SpliceAI_pred_DS_AL</th>\n",
       "      <th>SpliceAI_pred_DS_DG</th>\n",
       "      <th>SpliceAI_pred_DS_DL</th>\n",
       "      <th>SpliceAI_pred_SYMBOL</th>\n",
       "      <th>LOEUF</th>\n",
       "      <th>PHENOTYPES</th>\n",
       "      <th>NMD</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>15820</td>\n",
       "      <td>rs2691315</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>T</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17385</td>\n",
       "      <td>rs201535981</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>A</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17697</td>\n",
       "      <td>rs71260069</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>C</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>133129</td>\n",
       "      <td>rs367730352</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>A</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>183629</td>\n",
       "      <td>rs71267774</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>A</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102512</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56856512</td>\n",
       "      <td>rs76439603</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>C</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102513</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56856517</td>\n",
       "      <td>rs75739861</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>G</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102514</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56856554</td>\n",
       "      <td>rs77088808</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>G</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102515</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56856581</td>\n",
       "      <td>rs9320083</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>A</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102516</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56856587</td>\n",
       "      <td>rs7067511</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>C</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>102517 rows × 103 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       #CHROM       POS           ID REF ALT QUAL FILTER Allele  \\\n",
       "0        chr1     15820    rs2691315   G   T    .   PASS      T   \n",
       "1        chr1     17385  rs201535981   G   A    .   PASS      A   \n",
       "2        chr1     17697   rs71260069   G   C    .   PASS      C   \n",
       "3        chr1    133129  rs367730352   G   A    .   PASS      A   \n",
       "4        chr1    183629   rs71267774   G   A    .   PASS      A   \n",
       "...       ...       ...          ...  ..  ..  ...    ...    ...   \n",
       "102512   chrY  56856512   rs76439603   A   C    .   PASS      C   \n",
       "102513   chrY  56856517   rs75739861   A   G    .   PASS      G   \n",
       "102514   chrY  56856554   rs77088808   T   G    .   PASS      G   \n",
       "102515   chrY  56856581    rs9320083   G   A    .   PASS      A   \n",
       "102516   chrY  56856587    rs7067511   A   C    .   PASS      C   \n",
       "\n",
       "                               Consequence    IMPACT  ... SpliceAI_pred_DP_DG  \\\n",
       "0                  downstream_gene_variant  MODIFIER  ...                 NaN   \n",
       "1                  downstream_gene_variant  MODIFIER  ...                 NaN   \n",
       "2                  downstream_gene_variant  MODIFIER  ...                 NaN   \n",
       "3       non_coding_transcript_exon_variant  MODIFIER  ...                 NaN   \n",
       "4                  downstream_gene_variant  MODIFIER  ...                 NaN   \n",
       "...                                    ...       ...  ...                 ...   \n",
       "102512             downstream_gene_variant  MODIFIER  ...                 NaN   \n",
       "102513             downstream_gene_variant  MODIFIER  ...                 NaN   \n",
       "102514             downstream_gene_variant  MODIFIER  ...                 NaN   \n",
       "102515             downstream_gene_variant  MODIFIER  ...                 NaN   \n",
       "102516             downstream_gene_variant  MODIFIER  ...                 NaN   \n",
       "\n",
       "       SpliceAI_pred_DP_DL SpliceAI_pred_DS_AG SpliceAI_pred_DS_AL  \\\n",
       "0                      NaN                 NaN                 NaN   \n",
       "1                      NaN                 NaN                 NaN   \n",
       "2                      NaN                 NaN                 NaN   \n",
       "3                      NaN                 NaN                 NaN   \n",
       "4                      NaN                 NaN                 NaN   \n",
       "...                    ...                 ...                 ...   \n",
       "102512                 NaN                 NaN                 NaN   \n",
       "102513                 NaN                 NaN                 NaN   \n",
       "102514                 NaN                 NaN                 NaN   \n",
       "102515                 NaN                 NaN                 NaN   \n",
       "102516                 NaN                 NaN                 NaN   \n",
       "\n",
       "       SpliceAI_pred_DS_DG SpliceAI_pred_DS_DL SpliceAI_pred_SYMBOL LOEUF  \\\n",
       "0                      NaN                 NaN                  NaN   NaN   \n",
       "1                      NaN                 NaN                  NaN   NaN   \n",
       "2                      NaN                 NaN                  NaN   NaN   \n",
       "3                      NaN                 NaN                  NaN   NaN   \n",
       "4                      NaN                 NaN                  NaN   NaN   \n",
       "...                    ...                 ...                  ...   ...   \n",
       "102512                 NaN                 NaN                  NaN   NaN   \n",
       "102513                 NaN                 NaN                  NaN   NaN   \n",
       "102514                 NaN                 NaN                  NaN   NaN   \n",
       "102515                 NaN                 NaN                  NaN   NaN   \n",
       "102516                 NaN                 NaN                  NaN   NaN   \n",
       "\n",
       "       PHENOTYPES  NMD  \n",
       "0             NaN  NaN  \n",
       "1             NaN  NaN  \n",
       "2             NaN  NaN  \n",
       "3             NaN  NaN  \n",
       "4             NaN  NaN  \n",
       "...           ...  ...  \n",
       "102512        NaN  NaN  \n",
       "102513        NaN  NaN  \n",
       "102514        NaN  NaN  \n",
       "102515        NaN  NaN  \n",
       "102516        NaN  NaN  \n",
       "\n",
       "[102517 rows x 103 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_015_csq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dtype: float64:\n",
      " - DISTANCE\n",
      " - STRAND\n",
      " - TSL\n",
      " - REFSEQ_MATCH\n",
      " - REFSEQ_OFFSET\n",
      " - HGVS_OFFSET\n",
      " - AF\n",
      " - AFR_AF\n",
      " - AMR_AF\n",
      " - EAS_AF\n",
      " - EUR_AF\n",
      " - SAS_AF\n",
      " - gnomADe_AF\n",
      " - gnomADe_AFR_AF\n",
      " - gnomADe_AMR_AF\n",
      " - gnomADe_ASJ_AF\n",
      " - gnomADe_EAS_AF\n",
      " - gnomADe_FIN_AF\n",
      " - gnomADe_NFE_AF\n",
      " - gnomADe_OTH_AF\n",
      " - gnomADe_SAS_AF\n",
      " - gnomADg_AF\n",
      " - gnomADg_AFR_AF\n",
      " - gnomADg_AMI_AF\n",
      " - gnomADg_AMR_AF\n",
      " - gnomADg_ASJ_AF\n",
      " - gnomADg_EAS_AF\n",
      " - gnomADg_FIN_AF\n",
      " - gnomADg_MID_AF\n",
      " - gnomADg_NFE_AF\n",
      " - gnomADg_OTH_AF\n",
      " - gnomADg_SAS_AF\n",
      " - MOTIF_POS\n",
      " - MOTIF_SCORE_CHANGE\n",
      " - EVE_SCORE\n",
      " - CADD_PHRED\n",
      " - CADD_RAW\n",
      " - SpliceAI_pred_DP_AG\n",
      " - SpliceAI_pred_DP_AL\n",
      " - SpliceAI_pred_DP_DG\n",
      " - SpliceAI_pred_DP_DL\n",
      " - SpliceAI_pred_DS_AG\n",
      " - SpliceAI_pred_DS_AL\n",
      " - SpliceAI_pred_DS_DG\n",
      " - SpliceAI_pred_DS_DL\n",
      " - LOEUF\n",
      "Dtype: object:\n",
      " - #CHROM\n",
      " - ID\n",
      " - REF\n",
      " - ALT\n",
      " - QUAL\n",
      " - FILTER\n",
      " - Allele\n",
      " - Consequence\n",
      " - IMPACT\n",
      " - SYMBOL\n",
      " - Gene\n",
      " - Feature_type\n",
      " - Feature\n",
      " - BIOTYPE\n",
      " - EXON\n",
      " - INTRON\n",
      " - HGVSc\n",
      " - HGVSp\n",
      " - cDNA_position\n",
      " - CDS_position\n",
      " - Protein_position\n",
      " - Amino_acids\n",
      " - Codons\n",
      " - Existing_variation\n",
      " - FLAGS\n",
      " - SYMBOL_SOURCE\n",
      " - HGNC_ID\n",
      " - CANONICAL\n",
      " - MANE_SELECT\n",
      " - MANE_PLUS_CLINICAL\n",
      " - APPRIS\n",
      " - CCDS\n",
      " - ENSP\n",
      " - SWISSPROT\n",
      " - TREMBL\n",
      " - UNIPARC\n",
      " - UNIPROT_ISOFORM\n",
      " - SOURCE\n",
      " - GIVEN_REF\n",
      " - USED_REF\n",
      " - BAM_EDIT\n",
      " - SIFT\n",
      " - PolyPhen\n",
      " - DOMAINS\n",
      " - CLIN_SIG\n",
      " - SOMATIC\n",
      " - PHENO\n",
      " - PUBMED\n",
      " - VAR_SYNONYMS\n",
      " - MOTIF_NAME\n",
      " - HIGH_INF_POS\n",
      " - TRANSCRIPTION_FACTORS\n",
      " - EVE_CLASS\n",
      " - SpliceAI_pred_SYMBOL\n",
      " - PHENOTYPES\n",
      " - NMD\n",
      "Dtype: int64:\n",
      " - POS\n"
     ]
    }
   ],
   "source": [
    "dt = unvcf_df_csq.dtypes.to_numpy()\n",
    "cols = unvcf_df_csq.columns\n",
    "dt_cols = [(d, col) for d, col in zip(dt, cols)]\n",
    "for d in set(dt):\n",
    "    print(f\"Dtype: {d}:\")\n",
    "    for _d, col in dt_cols:\n",
    "        if d == _d:\n",
    "            print(f\" - {col}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(df_050_csq.columns) == set(df_015_csq.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(df_050_csq.columns) == set(df_069_csq.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(df_015_csq.columns) == set(df_069_csq.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "49"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "object_columns = set(unvcf_df_csq.select_dtypes(object).columns).difference(set(unvcf_df.select_dtypes(object).columns))\n",
    "object_columns = list(object_columns)\n",
    "len(object_columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "UNIPARC\n",
      "18096\n",
      "[nan 'UPI000041820C' 'UPI00001DFBF0' 'UPI000022DAF4' 'UPI00001D7C8B']\n",
      "UPI000041820C\n",
      "========================================\n",
      "1\n",
      "APPRIS\n",
      "8\n",
      "[nan 'P1' 'A2' 'P4' 'P3']\n",
      "P1\n",
      "========================================\n",
      "2\n",
      "SYMBOL\n",
      "23275\n",
      "['DDX11L1' 'CICP27' 'MIR6859-2' nan 'LINC01409']\n",
      "PTGER4P2-CDK2AP2P2\n",
      "========================================\n",
      "3\n",
      "EXON\n",
      "2107\n",
      "[nan '1/1' '12/16' '1/5' '19/19']\n",
      "107/107\n",
      "========================================\n",
      "4\n",
      "Amino_acids\n",
      "794\n",
      "[nan 'T' 'E' 'I/V' 'V']\n",
      "PPMTVTTSGTSQSRSSFSTAKTSTSLHSHTSSTHHPEVTSTSTTSITPNHTST/-\n",
      "========================================\n",
      "5\n",
      "SOURCE\n",
      "3\n",
      "['Ensembl' nan 'RefSeq']\n",
      "Ensembl\n",
      "========================================\n",
      "6\n",
      "USED_REF\n",
      "2777\n",
      "['G' 'C' 'T' 'A' nan]\n",
      "CTAATCGGCCCATCTCTTTCCCTTTCTGTGTGGTTACCAGGTTACTGGAGGCCACTTAGTTGTTGGTTTTGGTTGGGACTTTCTGGACTTCAGATCTCAGCCCTCGTTTACTCACCCCTGAGCCCTCCCTGCCCTGCCCCCAGCAGTTCCTGACTTCGCCCCTATACACTGTCCCCGGAGCTCTTCCCTGGGC\n",
      "========================================\n",
      "7\n",
      "PHENOTYPES\n",
      "54932\n",
      "[nan 'Retinitis_pigmentosa+G2P+ENSG00000187634'\n",
      " 'Retinitis_pigmentosa+G2P+ENSG00000187634&ClinVar:_phenotype_not_specified+ClinVar+rs6672356'\n",
      " 'Skin_reflectance__Melanin_index_+NHGRI-EBI_GWAS_catalog+rs2272756'\n",
      " 'Pancreatic_Cancer+NHGRI-EBI_GWAS_catalog+rs13303010']\n",
      "FASTING_PLASMA_GLUCOSE_LEVEL_QUANTITATIVE_TRAIT_LOCUS_5+MIM_morbid+ENSG00000084734&ClinVar:_phenotype_not_specified+ClinVar+rs1260326&FASTING_PLASMA_GLUCOSE_LEVEL_QUANTITATIVE_TRAIT_LOCUS_5+ClinVar+rs1260326&Cholesterol+dbGaP+rs1260326&Cholesterol+dbGaP+rs1260326&Triglycerides+dbGaP+rs1260326&Triglycerides+dbGaP+rs1260326&Fasting_Glucose+MAGIC+rs1260326&1-carboxyethylisoleucine_levels+NHGRI-EBI_GWAS_catalog+rs1260326&1-carboxyethylleucine_levels+NHGRI-EBI_GWAS_catalog+rs1260326&1-carboxyethylphenylalanine_levels+NHGRI-EBI_GWAS_catalog+rs1260326&1-carboxyethylvaline_levels+NHGRI-EBI_GWAS_catalog+rs1260326&1-myristoyl-2-arachidonoyl-GPC_14:0/20:4_levels+NHGRI-EBI_GWAS_catalog+rs1260326&1-myristoyl-2-docosahexaenoyl-GPC_14:0/22:6_levels+NHGRI-EBI_GWAS_catalog+rs1260326&1-myristoyl-2-linoleoyl-GPC_14:0/18:2_levels+NHGRI-EBI_GWAS_catalog+rs1260326&1-myristoylglycerol_14:0_levels+NHGRI-EBI_GWAS_catalog+rs1260326&1-myristoyl-GPC_14:0_levels+NHGRI-EBI_GWAS_catalog+rs1260326&1-palmitoleoylglycerol_16:1_levels+NHGRI-EBI_GWAS_catalog+rs1260326&1-palmitoyl-2-oleoyl-GPE_16:0/18:1_levels+NHGRI-EBI_GWAS_catalog+rs1260326&1-palmitoyl-GPE_16:0_levels+NHGRI-EBI_GWAS_catalog+rs1260326&1-stearoyl-2-docosahexaenoyl-GPE_18:0/22:6_levels+NHGRI-EBI_GWAS_catalog+rs1260326&1-stearoyl-2-oleoyl-GPE_18:0/18:1_levels+NHGRI-EBI_GWAS_catalog+rs1260326&1-stearoyl-GPE_18:0_levels+NHGRI-EBI_GWAS_catalog+rs1260326&2-myristoyl-GPC_14:0_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Acetate_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Alanine_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Alanine_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Alanine_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Albumin_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Albumin_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Albumin_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Alcohol_consumption_drinkers_vs_non-drinkers+NHGRI-EBI_GWAS_catalog+rs1260326&Alcohol_consumption_drinks_per_month_UKB_data_field_1578__4424+NHGRI-EBI_GWAS_catalog+rs1260326&Alcohol_consumption__drinks_per_week___MTAG_+NHGRI-EBI_GWAS_catalog+rs1260326&Alcohol_consumption__drinks_per_week___MTAG_+NHGRI-EBI_GWAS_catalog+rs1260326&Alcohol_consumption_drinks_per_week+NHGRI-EBI_GWAS_catalog+rs1260326&Alcohol_consumption_drinks_per_week+NHGRI-EBI_GWAS_catalog+rs1260326&Alcohol_consumption__heavy_vs._light/non-drinkers_+NHGRI-EBI_GWAS_catalog+rs1260326&Alcohol_consumption_in_current_drinkers+NHGRI-EBI_GWAS_catalog+rs1260326&Alcohol_consumption+NHGRI-EBI_GWAS_catalog+rs1260326&Alcohol_consumption+NHGRI-EBI_GWAS_catalog+rs1260326&Alcohol_use_disorder__consumption_score_+NHGRI-EBI_GWAS_catalog+rs1260326&Alcohol_use_disorder__consumption_score_+NHGRI-EBI_GWAS_catalog+rs1260326&Alcohol_use_disorder__total_score_+NHGRI-EBI_GWAS_catalog+rs1260326&Alcohol_use_disorder+NHGRI-EBI_GWAS_catalog+rs1260326&Alcohol_use_disorder+NHGRI-EBI_GWAS_catalog+rs1260326&Apolipoprotein_A1_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Apolipoprotein_B_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Appendicular_lean_mass+NHGRI-EBI_GWAS_catalog+rs1260326&Appendicular_lean_mass+NHGRI-EBI_GWAS_catalog+rs1260326&Appendicular_lean_mass+NHGRI-EBI_GWAS_catalog+rs1260326&Asparagine_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Aspartate_aminotransferase_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Average_diameter_for_HDL_particles+NHGRI-EBI_GWAS_catalog+rs1260326&Average_diameter_for_LDL_particles+NHGRI-EBI_GWAS_catalog+rs1260326&Average_diameter_for_VLDL_particles+NHGRI-EBI_GWAS_catalog+rs1260326&Biological_age_PhenoAge+NHGRI-EBI_GWAS_catalog+rs1260326&Bitter_alcoholic_beverage_consumption+NHGRI-EBI_GWAS_catalog+rs1260326&Bitter_beverage_consumption+NHGRI-EBI_GWAS_catalog+rs1260326&Blood_metabolite_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Blood_protein_levels_in_cardiovascular_risk+NHGRI-EBI_GWAS_catalog+rs1260326&Blood_sugar_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Blood_urea_nitrogen_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Blood_urea_nitrogen_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Body_mass_index_and_LDL-C_pairwise+NHGRI-EBI_GWAS_catalog+rs1260326&Body_mass_index_and_triglycerides_pairwise+NHGRI-EBI_GWAS_catalog+rs1260326&Body_mass_index_MTAG+NHGRI-EBI_GWAS_catalog+rs1260326&Body_Mass_Index+NHGRI-EBI_GWAS_catalog+rs1260326&Body_surface_area+NHGRI-EBI_GWAS_catalog+rs1260326&Branched-chain_amino_acid_levels__Isoleucine_+NHGRI-EBI_GWAS_catalog+rs1260326&Branched-chain_amino_acid_levels__Leucine_+NHGRI-EBI_GWAS_catalog+rs1260326&Branched-chain_amino_acid_levels__Valine_+NHGRI-EBI_GWAS_catalog+rs1260326&Caffeine_consumption_from_coffee_or_tea+NHGRI-EBI_GWAS_catalog+rs1260326&Caffeine_metabolism__plasma_1_7-dimethylxanthine__paraxanthine__to_1_3_7-trimethylxanthine__caffeine__ratio_+NHGRI-EBI_GWAS_catalog+rs1260326&Calcium_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Calcium_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Cardiovascular_disease_risk_factors+NHGRI-EBI_GWAS_catalog+rs1260326&Cathepsin_D_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Chitinase-3-like_protein_1_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Cholecystitis+NHGRI-EBI_GWAS_catalog+rs1260326&Cholelithiasis+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesterol_levels_in_chylomicrons_and_extremely_large_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesterol_levels_in_large_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesterol_levels_in_large_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesterol_levels_in_large_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesterol_levels_in_medium_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesterol_levels_in_medium_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesterol_levels_in_small_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesterol_levels_in_small_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesterol_levels_in_small_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesterol_levels_in_very_large_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesterol_levels_in_very_large_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesterol__total+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesterol__total+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesterol__total+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesterol_to_total_lipids_ratio_in_IDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesterol_to_total_lipids_ratio_in_large_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesterol_to_total_lipids_ratio_in_large_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesterol_to_total_lipids_ratio_in_large_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesterol_to_total_lipids_ratio_in_medium_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesterol_to_total_lipids_ratio_in_medium_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesterol_to_total_lipids_ratio_in_medium_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesterol_to_total_lipids_ratio_in_small_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesterol_to_total_lipids_ratio_in_small_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesterol_to_total_lipids_ratio_in_very_large_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesterol_to_total_lipids_ratio_in_very_small_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesteryl_ester_levels_in_chylomicrons_and_extremely_large_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesteryl_ester_levels_in_large_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesteryl_ester_levels_in_large_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesteryl_ester_levels_in_large_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesteryl_ester_levels_in_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesteryl_ester_levels_in_medium_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesteryl_ester_levels_in_medium_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesteryl_ester_levels_in_small_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesteryl_ester_levels_in_small_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesteryl_ester_levels_in_small_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesteryl_ester_levels_in_very_large_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesteryl_ester_levels_in_very_large_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesteryl_ester_levels_in_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesteryl_esters_to_total_lipids_ratio_in_IDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesteryl_esters_to_total_lipids_ratio_in_large_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesteryl_esters_to_total_lipids_ratio_in_large_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesteryl_esters_to_total_lipids_ratio_in_large_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesteryl_esters_to_total_lipids_ratio_in_medium_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesteryl_esters_to_total_lipids_ratio_in_medium_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesteryl_esters_to_total_lipids_ratio_in_medium_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesteryl_esters_to_total_lipids_ratio_in_small_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesteryl_esters_to_total_lipids_ratio_in_small_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesteryl_esters_to_total_lipids_ratio_in_small_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesteryl_esters_to_total_lipids_ratio_in_very_large_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesteryl_esters_to_total_lipids_ratio_in_very_small_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Chronic_inflammatory_diseases__ankylosing_spondylitis__Crohn's_disease__psoriasis__primary_sclerosing_cholangitis__ulcerative_colitis___pleiotropy_+NHGRI-EBI_GWAS_catalog+rs1260326&Chronic_kidney_disease+NHGRI-EBI_GWAS_catalog+rs1260326&Circulating_docosahexaenoic_acid_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Circulating_fibroblast_growth_factor_21_levels+NHGRI-EBI_GWAS_catalog+rs1260326&circulating_leptin_levels_adjusted_for_BMI+NHGRI-EBI_GWAS_catalog+rs1260326&circulating_leptin_levels_adjusted_for_BMI+NHGRI-EBI_GWAS_catalog+rs1260326&circulating_leptin_levels_adjusted_for_BMI+NHGRI-EBI_GWAS_catalog+rs1260326&circulating_leptin_levels+NHGRI-EBI_GWAS_catalog+rs1260326&circulating_leptin_levels+NHGRI-EBI_GWAS_catalog+rs1260326&circulating_leptin_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Coagulation_factor_IXab_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Coagulation_factor_IX_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Coffee_consumption__cups_per_day_+NHGRI-EBI_GWAS_catalog+rs1260326&Coffee_consumption__cups_per_day_+NHGRI-EBI_GWAS_catalog+rs1260326&Coffee_consumption+NHGRI-EBI_GWAS_catalog+rs1260326&Coffee_consumption+NHGRI-EBI_GWAS_catalog+rs1260326&Concentration_of_chylomicrons_and_extremely_large_VLDL_particles+NHGRI-EBI_GWAS_catalog+rs1260326&Concentration_of_HDL_particles+NHGRI-EBI_GWAS_catalog+rs1260326&Concentration_of_large_HDL_particles+NHGRI-EBI_GWAS_catalog+rs1260326&Concentration_of_large_VLDL_particles+NHGRI-EBI_GWAS_catalog+rs1260326&Concentration_of_medium_HDL_particles+NHGRI-EBI_GWAS_catalog+rs1260326&Concentration_of_medium_LDL_particles+NHGRI-EBI_GWAS_catalog+rs1260326&Concentration_of_medium_VLDL_particles+NHGRI-EBI_GWAS_catalog+rs1260326&Concentration_of_small_HDL_particles+NHGRI-EBI_GWAS_catalog+rs1260326&Concentration_of_small_LDL_particles+NHGRI-EBI_GWAS_catalog+rs1260326&Concentration_of_small_VLDL_particles+NHGRI-EBI_GWAS_catalog+rs1260326&Concentration_of_very_large_HDL_particles+NHGRI-EBI_GWAS_catalog+rs1260326&Concentration_of_very_large_VLDL_particles+NHGRI-EBI_GWAS_catalog+rs1260326&Concentration_of_very_small_VLDL_particles+NHGRI-EBI_GWAS_catalog+rs1260326&Concentration_of_VLDL_particles+NHGRI-EBI_GWAS_catalog+rs1260326&Coronary_artery_disease_or_factor_VII_levels_pleiotropy+NHGRI-EBI_GWAS_catalog+rs1260326&C-reactive_protein_levels_MTAG+NHGRI-EBI_GWAS_catalog+rs1260326&C-reactive_protein_levels_or_total_cholesterol_levels__pleiotropy_+NHGRI-EBI_GWAS_catalog+rs1260326&C-reactive_protein_levels_or_triglyceride_levels__pleiotropy_+NHGRI-EBI_GWAS_catalog+rs1260326&C-reactive_protein_levels+NHGRI-EBI_GWAS_catalog+rs1260326&C-reactive_protein_levels+NHGRI-EBI_GWAS_catalog+rs1260326&C-reactive_protein_levels+NHGRI-EBI_GWAS_catalog+rs1260326&C-reactive_protein_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Creatinine_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Crohn's_disease+NHGRI-EBI_GWAS_catalog+rs1260326&Degree_of_unsaturation+NHGRI-EBI_GWAS_catalog+rs1260326&Diacylglycerol_32:2_[M+H-H2O]1+_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Diacylglycerol_34:1_[M+H-H2O]1+_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Diacylglycerol_34:2_[M+H-H2O]1+_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Diacylglycerol_34:3_[M+H-H2O]1+_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Diacylglycerol_44:7_[M+H-H2O]1+_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Docosahexaenoic_acid_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Drinks_per_week+NHGRI-EBI_GWAS_catalog+rs1260326&Drinks_per_week+NHGRI-EBI_GWAS_catalog+rs1260326&Drinks_per_week+NHGRI-EBI_GWAS_catalog+rs1260326&Dyslipidemia+NHGRI-EBI_GWAS_catalog+rs1260326&Estimated_glomerular_filtration_rate_creatinine+NHGRI-EBI_GWAS_catalog+rs1260326&Estimated_glomerular_filtration_rate_creatinine+NHGRI-EBI_GWAS_catalog+rs1260326&Estimated_glomerular_filtration_rate_cystatin_c+NHGRI-EBI_GWAS_catalog+rs1260326&Estimated_glomerular_filtration_rate_in_non-diabetics+NHGRI-EBI_GWAS_catalog+rs1260326&Estimated_glomerular_filtration_rate+NHGRI-EBI_GWAS_catalog+rs1260326&Estimated_glomerular_filtration_rate+NHGRI-EBI_GWAS_catalog+rs1260326&Estimated_glomerular_filtration_rate+NHGRI-EBI_GWAS_catalog+rs1260326&Estradiol_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Factor_VII_activity_or_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Factor_VII_activity+NHGRI-EBI_GWAS_catalog+rs1260326&Factor_VII_activity+NHGRI-EBI_GWAS_catalog+rs1260326&Factor_VII_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Familial_combined_hyperlipidemia_defined_by_Brunzell_criteria+NHGRI-EBI_GWAS_catalog+rs1260326&Familial_combined_hyperlipidemia_defined_by_Consensus_criteria+NHGRI-EBI_GWAS_catalog+rs1260326&Familial_combined_hyperlipidemia_defined_by_Dutch_criteria+NHGRI-EBI_GWAS_catalog+rs1260326&Familial_combined_hyperlipidemia_defined_by_Goldstein_criteria+NHGRI-EBI_GWAS_catalog+rs1260326&Familial_combined_hyperlipidemia_defined_by_Mexico_criteria+NHGRI-EBI_GWAS_catalog+rs1260326&Fasting_blood_glucose_adjusted_for_BMI+NHGRI-EBI_GWAS_catalog+rs1260326&Fasting_blood_glucose+NHGRI-EBI_GWAS_catalog+rs1260326&Fasting_blood_insulin_adjusted_for_BMI+NHGRI-EBI_GWAS_catalog+rs1260326&Fasting_glucose_adjusted_for_BMI+NHGRI-EBI_GWAS_catalog+rs1260326&Fasting_glucose_adjusted_for_BMI+NHGRI-EBI_GWAS_catalog+rs1260326&Fasting_Glucose+NHGRI-EBI_GWAS_catalog+rs1260326&Fasting_Glucose+NHGRI-EBI_GWAS_catalog+rs1260326&Fasting_Glucose+NHGRI-EBI_GWAS_catalog+rs1260326&Fasting_insulin_adjusted_for_BMI+NHGRI-EBI_GWAS_catalog+rs1260326&Fasting_insulin+NHGRI-EBI_GWAS_catalog+rs1260326&Fasting_insulin+NHGRI-EBI_GWAS_catalog+rs1260326&Fat-free_mass+NHGRI-EBI_GWAS_catalog+rs1260326&Fat-free_mass+NHGRI-EBI_GWAS_catalog+rs1260326&Fat-free_mass+NHGRI-EBI_GWAS_catalog+rs1260326&Fibrinogen_levels_or_factor_VII_levels_or_factor_XI_levels_or_tissue_plasminogen_activator_levels_pleiotropy+NHGRI-EBI_GWAS_catalog+rs1260326&Fibrinogen_levels_or_factor_VII_levels_pleiotropy+NHGRI-EBI_GWAS_catalog+rs1260326&Fibroblast_growth_factor_21_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Follistatin_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Free_cholesterol_levels_in_chylomicrons_and_extremely_large_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Free_cholesterol_levels_in_large_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Free_cholesterol_levels_in_large_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Free_cholesterol_levels_in_medium_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Free_cholesterol_levels_in_medium_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Free_cholesterol_levels_in_small_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Free_cholesterol_levels_in_small_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Free_cholesterol_levels_in_very_large_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Free_cholesterol_levels_in_very_large_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Free_cholesterol_levels_in_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Free_cholesterol_to_total_lipids_ratio_in_chylomicrons_and_extremely_large_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Free_cholesterol_to_total_lipids_ratio_in_IDL+NHGRI-EBI_GWAS_catalog+rs1260326&Free_cholesterol_to_total_lipids_ratio_in_large_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Free_cholesterol_to_total_lipids_ratio_in_large_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Free_cholesterol_to_total_lipids_ratio_in_medium_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Free_cholesterol_to_total_lipids_ratio_in_medium_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Free_cholesterol_to_total_lipids_ratio_in_small_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Free_cholesterol_to_total_lipids_ratio_in_small_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Free_cholesterol_to_total_lipids_ratio_in_very_large_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Free_cholesterol_to_total_lipids_ratio_in_very_small_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Gallstone_disease+NHGRI-EBI_GWAS_catalog+rs1260326&Gallstone_disease+NHGRI-EBI_GWAS_catalog+rs1260326&Gamma-glutamylthreonine_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Gamma_glutamyl_transferase_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Gamma_glutamyl_transferase_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Gamma_glutamyl_transpeptidase+NHGRI-EBI_GWAS_catalog+rs1260326&Gamma_glutamyl_transpeptidase+NHGRI-EBI_GWAS_catalog+rs1260326&Gestational_diabetes_mellitus+NHGRI-EBI_GWAS_catalog+rs1260326&Gestational_insulin_sensitivity+NHGRI-EBI_GWAS_catalog+rs1260326&Glomerular_filtration_rate+NHGRI-EBI_GWAS_catalog+rs1260326&Glomerular_filtration_rate_creatinine+NHGRI-EBI_GWAS_catalog+rs1260326&Glomerular_filtration_rate_in_non_diabetics__creatinine_+NHGRI-EBI_GWAS_catalog+rs1260326&Glomerular_filtration_rate+NHGRI-EBI_GWAS_catalog+rs1260326&Glucokinase_regulatory_protein_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Glucose_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Glucose_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Glucose-to-mannose_ratio+NHGRI-EBI_GWAS_catalog+rs1260326&Glutamine_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Glutamine_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Glycated_haemoglobin_HbA1c_levels_adjusted_for_BMI+NHGRI-EBI_GWAS_catalog+rs1260326&Glycated_haemoglobin_HbA1c_levels_adjusted_for_BMI+NHGRI-EBI_GWAS_catalog+rs1260326&Glycemic_traits_pleiotropy+NHGRI-EBI_GWAS_catalog+rs1260326&Glycemic_traits__pregnancy_+NHGRI-EBI_GWAS_catalog+rs1260326&Glycemic_traits__pregnancy_+NHGRI-EBI_GWAS_catalog+rs1260326&Glycoprotein_acetyls_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Gout+NHGRI-EBI_GWAS_catalog+rs1260326&Gout_combined_type+NHGRI-EBI_GWAS_catalog+rs1260326&Gout+NHGRI-EBI_GWAS_catalog+rs1260326&Gout_vs_asymptomatic_hyperuricemia+NHGRI-EBI_GWAS_catalog+rs1260326&Granulocyte_count+NHGRI-EBI_GWAS_catalog+rs1260326&Height+NHGRI-EBI_GWAS_catalog+rs1260326&Height+NHGRI-EBI_GWAS_catalog+rs1260326&Height+NHGRI-EBI_GWAS_catalog+rs1260326&Height+NHGRI-EBI_GWAS_catalog+rs1260326&Hematocrit+NHGRI-EBI_GWAS_catalog+rs1260326&Hematocrit+NHGRI-EBI_GWAS_catalog+rs1260326&Hematological_and_biochemical_traits+NHGRI-EBI_GWAS_catalog+rs1260326&High_density_lipoprotein_cholesterol_levels+NHGRI-EBI_GWAS_catalog+rs1260326&High_light_scatter_reticulocyte_count+NHGRI-EBI_GWAS_catalog+rs1260326&High_light_scatter_reticulocyte_percentage_of_red_cells+NHGRI-EBI_GWAS_catalog+rs1260326&High_light_scatter_reticulocyte_percentage_of_red_cells+NHGRI-EBI_GWAS_catalog+rs1260326&Hypertriglyceridemia+NHGRI-EBI_GWAS_catalog+rs1260326&IGF-1_and_IGFBP-3_levels_bivariate_analysis+NHGRI-EBI_GWAS_catalog+rs1260326&Inflammatory_bowel_disease+NHGRI-EBI_GWAS_catalog+rs1260326&Inflammatory_bowel_disease+NHGRI-EBI_GWAS_catalog+rs1260326&Intrahepatic_cholestasis_of_pregnancy+NHGRI-EBI_GWAS_catalog+rs1260326&Iron_status_biomarkers__ferritin_levels_+NHGRI-EBI_GWAS_catalog+rs1260326&Ischemic_stroke_or_factor_VII_levels_pleiotropy+NHGRI-EBI_GWAS_catalog+rs1260326&Isoleucine_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Isoleucine_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Lactate_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Lactate_levels+NHGRI-EBI_GWAS_catalog+rs1260326&LDL_cholesterol_levels+NHGRI-EBI_GWAS_catalog+rs1260326&LDL_cholesterol_levels+NHGRI-EBI_GWAS_catalog+rs1260326&LDL_cholesterol_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Leucine_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Leucine_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Linoleic_acid_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Lipid_metabolism_phenotypes+NHGRI-EBI_GWAS_catalog+rs1260326&Lipid_metabolism_phenotypes+NHGRI-EBI_GWAS_catalog+rs1260326&Lipid_metabolism_phenotypes+NHGRI-EBI_GWAS_catalog+rs1260326&Lipid_traits+NHGRI-EBI_GWAS_catalog+rs1260326&Lipoprotein-associated_phospholipase_A2_activity_and_mass+NHGRI-EBI_GWAS_catalog+rs1260326&Liver_enzyme_levels__alkaline_phosphatase_+NHGRI-EBI_GWAS_catalog+rs1260326&Liver_enzyme_levels__gamma-glutamyl_transferase_+NHGRI-EBI_GWAS_catalog+rs1260326&Liver_fat_content_MRI_proton_density_fat_fraction_measure+NHGRI-EBI_GWAS_catalog+rs1260326&Liver_volume+NHGRI-EBI_GWAS_catalog+rs1260326&Low_density_lipoprotein_cholesterol_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Low_density_lipoprotein_cholesterol_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Low_density_lipoprotein_cholesterol_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Low_testosterone_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Lymphocyte_count+NHGRI-EBI_GWAS_catalog+rs1260326&Lymphocyte_count+NHGRI-EBI_GWAS_catalog+rs1260326&Lymphocyte_count+NHGRI-EBI_GWAS_catalog+rs1260326&lysoPhosphatidylcholine_acyl_C14:0_levels+NHGRI-EBI_GWAS_catalog+rs1260326&lysoPhosphatidylcholine_acyl_C16:0_levels+NHGRI-EBI_GWAS_catalog+rs1260326&lysoPhosphatidylcholine_acyl_C16:0_levels+NHGRI-EBI_GWAS_catalog+rs1260326&lysoPhosphatidylcholine_acyl_C16:1_levels+NHGRI-EBI_GWAS_catalog+rs1260326&lysoPhosphatidylcholine_acyl_C16:1_levels+NHGRI-EBI_GWAS_catalog+rs1260326&lysoPhosphatidylcholine_acyl_C18:0_levels+NHGRI-EBI_GWAS_catalog+rs1260326&lysoPhosphatidylcholine_acyl_C20:3_levels+NHGRI-EBI_GWAS_catalog+rs1260326&lysoPhosphatidylcholine_acyl_C20:3_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Lysophosphatidylethanolamine_16:0_[M-H]1-_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Macular_telangiectasia_type_2+NHGRI-EBI_GWAS_catalog+rs1260326&Macular_telangiectasia_type_2+NHGRI-EBI_GWAS_catalog+rs1260326&Mannonate_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Mannose_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Mannose_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Mannose_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Maximum_habitual_alcohol_consumption_MTAG+NHGRI-EBI_GWAS_catalog+rs1260326&Maximum_habitual_alcohol_consumption+NHGRI-EBI_GWAS_catalog+rs1260326&Maximum_habitual_alcohol_consumption+NHGRI-EBI_GWAS_catalog+rs1260326&Mean_platelet_volume+NHGRI-EBI_GWAS_catalog+rs1260326&Mean_reticulocyte_volume+NHGRI-EBI_GWAS_catalog+rs1260326&Mean_spheric_corpuscular_volume+NHGRI-EBI_GWAS_catalog+rs1260326&Medication_use_drugs_used_in_diabetes+NHGRI-EBI_GWAS_catalog+rs1260326&Medication_use_drugs_used_in_diabetes+NHGRI-EBI_GWAS_catalog+rs1260326&Medication_use_HMG_CoA_reductase_inhibitors+NHGRI-EBI_GWAS_catalog+rs1260326&Medication_use_HMG_CoA_reductase_inhibitors+NHGRI-EBI_GWAS_catalog+rs1260326&Medication_use_HMG_CoA_reductase_inhibitors+NHGRI-EBI_GWAS_catalog+rs1260326&Metabolic_syndrome+NHGRI-EBI_GWAS_catalog+rs1260326&Metabolic_traits+NHGRI-EBI_GWAS_catalog+rs1260326&Metabolite_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Metabolite_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Metabolite_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Metabolite_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Monocyte_percentage_of_white_cells+NHGRI-EBI_GWAS_catalog+rs1260326&Monounsaturated_fatty_acid_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Myeloid_white_cell_count+NHGRI-EBI_GWAS_catalog+rs1260326&Myristoyl-linoleoyl-glycerol_14:0/18:2_[1]_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Neutrophil_count+NHGRI-EBI_GWAS_catalog+rs1260326&Neutrophil_count+NHGRI-EBI_GWAS_catalog+rs1260326&Neutrophil_count+NHGRI-EBI_GWAS_catalog+rs1260326&Neutrophil_percentage_of_white_cells+NHGRI-EBI_GWAS_catalog+rs1260326&Non-albumin_protein_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Nonalcoholic_fatty_liver_disease+NHGRI-EBI_GWAS_catalog+rs1260326&Nonalcoholic_fatty_liver_disease+NHGRI-EBI_GWAS_catalog+rs1260326&Nonalcoholic_fatty_liver_disease+NHGRI-EBI_GWAS_catalog+rs1260326&Nonalcoholic_fatty_liver_disease+NHGRI-EBI_GWAS_catalog+rs1260326&Nonalcoholic_steatohepatitis+NHGRI-EBI_GWAS_catalog+rs1260326&Non-HDL_cholesterol_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Non-HDL_cholesterol_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Non-HDL_cholesterol_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Non-HDL_cholesterol_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Non-oily_fish_consumption+NHGRI-EBI_GWAS_catalog+rs1260326&Oleoyl-arachidonoyl-glycerol_18:1/20:4_[1]_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Omega-3_fatty_acid_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Omega-6_fatty_acid_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Palmitoleoyl-linoleoyl-glycerol_16:1/18:2_[1]_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Percent_glycated_albumin+NHGRI-EBI_GWAS_catalog+rs1260326&Phosphatidylcholine-2O_32:0_[M+H]1+_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Phosphatidylcholine_diacyl_C32:1_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Phosphatidylcholine_diacyl_C32:2_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Phosphatidylcholine_diacyl_C34:1_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Phosphatidylcholine_diacyl_C34:3_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Phosphatidylcholine_diacyl_C34:4_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Phosphatidylcholine_diacyl_C34:4_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Phosphatidylcholine_diacyl_C36:6_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Phosphatidylcholine_diacyl_C38:5_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Phosphatidylcholine_diacyl_C40:5_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Phosphatidylcholine_diacyl_C40:6_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Phosphatidylcholine_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Phosphatidylcholine-O_30:1_[M+H]1+/Phosphatidylcholine-P_30:0_[M+H]1+/Phosphatidylethanolamine-O_33:1_[M+H]1+_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Phosphatidylcholine-O_31:0_[M+H]1+/Phosphatidylethanolamine-O_34:0_[M+H]1+_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Phosphatidylcholine-O_32:1_[M+H]1+/Phosphatidylcholine-P_32:0_[M+H]1+/Phosphatidylethanolamine-O_35:1_[M+H]1+_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Phosphatidylcholine-O_34:1_[M+H]1+/Phosphatidylcholine-P_34:0_[M+H]1+/Phosphatidylethanolamine-O_37:1_[M+H]1+_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Phosphatidylcholine-O_34:2_[M+H]1+/Phosphatidylcholine-P_34:1_[M+H]1+/Phosphatidylethanolamine-O_37:2_[M+H]1+_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Phosphatidylcholine-O_34:3_[M+H]1+/Phosphatidylcholine-P_34:2_[M+H]1+/Phosphatidylethanolamine-P_37:2_[M+H]1+_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Phosphatidylcholine-O_36:2_[M+H]1+/Phosphatidylcholine-P_36:1_[M+H]1+/Phosphatidylethanolamine-P_39:1_[M+H]1+_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Phosphatidylcholine-O_36:3_[M+H]1+/Phosphatidylcholine-P_36:2_[M+H]1+_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Phosphatidylethanolamine_34:1_[M-H]1-_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Phosphoglycerides_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Phospholipid_levels_in_chylomicrons_and_extremely_large_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Phospholipid_levels_in_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Phospholipid_levels_in_large_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Phospholipid_levels_in_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Phospholipid_levels_in_medium_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Phospholipid_levels_in_medium_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Phospholipid_levels_in_medium_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Phospholipid_levels_in_small_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Phospholipid_levels_in_small_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Phospholipid_levels_in_very_large_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Phospholipid_levels_in_very_large_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Phospholipid_levels_in_very_small_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Phospholipid_levels_in_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Phospholipids_to_total_lipids_ratio_in_chylomicrons_and_extremely_large_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Phospholipids_to_total_lipids_ratio_in_large_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Phospholipids_to_total_lipids_ratio_in_large_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Phospholipids_to_total_lipids_ratio_in_large_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Phospholipids_to_total_lipids_ratio_in_medium_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Phospholipids_to_total_lipids_ratio_in_medium_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Phospholipids_to_total_lipids_ratio_in_medium_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Phospholipids_to_total_lipids_ratio_in_small_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Phospholipids_to_total_lipids_ratio_in_small_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Phospholipids_to_total_lipids_ratio_in_very_large_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Phospholipids_to_total_lipids_ratio_in_very_small_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Plasma_ANGPTL8_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Plasma_lactate_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Platelet_Count+NHGRI-EBI_GWAS_catalog+rs1260326&Platelet_Count+NHGRI-EBI_GWAS_catalog+rs1260326&Platelet_Count+NHGRI-EBI_GWAS_catalog+rs1260326&Plateletcrit+NHGRI-EBI_GWAS_catalog+rs1260326&Plateletcrit+NHGRI-EBI_GWAS_catalog+rs1260326&Polyunsaturated_fatty_acid_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Problematic_alcohol_use+NHGRI-EBI_GWAS_catalog+rs1260326&Problematic_alcohol_use_MTAG+NHGRI-EBI_GWAS_catalog+rs1260326&Problematic_alcohol_use_MTAG+NHGRI-EBI_GWAS_catalog+rs1260326&Protein_C_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Pyruvate_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Pyruvate_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Ratio_of_linoleic_acid_to_total_fatty_acids+NHGRI-EBI_GWAS_catalog+rs1260326&Ratio_of_monounsaturated_fatty_acids_to_total_fatty_acids+NHGRI-EBI_GWAS_catalog+rs1260326&Ratio_of_omega-3_fatty_acids_to_total_fatty_acids+NHGRI-EBI_GWAS_catalog+rs1260326&Ratio_of_omega-6_fatty_acids_to_omega-3_fatty_acids+NHGRI-EBI_GWAS_catalog+rs1260326&Ratio_of_omega-6_fatty_acids_to_total_fatty_acids+NHGRI-EBI_GWAS_catalog+rs1260326&Ratio_of_polyunsaturated_fatty_acids_to_monounsaturated_fatty_acids+NHGRI-EBI_GWAS_catalog+rs1260326&Ratio_of_polyunsaturated_fatty_acids_to_total_fatty_acids+NHGRI-EBI_GWAS_catalog+rs1260326&Ratio_of_saturated_fatty_acids_to_total_fatty_acids+NHGRI-EBI_GWAS_catalog+rs1260326&Ratio_of_triglycerides_to_phosphoglycerides+NHGRI-EBI_GWAS_catalog+rs1260326&Red_cell_distribution_width+NHGRI-EBI_GWAS_catalog+rs1260326&Red_cell_distribution_width+NHGRI-EBI_GWAS_catalog+rs1260326&Red_cell_distribution_width+NHGRI-EBI_GWAS_catalog+rs1260326&Refractive_error+NHGRI-EBI_GWAS_catalog+rs1260326&Renal_overload_gout+NHGRI-EBI_GWAS_catalog+rs1260326&Resting_heart_rate+NHGRI-EBI_GWAS_catalog+rs1260326&Resting_heart_rate+NHGRI-EBI_GWAS_catalog+rs1260326&Reticulocyte_count+NHGRI-EBI_GWAS_catalog+rs1260326&Reticulocyte_fraction_of_red_cells+NHGRI-EBI_GWAS_catalog+rs1260326&Saturated_fatty_acid_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Serine_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Serum_25-Hydroxyvitamin_D_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Serum_25-Hydroxyvitamin_D_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Serum_albumin_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Serum_albumin_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Serum_albumin_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Serum_alkaline_phosphatase_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Serum_alkaline_phosphatase_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Serum_alkaline_phosphatase_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Serum_alpha1-antitrypsin_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Serum_creatinine_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Serum_metabolite_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Serum_metabolite_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Serum_metabolite_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Serum_metabolite_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Serum_metabolite_levels_CMS+NHGRI-EBI_GWAS_catalog+rs1260326&Serum_metabolite_levels_CMS+NHGRI-EBI_GWAS_catalog+rs1260326&Serum_metabolite_levels_CMS+NHGRI-EBI_GWAS_catalog+rs1260326&Serum_metabolite_levels_CMS+NHGRI-EBI_GWAS_catalog+rs1260326&Serum_total_protein_level+NHGRI-EBI_GWAS_catalog+rs1260326&Serum_total_protein_level+NHGRI-EBI_GWAS_catalog+rs1260326&Serum_total_protein_level+NHGRI-EBI_GWAS_catalog+rs1260326&Serum_uric_acid_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Serum_uric_acid_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Serum_uric_acid_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Sex_hormone-binding_globulin_levels_adjusted_for_BMI+NHGRI-EBI_GWAS_catalog+rs1260326&Sex_hormone-binding_globulin_levels_adjusted_for_BMI+NHGRI-EBI_GWAS_catalog+rs1260326&Sex_hormone-binding_globulin_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Sex_hormone-binding_globulin_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Sex_hormone-binding_globulin_levels_in_postmenopausal_women+NHGRI-EBI_GWAS_catalog+rs1260326&Sex_hormone-binding_globulin_levels_in_premenopausal_women+NHGRI-EBI_GWAS_catalog+rs1260326&Sphingomyelin33:1_[M-CH3]1-_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Sphingomyelin_33:1_[M+H]1+_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Sphingomyelin_34:0_[M+H]1+_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Sphingomyelin_34:1_[M+H]1+_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Sphingomyelin_34:2_[M+H]1+_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Sphingomyelin_35:1_[M+H]1+_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Sphingomyelin37:1_[M+H]1+_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Sphingomyelin_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Sum_basophil_neutrophil_counts+NHGRI-EBI_GWAS_catalog+rs1260326&Sum_neutrophil_eosinophil_counts+NHGRI-EBI_GWAS_catalog+rs1260326&Testosterone_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Testosterone_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Testosterone_levels_in_postmenopausal_women+NHGRI-EBI_GWAS_catalog+rs1260326&Threonine_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Total_cholesterol_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Total_cholesterol_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Total_cholesterol_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Total_cholesterol_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Total_cholesterol_minus_HDL-C_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Total_cholines_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Total_concentration_of_branched-chain_amino_acids_leucine_+_isoleucine_+_valine+NHGRI-EBI_GWAS_catalog+rs1260326&Total_concentration_of_lipoprotein_particles+NHGRI-EBI_GWAS_catalog+rs1260326&Total_esterified_cholesterol_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Total_fatty_acid_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Total_free_cholesterol_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Total_lipid_levels_in_chylomicrons_and_extremely_large_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Total_lipid_levels_in_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Total_lipid_levels_in_large_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Total_lipid_levels_in_large_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Total_lipid_levels_in_large_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Total_lipid_levels_in_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Total_lipid_levels_in_lipoprotein_particles+NHGRI-EBI_GWAS_catalog+rs1260326&Total_lipid_levels_in_medium_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Total_lipid_levels_in_medium_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Total_lipid_levels_in_medium_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Total_lipid_levels_in_small_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Total_lipid_levels_in_small_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Total_lipid_levels_in_small_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Total_lipid_levels_in_very_large_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Total_lipid_levels_in_very_large_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Total_lipid_levels_in_very_small_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Total_lipid_levels_in_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Total_omega-3_fatty_acid_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Total_omega-6_fatty_acid_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Total_phospholipid_levels_in_lipoprotein_particles+NHGRI-EBI_GWAS_catalog+rs1260326&Total_testosterone_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Total_testosterone_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Total_testosterone_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Total_triglycerides_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Total_triglycerides_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Triacylglyceride_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Triacylglycerol_48:2_[NL-16:0]_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Triacylglycerol_50:2_[M+NH4]1+_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Triacylglycerol_50:3_[M+NH4]1+_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Triacylglycerol_50:4_[M+NH4]1+_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Triacylglycerol_50:5_[M+NH4]1+_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Triacylglycerol_52:6_[M+NH4]1+_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Triglyceride_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Triglyceride_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Triglyceride_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Triglyceride_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Triglyceride_levels_in_chylomicrons_and_extremely_large_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Triglyceride_levels_in_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Triglyceride_levels_in_IDL+NHGRI-EBI_GWAS_catalog+rs1260326&Triglyceride_levels_in_large_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Triglyceride_levels_in_large_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Triglyceride_levels_in_large_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Triglyceride_levels_in_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Triglyceride_levels_in_medium_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Triglyceride_levels_in_medium_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Triglyceride_levels_in_medium_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Triglyceride_levels_in_small_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Triglyceride_levels_in_small_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Triglyceride_levels_in_small_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Triglyceride_levels_in_very_large_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Triglyceride_levels_in_very_large_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Triglyceride_levels_in_very_small_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Triglyceride_levels_in_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Triglyceride_levels_x_long_total_sleep_time_interaction_2df_test+NHGRI-EBI_GWAS_catalog+rs1260326&Triglyceride_levels_x_short_total_sleep_time_interaction_2df_test+NHGRI-EBI_GWAS_catalog+rs1260326&Triglycerides+NHGRI-EBI_GWAS_catalog+rs1260326&Triglycerides+NHGRI-EBI_GWAS_catalog+rs1260326&Triglycerides+NHGRI-EBI_GWAS_catalog+rs1260326&Triglycerides+NHGRI-EBI_GWAS_catalog+rs1260326&Triglycerides_to_total_lipids_ratio_in_IDL+NHGRI-EBI_GWAS_catalog+rs1260326&Triglycerides_to_total_lipids_ratio_in_large_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Triglycerides_to_total_lipids_ratio_in_large_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Triglycerides_to_total_lipids_ratio_in_medium_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Triglycerides_to_total_lipids_ratio_in_medium_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Triglycerides_to_total_lipids_ratio_in_medium_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Triglycerides_to_total_lipids_ratio_in_small_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Triglycerides_to_total_lipids_ratio_in_small_LDL+NHGRI-EBI_GWAS_catalog+rs1260326&Triglycerides_to_total_lipids_ratio_in_small_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Triglycerides_to_total_lipids_ratio_in_very_large_HDL+NHGRI-EBI_GWAS_catalog+rs1260326&Triglycerides_to_total_lipids_ratio_in_very_large_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Triglycerides_to_total_lipids_ratio_in_very_small_VLDL+NHGRI-EBI_GWAS_catalog+rs1260326&Triglycerides_x_physical_activity_interaction__2df_test_+NHGRI-EBI_GWAS_catalog+rs1260326&Two-hour_glucose+NHGRI-EBI_GWAS_catalog+rs1260326&Two-hour_glucose_challenge+NHGRI-EBI_GWAS_catalog+rs1260326&Type_2_diabetes__adjusted_for_BMI_+NHGRI-EBI_GWAS_catalog+rs1260326&Type_2_diabetes__adjusted_for_BMI_+NHGRI-EBI_GWAS_catalog+rs1260326&Type_2_diabetes+NHGRI-EBI_GWAS_catalog+rs1260326&Type_2_diabetes+NHGRI-EBI_GWAS_catalog+rs1260326&Type_2_diabetes+NHGRI-EBI_GWAS_catalog+rs1260326&Type_2_diabetes+NHGRI-EBI_GWAS_catalog+rs1260326&Urate_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Urate_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Urate_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Urate_levels_in_overweight_individuals+NHGRI-EBI_GWAS_catalog+rs1260326&Urate_levels_in_overweight_individuals+NHGRI-EBI_GWAS_catalog+rs1260326&Urinary_sodium_excretion+NHGRI-EBI_GWAS_catalog+rs1260326&Urinary_sodium_to_creatinine_ratio+NHGRI-EBI_GWAS_catalog+rs1260326&Urolithiasis+NHGRI-EBI_GWAS_catalog+rs1260326&Urolithiasis+NHGRI-EBI_GWAS_catalog+rs1260326&Urolithiasis+NHGRI-EBI_GWAS_catalog+rs1260326&Valine_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Venous_thromboembolism_or_factor_VII_levels_pleiotropy+NHGRI-EBI_GWAS_catalog+rs1260326&VLDL_cholesterol_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Waist_circumference_and_related_phenotypes+NHGRI-EBI_GWAS_catalog+rs1260326&Weight+NHGRI-EBI_GWAS_catalog+rs1260326&White_blood_cell_count+NHGRI-EBI_GWAS_catalog+rs1260326&White_blood_cell_count+NHGRI-EBI_GWAS_catalog+rs1260326&White_blood_cell_count+NHGRI-EBI_GWAS_catalog+rs1260326&White_blood_cell_count+NHGRI-EBI_GWAS_catalog+rs1260326&X-19438_levels+NHGRI-EBI_GWAS_catalog+rs1260326&X-24295_levels+NHGRI-EBI_GWAS_catalog+rs1260326&Cholesterol__total+Teslovich+rs1260326\n",
      "========================================\n",
      "8\n",
      "CLIN_SIG\n",
      "184\n",
      "[nan 'benign' 'pathogenic' 'not_provided' 'likely_benign']\n",
      "risk_factor&benign&likely_benign&association&pathogenic&other&likely_pathogenic&conflicting_interpretations_of_pathogenicity\n",
      "========================================\n",
      "9\n",
      "Allele\n",
      "2350\n",
      "['T' 'A' 'C' 'G' 'ACCCTGGTCCCCCTGGTCCCTTTGGCCCTGCACCTGGCTGG']\n",
      "TCTGCTCAGCTCCTGTCCTCAGCGTCTCCTCCCAACCCCTGCCCTGCGCTGTGTGGGGCCGCACCTGTGTCCTCACCTCGCCTCCCAGCACCTCTGCTCAACTCCTGTCCTCAGCGTCTCCTCCCAGCCCCTGCCCTGCACTGTATGGGGCCGAACCTGGGTCCTTGCCTCCCAGCACC\n",
      "========================================\n",
      "10\n",
      "HGVSc\n",
      "207584\n",
      "[nan 'ENST00000442987.3:n.2105G>A' 'ENST00000506640.3:n.2100G>T'\n",
      " 'ENST00000412115.2:n.301G>A' 'ENST00000415481.1:n.251G>A']\n",
      "ENST00000457478.1:n.439-68507_439-68506insGGTGCTGGGAGGCAAGGACCCAGGTTCGGCCCCATACAGTGCAGGGCAGGGGCTGGGAGGAGACGCTGAGGACAGGAGTTGAGCAGAGGTGCTGGGAGGCGAGGTGAGGACACAGGTGCGGCCCCACACAGCGCAGGGCAGGGGTTGGGAGGAGACGCTGAGGACAGGAGCTGAGCAGA\n",
      "========================================\n",
      "11\n",
      "UNIPROT_ISOFORM\n",
      "9307\n",
      "[nan 'Q5SV97-3' 'O00468-6' 'Q96HA4-4' 'Q6ZVT0-1']\n",
      "A0A0K2S4Q6-2\n",
      "========================================\n",
      "12\n",
      "CCDS\n",
      "15897\n",
      "[nan 'CCDS3.1' 'CCDS30550.1' 'CCDS90836.1' 'CCDS30551.1']\n",
      "CCDS30550.1\n",
      "========================================\n",
      "13\n",
      "ENSP\n",
      "18398\n",
      "[nan 'ENSP00000317992' 'ENSP00000343930' 'ENSP00000343864'\n",
      " 'ENSP00000368678']\n",
      "ENSP00000317992\n",
      "========================================\n",
      "14\n",
      "CDS_position\n",
      "6519\n",
      "[nan '1182' '918' '898' '1863']\n",
      "11857-11858\n",
      "========================================\n",
      "15\n",
      "SOMATIC\n",
      "19\n",
      "[nan '0&1' '0&0&1' '0&1&1' '0&1&1&1']\n",
      "1&1&1&1&1&1&1&1&1\n",
      "========================================\n",
      "16\n",
      "DOMAINS\n",
      "20172\n",
      "[nan\n",
      " 'Pfam:PF03715&Superfamily:SSF48371&PANTHER:PTHR12687&AlphaFold_DB_import:AF-Q9Y3T9-F1'\n",
      " 'Superfamily:SSF48371&PANTHER:PTHR12687&AlphaFold_DB_import:AF-Q9Y3T9-F1'\n",
      " 'Gene3D:2.120.10.80&AlphaFold_DB_import:AF-Q6TDP4-F1&SMART:SM00612&Superfamily:SSF50965&PIRSF:PIRSF037037&PDB-ENSP_mappings:6hrl.A&PDB-ENSP_mappings:6hrl.B'\n",
      " 'AlphaFold_DB_import:AF-Q5SV97-F1&PANTHER:PTHR47282']\n",
      "PDB-ENSP_mappings:12ca.A&PDB-ENSP_mappings:1a42.A&PDB-ENSP_mappings:1am6.A&PDB-ENSP_mappings:1avn.A&PDB-ENSP_mappings:1bcd.A&PDB-ENSP_mappings:1bic.A&PDB-ENSP_mappings:1bn1.A&PDB-ENSP_mappings:1bn3.A&PDB-ENSP_mappings:1bn4.A&PDB-ENSP_mappings:1bnm.A&PDB-ENSP_mappings:1bnn.A&PDB-ENSP_mappings:1bnq.A&PDB-ENSP_mappings:1bnt.A&PDB-ENSP_mappings:1bnu.A&PDB-ENSP_mappings:1bnv.A&PDB-ENSP_mappings:1bnw.A&PDB-ENSP_mappings:1bv3.A&PDB-ENSP_mappings:1ca2.A&PDB-ENSP_mappings:1ca3.A&PDB-ENSP_mappings:1cah.A&PDB-ENSP_mappings:1cai.A&PDB-ENSP_mappings:1caj.A&PDB-ENSP_mappings:1cak.A&PDB-ENSP_mappings:1cal.A&PDB-ENSP_mappings:1cam.A&PDB-ENSP_mappings:1can.A&PDB-ENSP_mappings:1cao.A&PDB-ENSP_mappings:1cay.A&PDB-ENSP_mappings:1caz.A&PDB-ENSP_mappings:1ccs.A&PDB-ENSP_mappings:1cct.A&PDB-ENSP_mappings:1ccu.A&PDB-ENSP_mappings:1cil.A&PDB-ENSP_mappings:1cim.A&PDB-ENSP_mappings:1cin.A&PDB-ENSP_mappings:1cnb.A&PDB-ENSP_mappings:1cnc.A&PDB-ENSP_mappings:1cng.A&PDB-ENSP_mappings:1cnh.A&PDB-ENSP_mappings:1cni.A&PDB-ENSP_mappings:1cnj.A&PDB-ENSP_mappings:1cnk.A&PDB-ENSP_mappings:1cnw.A&PDB-ENSP_mappings:1cnx.A&PDB-ENSP_mappings:1cny.A&PDB-ENSP_mappings:1cra.A&PDB-ENSP_mappings:1cva.A&PDB-ENSP_mappings:1cvb.A&PDB-ENSP_mappings:1cvc.A&PDB-ENSP_mappings:1cvd.A&PDB-ENSP_mappings:1cve.A&PDB-ENSP_mappings:1cvf.A&PDB-ENSP_mappings:1cvh.A&PDB-ENSP_mappings:1dca.A&PDB-ENSP_mappings:1dcb.A&PDB-ENSP_mappings:1eou.A&PDB-ENSP_mappings:1f2w.A&PDB-ENSP_mappings:1fql.A&PDB-ENSP_mappings:1fqm.A&PDB-ENSP_mappings:1fqn.A&PDB-ENSP_mappings:1fqr.A&PDB-ENSP_mappings:1fr4.A&PDB-ENSP_mappings:1fr7.A&PDB-ENSP_mappings:1fr7.B&PDB-ENSP_mappings:1fsn.A&PDB-ENSP_mappings:1fsn.B&PDB-ENSP_mappings:1fsq.A&PDB-ENSP_mappings:1fsq.B&PDB-ENSP_mappings:1fsr.A&PDB-ENSP_mappings:1fsr.B&PDB-ENSP_mappings:1g0e.A&PDB-ENSP_mappings:1g0f.A&PDB-ENSP_mappings:1g1d.A&PDB-ENSP_mappings:1g3z.A&PDB-ENSP_mappings:1g45.A&PDB-ENSP_mappings:1g46.A&PDB-ENSP_mappings:1g48.A&PDB-ENSP_mappings:1g4j.A&PDB-ENSP_mappings:1g4o.A&PDB-ENSP_mappings:1g52.A&PDB-ENSP_mappings:1g53.A&PDB-ENSP_mappings:1g54.A&PDB-ENSP_mappings:1h4n.A&PDB-ENSP_mappings:1h9n.A&PDB-ENSP_mappings:1h9q.A&PDB-ENSP_mappings:1hca.A&PDB-ENSP_mappings:1hea.A&PDB-ENSP_mappings:1heb.A&PDB-ENSP_mappings:1hec.A&PDB-ENSP_mappings:1hed.A&PDB-ENSP_mappings:1hva.A&PDB-ENSP_mappings:1i8z.A&PDB-ENSP_mappings:1i90.A&PDB-ENSP_mappings:1i91.A&PDB-ENSP_mappings:1i9l.A&PDB-ENSP_mappings:1i9m.A&PDB-ENSP_mappings:1i9n.A&PDB-ENSP_mappings:1i9o.A&PDB-ENSP_mappings:1i9p.A&PDB-ENSP_mappings:1i9q.A&PDB-ENSP_mappings:1if4.A&PDB-ENSP_mappings:1if5.A&PDB-ENSP_mappings:1if6.A&PDB-ENSP_mappings:1if7.A&PDB-ENSP_mappings:1if8.A&PDB-ENSP_mappings:1if9.A&PDB-ENSP_mappings:1kwq.A&PDB-ENSP_mappings:1kwr.A&PDB-ENSP_mappings:1lg5.A&PDB-ENSP_mappings:1lg6.A&PDB-ENSP_mappings:1lgd.A&PDB-ENSP_mappings:1lug.A&PDB-ENSP_mappings:1lzv.A&PDB-ENSP_mappings:1moo.A&PDB-ENSP_mappings:1mua.A&PDB-ENSP_mappings:1okl.A&PDB-ENSP_mappings:1okm.A&PDB-ENSP_mappings:1okn.A&PDB-ENSP_mappings:1oq5.A&PDB-ENSP_mappings:1ray.A&PDB-ENSP_mappings:1raz.A&PDB-ENSP_mappings:1rza.A&PDB-ENSP_mappings:1rzb.A&PDB-ENSP_mappings:1rzc.A&PDB-ENSP_mappings:1rzd.A&PDB-ENSP_mappings:1rze.A&PDB-ENSP_mappings:1t9n.A&PDB-ENSP_mappings:1tb0.X&PDB-ENSP_mappings:1tbt.X&PDB-ENSP_mappings:1te3.X&PDB-ENSP_mappings:1teq.X&PDB-ENSP_mappings:1teu.X&PDB-ENSP_mappings:1tg3.A&PDB-ENSP_mappings:1tg9.A&PDB-ENSP_mappings:1th9.A&PDB-ENSP_mappings:1thk.A&PDB-ENSP_mappings:1ttm.A&PDB-ENSP_mappings:1uga.A&PDB-ENSP_mappings:1ugb.A&PDB-ENSP_mappings:1ugc.A&PDB-ENSP_mappings:1ugd.A&PDB-ENSP_mappings:1uge.A&PDB-ENSP_mappings:1ugf.A&PDB-ENSP_mappings:1ugg.A&PDB-ENSP_mappings:1xeg.A&PDB-ENSP_mappings:1xev.A&PDB-ENSP_mappings:1xev.B&PDB-ENSP_mappings:1xev.C&PDB-ENSP_mappings:1xev.D&PDB-ENSP_mappings:1xpz.A&PDB-ENSP_mappings:1xq0.A&PDB-ENSP_mappings:1yda.A&PDB-ENSP_mappings:1ydb.A&PDB-ENSP_mappings:1ydc.A&PDB-ENSP_mappings:1ydd.A&PDB-ENSP_mappings:1yo0.A&PDB-ENSP_mappings:1yo1.A&PDB-ENSP_mappings:1yo2.A&PDB-ENSP_mappings:1z9y.A&PDB-ENSP_mappings:1ze8.A&PDB-ENSP_mappings:1zfk.A&PDB-ENSP_mappings:1zfq.A&PDB-ENSP_mappings:1zge.A&PDB-ENSP_mappings:1zgf.A&PDB-ENSP_mappings:1zh9.A&PDB-ENSP_mappings:1zsa.A&PDB-ENSP_mappings:1zsb.A&PDB-ENSP_mappings:1zsc.A&PDB-ENSP_mappings:2abe.A&PDB-ENSP_mappings:2aw1.A&PDB-ENSP_mappings:2ax2.A&PDB-ENSP_mappings:2ca2.A&PDB-ENSP_mappings:2cba.A&PDB-ENSP_mappings:2cbb.A&PDB-ENSP_mappings:2cbc.A&PDB-ENSP_mappings:2cbd.A&PDB-ENSP_mappings:2cbe.A&PDB-ENSP_mappings:2eu2.A&PDB-ENSP_mappings:2eu3.A&PDB-ENSP_mappings:2ez7.A&PDB-ENSP_mappings:2f14.A&PDB-ENSP_mappings:2fmg.A&PDB-ENSP_mappings:2fmz.A&PDB-ENSP_mappings:2fnk.A&PDB-ENSP_mappings:2fnm.A&PDB-ENSP_mappings:2fnn.A&PDB-ENSP_mappings:2foq.A&PDB-ENSP_mappings:2fos.A&PDB-ENSP_mappings:2fou.A&PDB-ENSP_mappings:2fov.A&PDB-ENSP_mappings:2gd8.A&PDB-ENSP_mappings:2geh.A&PDB-ENSP_mappings:2h15.A&PDB-ENSP_mappings:2h4n.A&PDB-ENSP_mappings:2hd6.A&PDB-ENSP_mappings:2hkk.A&PDB-ENSP_mappings:2hl4.A&PDB-ENSP_mappings:2hnc.A&PDB-ENSP_mappings:2hoc.A&PDB-ENSP_mappings:2ili.A&PDB-ENSP_mappings:2nng.A&PDB-ENSP_mappings:2nno.A&PDB-ENSP_mappings:2nns.A&PDB-ENSP_mappings:2nnv.A&PDB-ENSP_mappings:2nwo.A&PDB-ENSP_mappings:2nwp.A&PDB-ENSP_mappings:2nwy.A&PDB-ENSP_mappings:2nwz.A&PDB-ENSP_mappings:2nxr.A&PDB-ENSP_mappings:2nxs.A&PDB-ENSP_mappings:2nxt.A&PDB-ENSP_mappings:2o4z.A&PDB-ENSP_mappings:2osf.A&PDB-ENSP_mappings:2osm.A&PDB-ENSP_mappings:2pou.A&PDB-ENSP_mappings:2pov.A&PDB-ENSP_mappings:2pow.A&PDB-ENSP_mappings:2q1b.A&PDB-ENSP_mappings:2q1q.A&PDB-ENSP_mappings:2q38.A&PDB-ENSP_mappings:2qo8.A&PDB-ENSP_mappings:2qoa.A&PDB-ENSP_mappings:2qp6.A&PDB-ENSP_mappings:2vva.X&PDB-ENSP_mappings:2vvb.X&PDB-ENSP_mappings:2wd2.A&PDB-ENSP_mappings:2wd3.A&PDB-ENSP_mappings:2weg.A&PDB-ENSP_mappings:2weh.A&PDB-ENSP_mappings:2wej.A&PDB-ENSP_mappings:2weo.A&PDB-ENSP_mappings:2x7s.A&PDB-ENSP_mappings:2x7t.A&PDB-ENSP_mappings:2x7u.A&PDB-ENSP_mappings:3b4f.A&PDB-ENSP_mappings:3bet.A&PDB-ENSP_mappings:3bl0.A&PDB-ENSP_mappings:3bl1.A&PDB-ENSP_mappings:3c7p.A&PDB-ENSP_mappings:3ca2.A&PDB-ENSP_mappings:3caj.A&PDB-ENSP_mappings:3cyu.A&PDB-ENSP_mappings:3d8w.A&PDB-ENSP_mappings:3d92.A&PDB-ENSP_mappings:3d93.A&PDB-ENSP_mappings:3d9z.A&PDB-ENSP_mappings:3daz.A&PDB-ENSP_mappings:3dbu.A&PDB-ENSP_mappings:3dc3.A&PDB-ENSP_mappings:3dc9.A&PDB-ENSP_mappings:3dcc.A&PDB-ENSP_mappings:3dcs.A&PDB-ENSP_mappings:3dcw.A&PDB-ENSP_mappings:3dd0.A&PDB-ENSP_mappings:3dd8.A&PDB-ENSP_mappings:3dv7.A&PDB-ENSP_mappings:3dvb.A&PDB-ENSP_mappings:3dvc.A&PDB-ENSP_mappings:3dvd.A&PDB-ENSP_mappings:3efi.A&PDB-ENSP_mappings:3eft.A&PDB-ENSP_mappings:3f4x.A&PDB-ENSP_mappings:3f8e.A&PDB-ENSP_mappings:3ffp.X&PDB-ENSP_mappings:3gz0.A&PDB-ENSP_mappings:3hfp.A&PDB-ENSP_mappings:3hkn.A&PDB-ENSP_mappings:3hkq.A&PDB-ENSP_mappings:3hkt.A&PDB-ENSP_mappings:3hku.A&PDB-ENSP_mappings:3hlj.A&PDB-ENSP_mappings:3hs4.A&PDB-ENSP_mappings:3ibi.A&PDB-ENSP_mappings:3ibl.A&PDB-ENSP_mappings:3ibn.A&PDB-ENSP_mappings:3ibu.A&PDB-ENSP_mappings:3ieo.A&PDB-ENSP_mappings:3igp.A&PDB-ENSP_mappings:3k2f.A&PDB-ENSP_mappings:3k34.A&PDB-ENSP_mappings:3k7k.A&PDB-ENSP_mappings:3kig.A&PDB-ENSP_mappings:3kkx.A&PDB-ENSP_mappings:3kne.A&PDB-ENSP_mappings:3koi.A&PDB-ENSP_mappings:3kok.A&PDB-ENSP_mappings:3kon.A&PDB-ENSP_mappings:3ks3.A&PDB-ENSP_mappings:3kwa.A&PDB-ENSP_mappings:3l14.A&PDB-ENSP_mappings:3m04.A&PDB-ENSP_mappings:3m14.A&PDB-ENSP_mappings:3m1j.A&PDB-ENSP_mappings:3m1k.A&PDB-ENSP_mappings:3m1q.A&PDB-ENSP_mappings:3m1w.A&PDB-ENSP_mappings:3m2n.A&PDB-ENSP_mappings:3m2x.A&PDB-ENSP_mappings:3m2y.A&PDB-ENSP_mappings:3m2z.A&PDB-ENSP_mappings:3m3x.A&PDB-ENSP_mappings:3m40.A&PDB-ENSP_mappings:3m5e.A&PDB-ENSP_mappings:3m5s.A&PDB-ENSP_mappings:3m5t.A&PDB-ENSP_mappings:3m67.A&PDB-ENSP_mappings:3m96.A&PDB-ENSP_mappings:3m98.A&PDB-ENSP_mappings:3mhc.A&PDB-ENSP_mappings:3mhi.A&PDB-ENSP_mappings:3mhl.A&PDB-ENSP_mappings:3mhm.A&PDB-ENSP_mappings:3mho.A&PDB-ENSP_mappings:3ml2.A&PDB-ENSP_mappings:3mmf.A&PDB-ENSP_mappings:3mna.A&PDB-ENSP_mappings:3mnh.A&PDB-ENSP_mappings:3mni.A&PDB-ENSP_mappings:3mnj.A&PDB-ENSP_mappings:3mnk.A&PDB-ENSP_mappings:3mnu.A&PDB-ENSP_mappings:3mwo.A&PDB-ENSP_mappings:3mwo.B&PDB-ENSP_mappings:3myq.A&PDB-ENSP_mappings:3mzc.A&PDB-ENSP_mappings:3n0n.A&PDB-ENSP_mappings:3n2p.A&PDB-ENSP_mappings:3n3j.A&PDB-ENSP_mappings:3n4b.A&PDB-ENSP_mappings:3nb5.A&PDB-ENSP_mappings:3ni5.A&PDB-ENSP_mappings:3nj9.A&PDB-ENSP_mappings:3oik.A&PDB-ENSP_mappings:3oil.A&PDB-ENSP_mappings:3oim.A&PDB-ENSP_mappings:3oku.A&PDB-ENSP_mappings:3okv.A&PDB-ENSP_mappings:3oy0.A&PDB-ENSP_mappings:3oyq.A&PDB-ENSP_mappings:3oys.A&PDB-ENSP_mappings:3p3h.A&PDB-ENSP_mappings:3p3j.A&PDB-ENSP_mappings:3p44.A&PDB-ENSP_mappings:3p4v.A&PDB-ENSP_mappings:3p55.A&PDB-ENSP_mappings:3p58.A&PDB-ENSP_mappings:3p5a.A&PDB-ENSP_mappings:3p5l.A&PDB-ENSP_mappings:3pjj.A&PDB-ENSP_mappings:3po6.A&PDB-ENSP_mappings:3pyk.A&PDB-ENSP_mappings:3qyk.A&PDB-ENSP_mappings:3r16.A&PDB-ENSP_mappings:3r17.B&PDB-ENSP_mappings:3rg3.A&PDB-ENSP_mappings:3rg4.A&PDB-ENSP_mappings:3rge.A&PDB-ENSP_mappings:3rj7.A&PDB-ENSP_mappings:3rld.A&PDB-ENSP_mappings:3ryj.B&PDB-ENSP_mappings:3ryv.B&PDB-ENSP_mappings:3ryx.B&PDB-ENSP_mappings:3ryy.A&PDB-ENSP_mappings:3ryz.A&PDB-ENSP_mappings:3rz0.B&PDB-ENSP_mappings:3rz1.B&PDB-ENSP_mappings:3rz5.A&PDB-ENSP_mappings:3rz7.A&PDB-ENSP_mappings:3rz8.A&PDB-ENSP_mappings:3s71.B&PDB-ENSP_mappings:3s72.B&PDB-ENSP_mappings:3s73.B&PDB-ENSP_mappings:3s74.B&PDB-ENSP_mappings:3s75.B&PDB-ENSP_mappings:3s76.A&PDB-ENSP_mappings:3s77.B&PDB-ENSP_mappings:3s78.B&PDB-ENSP_mappings:3s8x.A&PDB-ENSP_mappings:3s9t.A&PDB-ENSP_mappings:3sap.A&PDB-ENSP_mappings:3sax.A&PDB-ENSP_mappings:3sbh.A&PDB-ENSP_mappings:3sbi.A&PDB-ENSP_mappings:3t5u.A&PDB-ENSP_mappings:3t5z.A&PDB-ENSP_mappings:3t82.A&PDB-ENSP_mappings:3t83.A&PDB-ENSP_mappings:3t84.A&PDB-ENSP_mappings:3t85.A&PDB-ENSP_mappings:3tmj.A&PDB-ENSP_mappings:3tvn.X&PDB-ENSP_mappings:3tvo.X&PDB-ENSP_mappings:3u3a.X&PDB-ENSP_mappings:3u45.X&PDB-ENSP_mappings:3u47.A&PDB-ENSP_mappings:3u7c.A&PDB-ENSP_mappings:3v2j.A&PDB-ENSP_mappings:3v2m.A&PDB-ENSP_mappings:3v3f.A&PDB-ENSP_mappings:3v3g.B&PDB-ENSP_mappings:3v3h.B&PDB-ENSP_mappings:3v3i.B&PDB-ENSP_mappings:3v3j.A&PDB-ENSP_mappings:3v5g.A&PDB-ENSP_mappings:3v7x.A&PDB-ENSP_mappings:3vbd.A&PDB-ENSP_mappings:3zp9.A&PDB-ENSP_mappings:4bcw.A&PDB-ENSP_mappings:4bf1.A&PDB-ENSP_mappings:4bf6.A&PDB-ENSP_mappings:4ca2.A&PDB-ENSP_mappings:4cac.A&PDB-ENSP_mappings:4cq0.A&PDB-ENSP_mappings:4dz7.A&PDB-ENSP_mappings:4dz9.A&PDB-ENSP_mappings:4e3d.A&PDB-ENSP_mappings:4e3f.A&PDB-ENSP_mappings:4e3g.A&PDB-ENSP_mappings:4e3h.A&PDB-ENSP_mappings:4e49.A&PDB-ENSP_mappings:4e4a.A&PDB-ENSP_mappings:4e5q.A&PDB-ENSP_mappings:4fik.A&PDB-ENSP_mappings:4fl7.A&PDB-ENSP_mappings:4fpt.A&PDB-ENSP_mappings:4frc.A&PDB-ENSP_mappings:4fu5.A&PDB-ENSP_mappings:4fvn.A&PDB-ENSP_mappings:4fvo.A&PDB-ENSP_mappings:4g0c.A&PDB-ENSP_mappings:4gl1.X&PDB-ENSP_mappings:4hba.A&PDB-ENSP_mappings:4hew.A&PDB-ENSP_mappings:4hey.A&PDB-ENSP_mappings:4hez.A&PDB-ENSP_mappings:4hf3.A&PDB-ENSP_mappings:4ht0.A&PDB-ENSP_mappings:4idr.X&PDB-ENSP_mappings:4ilx.A&PDB-ENSP_mappings:4ito.A&PDB-ENSP_mappings:4itp.A&PDB-ENSP_mappings:4iwz.A&PDB-ENSP_mappings:4js6.A&PDB-ENSP_mappings:4jsa.A&PDB-ENSP_mappings:4jss.A&PDB-ENSP_mappings:4jsw.A&PDB-ENSP_mappings:4jsz.A&PDB-ENSP_mappings:4k0s.A&PDB-ENSP_mappings:4k0t.A&PDB-ENSP_mappings:4k0z.A&PDB-ENSP_mappings:4k13.A&PDB-ENSP_mappings:4k1q.A&PDB-ENSP_mappings:4kap.A&PDB-ENSP_mappings:4kni.A&PDB-ENSP_mappings:4knj.A&PDB-ENSP_mappings:4kuv.A&PDB-ENSP_mappings:4kuw.A&PDB-ENSP_mappings:4kuy.A&PDB-ENSP_mappings:4kv0.A&PDB-ENSP_mappings:4l5u.A&PDB-ENSP_mappings:4l5v.A&PDB-ENSP_mappings:4l5w.A&PDB-ENSP_mappings:4lhi.A&PDB-ENSP_mappings:4lp6.A&PDB-ENSP_mappings:4lp6.B&PDB-ENSP_mappings:4m2r.A&PDB-ENSP_mappings:4m2u.A&PDB-ENSP_mappings:4m2v.A&PDB-ENSP_mappings:4m2w.A&PDB-ENSP_mappings:4mdg.A&PDB-ENSP_mappings:4mdl.A&PDB-ENSP_mappings:4mdm.A&PDB-ENSP_mappings:4mlt.A&PDB-ENSP_mappings:4mlx.A&PDB-ENSP_mappings:4mo8.A&PDB-ENSP_mappings:4mty.A&PDB-ENSP_mappings:4n0x.B&PDB-ENSP_mappings:4n16.A&PDB-ENSP_mappings:4pq7.A&PDB-ENSP_mappings:4pxx.A&PDB-ENSP_mappings:4pyx.A&PDB-ENSP_mappings:4pyy.A&PDB-ENSP_mappings:4pzh.A&PDB-ENSP_mappings:4q06.A&PDB-ENSP_mappings:4q07.A&PDB-ENSP_mappings:4q08.A&PDB-ENSP_mappings:4q09.A&PDB-ENSP_mappings:4q49.A&PDB-ENSP_mappings:4q6d.A&PDB-ENSP_mappings:4q6e.A&PDB-ENSP_mappings:4q78.A&PDB-ENSP_mappings:4q7p.A&PDB-ENSP_mappings:4q7s.A&PDB-ENSP_mappings:4q7v.A&PDB-ENSP_mappings:4q7w.A&PDB-ENSP_mappings:4q81.A&PDB-ENSP_mappings:4q83.A&PDB-ENSP_mappings:4q87.A&PDB-ENSP_mappings:4q8x.A&PDB-ENSP_mappings:4q8y.A&PDB-ENSP_mappings:4q8z.A&PDB-ENSP_mappings:4q90.A&PDB-ENSP_mappings:4q99.A&PDB-ENSP_mappings:4q9y.A&PDB-ENSP_mappings:4qef.A&PDB-ENSP_mappings:4qiy.A&PDB-ENSP_mappings:4qiy.B&PDB-ENSP_mappings:4qiy.C&PDB-ENSP_mappings:4qiy.D&PDB-ENSP_mappings:4qjm.A&PDB-ENSP_mappings:4qk1.A&PDB-ENSP_mappings:4qk2.A&PDB-ENSP_mappings:4qk3.A&PDB-ENSP_mappings:4qsa.A&PDB-ENSP_mappings:4qsb.A&PDB-ENSP_mappings:4qsi.A&PDB-ENSP_mappings:4qtl.A&PDB-ENSP_mappings:4qy3.A&PDB-ENSP_mappings:4r59.A&PDB-ENSP_mappings:4r5a.A&PDB-ENSP_mappings:4r5b.A&PDB-ENSP_mappings:4rfc.A&PDB-ENSP_mappings:4rfd.A&PDB-ENSP_mappings:4rh2.A&PDB-ENSP_mappings:4riu.A&PDB-ENSP_mappings:4riv.A&PDB-ENSP_mappings:4rn4.A&PDB-ENSP_mappings:4rux.A&PDB-ENSP_mappings:4ruy.A&PDB-ENSP_mappings:4ruz.A&PDB-ENSP_mappings:4wl4.A&PDB-ENSP_mappings:4ww6.A&PDB-ENSP_mappings:4xe1.A&PDB-ENSP_mappings:4y0j.A&PDB-ENSP_mappings:4ygj.A&PDB-ENSP_mappings:4ygk.A&PDB-ENSP_mappings:4ygl.A&PDB-ENSP_mappings:4ygn.A&PDB-ENSP_mappings:4yvy.A&PDB-ENSP_mappings:4ywp.A&PDB-ENSP_mappings:4yx4.A&PDB-ENSP_mappings:4yxi.A&PDB-ENSP_mappings:4yxo.A&PDB-ENSP_mappings:4yxu.A&PDB-ENSP_mappings:4yyt.A&PDB-ENSP_mappings:4z0q.A&PDB-ENSP_mappings:4z1e.A&PDB-ENSP_mappings:4z1j.A&PDB-ENSP_mappings:4z1k.A&PDB-ENSP_mappings:4z1n.A&PDB-ENSP_mappings:4zao.A&PDB-ENSP_mappings:4zwi.A&PDB-ENSP_mappings:4zwx.A&PDB-ENSP_mappings:4zwy.A&PDB-ENSP_mappings:4zwz.A&PDB-ENSP_mappings:4zx0.A&PDB-ENSP_mappings:4zx1.A&PDB-ENSP_mappings:5a6h.A&PDB-ENSP_mappings:5amd.A&PDB-ENSP_mappings:5amg.A&PDB-ENSP_mappings:5aml.A&PDB-ENSP_mappings:5bnl.A&PDB-ENSP_mappings:5bru.A&PDB-ENSP_mappings:5brv.A&PDB-ENSP_mappings:5brw.A&PDB-ENSP_mappings:5byi.A&PDB-ENSP_mappings:5c8i.A&PDB-ENSP_mappings:5ca2.A&PDB-ENSP_mappings:5cac.A&PDB-ENSP_mappings:5clu.A&PDB-ENSP_mappings:5dog.A&PDB-ENSP_mappings:5doh.A&PDB-ENSP_mappings:5doh.B&PDB-ENSP_mappings:5drs.A&PDB-ENSP_mappings:5dsk.A&PDB-ENSP_mappings:5dsl.A&PDB-ENSP_mappings:5dsm.A&PDB-ENSP_mappings:5dso.A&PDB-ENSP_mappings:5dsp.A&PDB-ENSP_mappings:5dsq.A&PDB-ENSP_mappings:5dsr.A&PDB-ENSP_mappings:5e28.A&PDB-ENSP_mappings:5e2k.A&PDB-ENSP_mappings:5e2r.A&PDB-ENSP_mappings:5e2s.A&PDB-ENSP_mappings:5eh5.A&PDB-ENSP_mappings:5eh7.A&PDB-ENSP_mappings:5eh8.A&PDB-ENSP_mappings:5ehe.A&PDB-ENSP_mappings:5ehv.A&PDB-ENSP_mappings:5ehw.A&PDB-ENSP_mappings:5eij.A&PDB-ENSP_mappings:5ekh.A&PDB-ENSP_mappings:5ekj.A&PDB-ENSP_mappings:5ekm.A&PDB-ENSP_mappings:5eoi.A&PDB-ENSP_mappings:5fdc.A&PDB-ENSP_mappings:5fdi.A&PDB-ENSP_mappings:5flo.A&PDB-ENSP_mappings:5flp.A&PDB-ENSP_mappings:5flq.A&PDB-ENSP_mappings:5flr.A&PDB-ENSP_mappings:5fls.A&PDB-ENSP_mappings:5flt.A&PDB-ENSP_mappings:5fng.A&PDB-ENSP_mappings:5fnh.A&PDB-ENSP_mappings:5fni.A&PDB-ENSP_mappings:5fnj.A&PDB-ENSP_mappings:5fnk.A&PDB-ENSP_mappings:5fnl.A&PDB-ENSP_mappings:5fnm.A&PDB-ENSP_mappings:5g01.A&PDB-ENSP_mappings:5g03.A&PDB-ENSP_mappings:5g0b.A&PDB-ENSP_mappings:5g0c.A&PDB-ENSP_mappings:5gmn.A&PDB-ENSP_mappings:5j8z.A&PDB-ENSP_mappings:5jdv.B&PDB-ENSP_mappings:5je7.B&PDB-ENSP_mappings:5jeg.B&PDB-ENSP_mappings:5jeh.B&PDB-ENSP_mappings:5jep.B&PDB-ENSP_mappings:5jes.B&PDB-ENSP_mappings:5jg3.B&PDB-ENSP_mappings:5jg5.B&PDB-ENSP_mappings:5jgs.B&PDB-ENSP_mappings:5jgt.B&PDB-ENSP_mappings:5jmz.A&PDB-ENSP_mappings:5jn3.A&PDB-ENSP_mappings:5jn7.A&PDB-ENSP_mappings:5jq0.A&PDB-ENSP_mappings:5jqt.A&PDB-ENSP_mappings:5l3o.A&PDB-ENSP_mappings:5l3o.B&PDB-ENSP_mappings:5l6k.A&PDB-ENSP_mappings:5l6k.B&PDB-ENSP_mappings:5l6t.A&PDB-ENSP_mappings:5l6t.B&PDB-ENSP_mappings:5l70.A&PDB-ENSP_mappings:5l70.B&PDB-ENSP_mappings:5l9e.A&PDB-ENSP_mappings:5l9e.B&PDB-ENSP_mappings:5l9e.C&PDB-ENSP_mappings:5l9e.D&PDB-ENSP_mappings:5ljq.A&PDB-ENSP_mappings:5ljt.A&PDB-ENSP_mappings:5ll4.A&PDB-ENSP_mappings:5ll4.B&PDB-ENSP_mappings:5ll8.A&PDB-ENSP_mappings:5llc.A&PDB-ENSP_mappings:5lle.A&PDB-ENSP_mappings:5llg.A&PDB-ENSP_mappings:5llh.A&PDB-ENSP_mappings:5lmd.A&PDB-ENSP_mappings:5lvs.A&PDB-ENSP_mappings:5lvs.B&PDB-ENSP_mappings:5m78.A&PDB-ENSP_mappings:5mjn.A&PDB-ENSP_mappings:5n0d.A&PDB-ENSP_mappings:5n0e.A&PDB-ENSP_mappings:5n1r.A&PDB-ENSP_mappings:5n1s.A&PDB-ENSP_mappings:5n24.A&PDB-ENSP_mappings:5n25.A&PDB-ENSP_mappings:5nea.A&PDB-ENSP_mappings:5nee.A&PDB-ENSP_mappings:5nxg.A&PDB-ENSP_mappings:5nxi.A&PDB-ENSP_mappings:5nxm.A&PDB-ENSP_mappings:5nxo.A&PDB-ENSP_mappings:5nxp.A&PDB-ENSP_mappings:5nxv.A&PDB-ENSP_mappings:5nxw.A&PDB-ENSP_mappings:5ny1.A&PDB-ENSP_mappings:5ny3.A&PDB-ENSP_mappings:5ny6.A&PDB-ENSP_mappings:5nya.A&PDB-ENSP_mappings:5o07.A&PDB-ENSP_mappings:5ogn.A&PDB-ENSP_mappings:5ogn.B&PDB-ENSP_mappings:5ogo.A&PDB-ENSP_mappings:5ogp.A&PDB-ENSP_mappings:5sz0.A&PDB-ENSP_mappings:5sz1.A&PDB-ENSP_mappings:5sz2.A&PDB-ENSP_mappings:5sz3.A&PDB-ENSP_mappings:5sz4.A&PDB-ENSP_mappings:5sz5.A&PDB-ENSP_mappings:5sz6.A&PDB-ENSP_mappings:5sz7.A&PDB-ENSP_mappings:5t71.A&PDB-ENSP_mappings:5t72.A&PDB-ENSP_mappings:5t74.A&PDB-ENSP_mappings:5t75.A&PDB-ENSP_mappings:5tfx.A&PDB-ENSP_mappings:5th4.A&PDB-ENSP_mappings:5thi.A&PDB-ENSP_mappings:5thj.A&PDB-ENSP_mappings:5thn.A&PDB-ENSP_mappings:5ti0.A&PDB-ENSP_mappings:5txy.A&PDB-ENSP_mappings:5ty1.A&PDB-ENSP_mappings:5ty8.A&PDB-ENSP_mappings:5ty9.A&PDB-ENSP_mappings:5tya.A&PDB-ENSP_mappings:5u0d.A&PDB-ENSP_mappings:5u0e.A&PDB-ENSP_mappings:5u0f.A&PDB-ENSP_mappings:5u0g.A&PDB-ENSP_mappings:5uln.A&PDB-ENSP_mappings:5umc.A&PDB-ENSP_mappings:5vgy.A&PDB-ENSP_mappings:5w8b.A&PDB-ENSP_mappings:5wex.A&PDB-ENSP_mappings:5wg7.A&PDB-ENSP_mappings:5wgp.A&PDB-ENSP_mappings:5wlr.A&PDB-ENSP_mappings:5wlt.A&PDB-ENSP_mappings:5wlu.A&PDB-ENSP_mappings:5wlv.A&PDB-ENSP_mappings:5y2r.A&PDB-ENSP_mappings:5y2s.A&PDB-ENSP_mappings:5yui.A&PDB-ENSP_mappings:5yuj.A&PDB-ENSP_mappings:5yuk.A&PDB-ENSP_mappings:5zxw.A&PDB-ENSP_mappings:6b00.A&PDB-ENSP_mappings:6b4d.A&PDB-ENSP_mappings:6b59.A&PDB-ENSP_mappings:6b5a.A&PDB-ENSP_mappings:6bbs.A&PDB-ENSP_mappings:6bc9.A&PDB-ENSP_mappings:6bcc.A&PDB-ENSP_mappings:6c7w.A&PDB-ENSP_mappings:6c7x.A&PDB-ENSP_mappings:6ca2.A&PDB-ENSP_mappings:6ceh.A&PDB-ENSP_mappings:6cjv.A&PDB-ENSP_mappings:6d1l.A&PDB-ENSP_mappings:6d1m.A&PDB-ENSP_mappings:6e8p.A&PDB-ENSP_mappings:6e8x.A&PDB-ENSP_mappings:6e91.A&PDB-ENSP_mappings:6e92.A&PDB-ENSP_mappings:6ebe.A&PDB-ENSP_mappings:6ecz.A&PDB-ENSP_mappings:6eda.A&PDB-ENSP_mappings:6eea.A&PDB-ENSP_mappings:6eeh.A&PDB-ENSP_mappings:6eeo.A&PDB-ENSP_mappings:6equ.A&PDB-ENSP_mappings:6fji.A&PDB-ENSP_mappings:6fjj.A&PDB-ENSP_mappings:6g3q.A&PDB-ENSP_mappings:6g6t.A&PDB-ENSP_mappings:6gcy.A&PDB-ENSP_mappings:6gdc.A&PDB-ENSP_mappings:6gm9.A&PDB-ENSP_mappings:6got.A&PDB-ENSP_mappings:6gxb.A&PDB-ENSP_mappings:6gxe.A&PDB-ENSP_mappings:6h29.A&PDB-ENSP_mappings:6h2z.A&PDB-ENSP_mappings:6h33.A&PDB-ENSP_mappings:6h34.A&PDB-ENSP_mappings:6h3q.A&PDB-ENSP_mappings:6h6s.A&PDB-ENSP_mappings:6h6s.B&PDB-ENSP_mappings:6hd2.A&PDB-ENSP_mappings:6hqx.A&PDB-ENSP_mappings:6hr3.A&PDB-ENSP_mappings:6hx5.A&PDB-ENSP_mappings:6hxd.A&PDB-ENSP_mappings:6hzx.A&PDB-ENSP_mappings:6hzx.B&PDB-ENSP_mappings:6i0w.A&PDB-ENSP_mappings:6i1u.A&PDB-ENSP_mappings:6i2f.A&PDB-ENSP_mappings:6i3e.A&PDB-ENSP_mappings:6ic2.A&PDB-ENSP_mappings:6klz.A&PDB-ENSP_mappings:6km0.A&PDB-ENSP_mappings:6km1.A&PDB-ENSP_mappings:6km2.A&PDB-ENSP_mappings:6km3.A&PDB-ENSP_mappings:6km4.A&PDB-ENSP_mappings:6km5.A&PDB-ENSP_mappings:6km6.A&PDB-ENSP_mappings:6luu.A&PDB-ENSP_mappings:6luv.A&PDB-ENSP_mappings:6luw.A&PDB-ENSP_mappings:6lux.A&PDB-ENSP_mappings:6luy.A&PDB-ENSP_mappings:6luz.A&PDB-ENSP_mappings:6lv1.A&PDB-ENSP_mappings:6lv2.A&PDB-ENSP_mappings:6lv3.A&PDB-ENSP_mappings:6lv4.A&PDB-ENSP_mappings:6lv5.A&PDB-ENSP_mappings:6lv6.A&PDB-ENSP_mappings:6lv7.A&PDB-ENSP_mappings:6lv8.A&PDB-ENSP_mappings:6lv9.A&PDB-ENSP_mappings:6lva.A&PDB-ENSP_mappings:6mbv.A&PDB-ENSP_mappings:6mby.A&PDB-ENSP_mappings:6nj2.A&PDB-ENSP_mappings:6nj2.B&PDB-ENSP_mappings:6nj3.A&PDB-ENSP_mappings:6nj4.A&PDB-ENSP_mappings:6nj5.A&PDB-ENSP_mappings:6nj6.A&PDB-ENSP_mappings:6nj6.B&PDB-ENSP_mappings:6nlv.A&PDB-ENSP_mappings:6nm0.A&PDB-ENSP_mappings:6odz.A&PDB-ENSP_mappings:6oe0.A&PDB-ENSP_mappings:6oe1.A&PDB-ENSP_mappings:6oti.A&PDB-ENSP_mappings:6otk.A&PDB-ENSP_mappings:6otm.A&PDB-ENSP_mappings:6oto.A&PDB-ENSP_mappings:6otp.A&PDB-ENSP_mappings:6otq.A&PDB-ENSP_mappings:6oub.A&PDB-ENSP_mappings:6oud.A&PDB-ENSP_mappings:6oue.A&PDB-ENSP_mappings:6ouf.A&PDB-ENSP_mappings:6ouh.A&PDB-ENSP_mappings:6oui.A&PDB-ENSP_mappings:6ouj.A&PDB-ENSP_mappings:6ouk.A&PDB-ENSP_mappings:6oum.A&PDB-ENSP_mappings:6pdv.A&PDB-ENSP_mappings:6pea.A&PDB-ENSP_mappings:6pgx.A&PDB-ENSP_mappings:6q3o.A&PDB-ENSP_mappings:6q3o.B&PDB-ENSP_mappings:6q3o.C&PDB-ENSP_mappings:6q3o.D&PDB-ENSP_mappings:6q9t.A&PDB-ENSP_mappings:6qeb.A&PDB-ENSP_mappings:6qfu.A&PDB-ENSP_mappings:6qfv.A&PDB-ENSP_mappings:6qfw.A&PDB-ENSP_mappings:6qfx.A&PDB-ENSP_mappings:6ql1.A&PDB-ENSP_mappings:6ql2.A&PDB-ENSP_mappings:6ql3.A&PDB-ENSP_mappings:6r6f.A&PDB-ENSP_mappings:6r6j.A&PDB-ENSP_mappings:6rfh.A&PDB-ENSP_mappings:6rg3.A&PDB-ENSP_mappings:6rg4.A&PDB-ENSP_mappings:6rg5.A&PDB-ENSP_mappings:6rh4.A&PDB-ENSP_mappings:6rhj.A&PDB-ENSP_mappings:6rhk.A&PDB-ENSP_mappings:6rig.A&PDB-ENSP_mappings:6rit.A&PDB-ENSP_mappings:6rjj.A&PDB-ENSP_mappings:6rkn.A&PDB-ENSP_mappings:6rl9.A&PDB-ENSP_mappings:6rm1.A&PDB-ENSP_mappings:6rmp.A&PDB-ENSP_mappings:6rmx.A&PDB-ENSP_mappings:6rmy.A&PDB-ENSP_mappings:6rnp.A&PDB-ENSP_mappings:6rob.A&PDB-ENSP_mappings:6roe.A&PDB-ENSP_mappings:6rof.A&PDB-ENSP_mappings:6rqi.A&PDB-ENSP_mappings:6rrg.A&PDB-ENSP_mappings:6rri.A&PDB-ENSP_mappings:6rs5.A&PDB-ENSP_mappings:6rsz.A&PDB-ENSP_mappings:6rvf.A&PDB-ENSP_mappings:6rvk.A&PDB-ENSP_mappings:6rvl.A&PDB-ENSP_mappings:6rw1.A&PDB-ENSP_mappings:6rzx.A&PDB-ENSP_mappings:6s03.A&PDB-ENSP_mappings:6s9g.A&PDB-ENSP_mappings:6s9z.A&PDB-ENSP_mappings:6sac.A&PDB-ENSP_mappings:6sas.A&PDB-ENSP_mappings:6say.A&PDB-ENSP_mappings:6sb7.A&PDB-ENSP_mappings:6sbh.A&PDB-ENSP_mappings:6sbl.A&PDB-ENSP_mappings:6sbm.A&PDB-ENSP_mappings:6sd7.A&PDB-ENSP_mappings:6sdh.A&PDB-ENSP_mappings:6sdi.A&PDB-ENSP_mappings:6sdj.A&PDB-ENSP_mappings:6sdl.A&PDB-ENSP_mappings:6sds.A&PDB-ENSP_mappings:6sey.A&PDB-ENSP_mappings:6sfq.A&PDB-ENSP_mappings:6sfu.A&PDB-ENSP_mappings:6sg0.A&PDB-ENSP_mappings:6sg6.A&PDB-ENSP_mappings:6sx9.A&PDB-ENSP_mappings:6syb.A&PDB-ENSP_mappings:6sys.A&PDB-ENSP_mappings:6t4n.A&PDB-ENSP_mappings:6t4o.A&PDB-ENSP_mappings:6t4p.A&PDB-ENSP_mappings:6t5c.A&PDB-ENSP_mappings:6t7u.A&PDB-ENSP_mappings:6t81.A&PDB-ENSP_mappings:6t9z.A&PDB-ENSP_mappings:6u4q.A&PDB-ENSP_mappings:6u4t.A&PDB-ENSP_mappings:6ufb.A&PDB-ENSP_mappings:6ufc.A&PDB-ENSP_mappings:6ufd.A&PDB-ENSP_mappings:6ugn.A&PDB-ENSP_mappings:6ugo.A&PDB-ENSP_mappings:6ugp.A&PDB-ENSP_mappings:6ugq.A&PDB-ENSP_mappings:6ugr.A&PDB-ENSP_mappings:6ugz.A&PDB-ENSP_mappings:6uh0.A&PDB-ENSP_mappings:6ux1.A&PDB-ENSP_mappings:6uzu.A&PDB-ENSP_mappings:6vj3.A&PDB-ENSP_mappings:6vkg.A&PDB-ENSP_mappings:6wka.A&PDB-ENSP_mappings:6wq4.A&PDB-ENSP_mappings:6wq5.A&PDB-ENSP_mappings:6wq7.A&PDB-ENSP_mappings:6wq8.A&PDB-ENSP_mappings:6wq9.A&PDB-ENSP_mappings:6xvh.A&PDB-ENSP_mappings:6xwz.AAA&PDB-ENSP_mappings:6xxt.X&PDB-ENSP_mappings:6yh4.A&PDB-ENSP_mappings:6yh5.A&PDB-ENSP_mappings:6yh6.A&PDB-ENSP_mappings:6yh7.A&PDB-ENSP_mappings:6yh8.A&PDB-ENSP_mappings:6yh9.A&PDB-ENSP_mappings:6yha.A&PDB-ENSP_mappings:6yhb.A&PDB-ENSP_mappings:6yhc.A&PDB-ENSP_mappings:6yj3.A&PDB-ENSP_mappings:6ykc.A&PDB-ENSP_mappings:6ykh.A&PDB-ENSP_mappings:6yma.A&PDB-ENSP_mappings:6ymb.A&PDB-ENSP_mappings:6yo2.A&PDB-ENSP_mappings:6yo4.A&PDB-ENSP_mappings:6yo7.A&PDB-ENSP_mappings:6yoi.A&PDB-ENSP_mappings:6yok.A&PDB-ENSP_mappings:6yol.A&PDB-ENSP_mappings:6ypw.A&PDB-ENSP_mappings:6yqt.A&PDB-ENSP_mappings:6yqu.A&PDB-ENSP_mappings:6yri.A&PDB-ENSP_mappings:6yzj.A&PDB-ENSP_mappings:6yzk.A&PDB-ENSP_mappings:6yzl.A&PDB-ENSP_mappings:6yzm.A&PDB-ENSP_mappings:6yzn.A&PDB-ENSP_mappings:6yzo.A&PDB-ENSP_mappings:6yzp.A&PDB-ENSP_mappings:6yzq.A&PDB-ENSP_mappings:6yzr.A&PDB-ENSP_mappings:6yzs.A&PDB-ENSP_mappings:6yzt.A&PDB-ENSP_mappings:6yzu.A&PDB-ENSP_mappings:6yzv.A&PDB-ENSP_mappings:6yzw.A&PDB-ENSP_mappings:6yzx.A&PDB-ENSP_mappings:6z04.A&PDB-ENSP_mappings:6zr8.A&PDB-ENSP_mappings:7a6v.A&PDB-ENSP_mappings:7aeq.A&PDB-ENSP_mappings:7aes.A&PDB-ENSP_mappings:7agn.A&PDB-ENSP_mappings:7agn.B&PDB-ENSP_mappings:7agn.C&PDB-ENSP_mappings:7agn.D&PDB-ENSP_mappings:7asj.A&PDB-ENSP_mappings:7bfa.AAA&PDB-ENSP_mappings:7bg5.AAA&PDB-ENSP_mappings:7bhh.A&PDB-ENSP_mappings:7bi5.A&PDB-ENSP_mappings:7ca2.A&PDB-ENSP_mappings:7jnr.A&PDB-ENSP_mappings:7jnv.A&PDB-ENSP_mappings:7jnw.A&PDB-ENSP_mappings:7jnx.A&PDB-ENSP_mappings:7jnz.A&PDB-ENSP_mappings:7jo0.A&PDB-ENSP_mappings:7jo1.A&PDB-ENSP_mappings:7jo2.A&PDB-ENSP_mappings:7jo3.A&PDB-ENSP_mappings:7job.A&PDB-ENSP_mappings:7joc.A&PDB-ENSP_mappings:7k6i.A&PDB-ENSP_mappings:7k6j.A&PDB-ENSP_mappings:7k6k.A&PDB-ENSP_mappings:7k6l.A&PDB-ENSP_mappings:7k6t.A&PDB-ENSP_mappings:7k6u.A&PDB-ENSP_mappings:7k6x.A&PDB-ENSP_mappings:7k6z.A&PDB-ENSP_mappings:7m23.A&PDB-ENSP_mappings:7m24.A&PDB-ENSP_mappings:7m26.A&PDB-ENSP_mappings:7mu3.A&PDB-ENSP_mappings:7nh6.AAA&PDB-ENSP_mappings:7nh8.AAA&PDB-ENSP_mappings:7ntb.A&PDB-ENSP_mappings:7nzr.A&PDB-ENSP_mappings:7nzs.A&PDB-ENSP_mappings:7nzt.A&PDB-ENSP_mappings:7nzu.A&PDB-ENSP_mappings:7nzw.A&PDB-ENSP_mappings:7nzx.A&PDB-ENSP_mappings:7ok8.AAA&PDB-ENSP_mappings:7onm.AAA&PDB-ENSP_mappings:7onp.AAA&PDB-ENSP_mappings:7onq.AAA&PDB-ENSP_mappings:7onv.AAA&PDB-ENSP_mappings:7orp.AAA&PDB-ENSP_mappings:7orq.AAA&PDB-ENSP_mappings:7q0c.A&PDB-ENSP_mappings:7q0c.B&PDB-ENSP_mappings:7q0e.A&PDB-ENSP_mappings:7qbh.AAA&PDB-ENSP_mappings:7qgx.A&PDB-ENSP_mappings:7qgy.A&PDB-ENSP_mappings:7qgz.A&PDB-ENSP_mappings:7rny.A&PDB-ENSP_mappings:7rnz.A&PDB-ENSP_mappings:7rre.A&PDB-ENSP_mappings:7rrf.A&PDB-ENSP_mappings:7suw.A&PDB-ENSP_mappings:7suy.A&PDB-ENSP_mappings:7sv1.A&PDB-ENSP_mappings:7sv8.A&PDB-ENSP_mappings:8ca2.A&PDB-ENSP_mappings:8dj9.A&PDB-ENSP_mappings:9ca2.A&PROSITE_profiles:PS51144&CDD:cd03119&PANTHER:PTHR18952&Pfam:PF00194&Gene3D:3.10.200.10&SMART:SM01057&Superfamily:SSF51069&AlphaFold_DB_import:AF-P00918-F1\n",
      "========================================\n",
      "17\n",
      "Protein_position\n",
      "3401\n",
      "[nan '394' '306' '300' '621']\n",
      "13558-13559\n",
      "========================================\n",
      "18\n",
      "BIOTYPE\n",
      "41\n",
      "['transcribed_unprocessed_pseudogene' 'processed_pseudogene' 'miRNA'\n",
      " 'lncRNA' 'unprocessed_pseudogene']\n",
      "transcribed_unprocessed_pseudogene\n",
      "========================================\n",
      "19\n",
      "MANE_PLUS_CLINICAL\n",
      "17\n",
      "[nan 'NM_033056.4' 'NM_001368067.1' 'NM_014191.4' 'NM_006640.5']\n",
      "NM_001368067.1\n",
      "========================================\n",
      "20\n",
      "GIVEN_REF\n",
      "2779\n",
      "['G' 'C' 'T' 'A' nan]\n",
      "CTAATCGGCCCATCTCTTTCCCTTTCTGTGTGGTTACCAGGTTACTGGAGGCCACTTAGTTGTTGGTTTTGGTTGGGACTTTCTGGACTTCAGATCTCAGCCCTCGTTTACTCACCCCTGAGCCCTCCCTGCCCTGCCCCCAGCAGTTCCTGACTTCGCCCCTATACACTGTCCCCGGAGCTCTTCCCTGGGC\n",
      "========================================\n",
      "21\n",
      "PUBMED\n",
      "12572\n",
      "[nan '24447899&35987071' '35987071' '21130836' '17544380&33073494']\n",
      "20981351&21444836&21894447&30387422&27247849&20952631&20521218&21953487&22938532&23820649&25266489&26091847&27233804&18187665&20401335&19379518&20565774&22024213&26602921&31469255&27480026&22552919&21172166&18794727&23856471&28611364&27103841&23575227&27333884&22949529&30955239&25941324&20921971&32024468&26058836&12374873&17496726&17512307&19553224&29587766&31056593&18059082&17150099&27589963&16120061&26111150&18534365&18615004&20869266&21414566&21807569&26602751&26879662&27669015&30668166&31090079&21395649&19743955&19842931&22683993&24332148&26694722&27108086&30753253&30756358&10212248&16844790&17335470&22495925&22703382&23065660&23073218&29514624&28887371&24954895&29888336&22382501&28351962&25592601&31819590&25432901&28931564&21305273&18437562&25443111&28607931&28496656&23202124&31018568&23927520&29455858&31141530&24910551&31105432&24932357&24093767&23689943&18794726&28734628&26046407&24621099&22703928&20581797&29404309&29912962&27841878&30006737&26920668&28089639&30823438&30578418&27618447&23303660&2238000\n",
      "========================================\n",
      "22\n",
      "PolyPhen\n",
      "912\n",
      "[nan 'benign(0)' 'possibly_damaging(0.737)' 'benign(0.049)'\n",
      " 'benign(0.068)']\n",
      "possibly_damaging(0.737)\n",
      "========================================\n",
      "23\n",
      "HIGH_INF_POS\n",
      "3\n",
      "[nan 'N' 'Y']\n",
      "N\n",
      "========================================\n",
      "24\n",
      "CANONICAL\n",
      "2\n",
      "['YES' nan]\n",
      "YES\n",
      "========================================\n",
      "25\n",
      "MANE_SELECT\n",
      "11361\n",
      "[nan 'NM_015658.4' 'NM_198317.3' 'NM_198576.4' 'NM_001130045.2']\n",
      "NM_001130045.2\n",
      "========================================\n",
      "26\n",
      "EVE_CLASS\n",
      "4\n",
      "[nan 'Uncertain' 'Benign' 'Pathogenic']\n",
      "Pathogenic\n",
      "========================================\n",
      "27\n",
      "INTRON\n",
      "3346\n",
      "[nan '18/18' '14/18' '13/18' '12/18']\n",
      "109/114\n",
      "========================================\n",
      "28\n",
      "TREMBL\n",
      "5451\n",
      "[nan 'A0A024R082.62' 'A0A5F9UJX7.13' 'A6NGS0.130' 'A0A5K1VW56.15']\n",
      "A0A1V0E3R0.29&A0A1V0E3R8.29&A0A1V0E3N6.29&A0A1V0E3Q4.29&A0A1V0E3R4.29&A0A1V0E3Q7.29&A0A1V0E3S4.29&A0A1V0E3Q3.29&X5CF87.68&A0A1V0E3N1.29&A0A1V0E3Q5.29&A0A1V0E3Q8.29&A0A1V0E3S6.29&A0A1V0E3R3.29&A0A1V0E3M7.29&A0A1V0E3P8.29&A0A1V0E3P6.29&A0A1V0E3S7.29&A0A1V0E3Q2.29&A0A1V0E3T1.29&A0A1V0E3P9.30&A0A1V0E3R6.29&A0A1V0E3N7.29&A0A1V0E3Q6.29&A0A1V0E3P3.29&A0A1V0E3P1.29&A0A1V0E3P0.29&A0A1V0E3S0.29&A0A1V0E3N3.29&A0A1V0E3Q1.29\n",
      "========================================\n",
      "29\n",
      "Consequence\n",
      "94\n",
      "['downstream_gene_variant' 'non_coding_transcript_exon_variant'\n",
      " 'upstream_gene_variant' '3_prime_UTR_variant' 'intron_variant']\n",
      "splice_acceptor_variant&splice_donor_variant&splice_donor_5th_base_variant&splice_polypyrimidine_tract_variant&intron_variant\n",
      "========================================\n",
      "30\n",
      "HGVSp\n",
      "29180\n",
      "[nan 'ENSP00000317992.6:p.Thr394%3D' 'ENSP00000317992.6:p.Glu306%3D'\n",
      " 'ENSP00000317992.6:p.Ile300Val' 'ENSP00000343930.3:p.Val621%3D']\n",
      "ENSP00000252825.3:p.His321_Gln322insGlyGlyHisArgLysGluGluAspGluAspValSerAspGluHisHisHisHisSerProSerHisArgHisArgGlyHisLysGluGluGluGluAspAspAspAspAspAspAspValSerThrGluTyrGlyHisGlnAlaHisArgHis\n",
      "========================================\n",
      "31\n",
      "Codons\n",
      "1430\n",
      "[nan 'acT/acC' 'gaA/gaG' 'Atc/Gtc' 'gtG/gtC']\n",
      "CCACCAATGACAGTGACCACCAGTGGGACCAGCCAATCCCGAAGCTCATTTAGCACGGCCAAAACCTCTACATCCCTACATTCACACACTTCCTCCACACACCATCCTGAAGTCACCTCAACTTCTACCACCAGCATCACCCCCAACCACACCAGTACA/-\n",
      "========================================\n",
      "32\n",
      "Feature\n",
      "29671\n",
      "['ENST00000450305.2' 'ENST00000442987.3' 'ENST00000612080.1'\n",
      " 'ENST00000506640.3' 'ENST00000412115.2']\n",
      "compmerge.0.Brain.chrUn_KI270467v1.1\n",
      "========================================\n",
      "33\n",
      "FLAGS\n",
      "5\n",
      "[nan 'cds_start_NF' 'cds_end_NF' 'cds_start_NF&cds_end_NF'\n",
      " 'cds_end_NF&cds_start_NF']\n",
      "cds_start_NF&cds_end_NF\n",
      "========================================\n",
      "34\n",
      "PHENO\n",
      "17\n",
      "[nan '0&1' '1&1' '1' '0&1&1']\n",
      "1&1&1&1&1&1&1&1&1\n",
      "========================================\n",
      "35\n",
      "SIFT\n",
      "205\n",
      "[nan 'tolerated(0.55)' 'tolerated(0.14)' 'tolerated(1)' 'tolerated(0.24)']\n",
      "deleterious_low_confidence(0.01)\n",
      "========================================\n",
      "36\n",
      "SpliceAI_pred_SYMBOL\n",
      "15393\n",
      "[nan 'NOC2L' 'KLHL17' 'PERM1' 'AGRN']\n",
      "ST6GALNAC3\n",
      "========================================\n",
      "37\n",
      "HGNC_ID\n",
      "22706\n",
      "['HGNC:37102' 'HGNC:48835' 'HGNC:49954' nan 'HGNC:50701']\n",
      "HGNC:37102\n",
      "========================================\n",
      "38\n",
      "Existing_variation\n",
      "268961\n",
      "['rs2691315' 'rs201535981' 'rs71260069' 'rs367730352' 'rs71267774']\n",
      "COSV106097567&COSV106097597&COSV106097613&COSV106097618&COSV51544217&COSV51551832&COSV51558469&COSV51558827&COSV51561854\n",
      "========================================\n",
      "39\n",
      "SWISSPROT\n",
      "15309\n",
      "[nan 'Q9Y3T9.189' 'Q6TDP4.147' 'Q5SV97.109' 'O00468.205']\n",
      "P0DUQ1.6&P0DUQ2.6\n",
      "========================================\n",
      "40\n",
      "Feature_type\n",
      "4\n",
      "['Transcript' nan 'RegulatoryFeature' 'MotifFeature']\n",
      "RegulatoryFeature\n",
      "========================================\n",
      "41\n",
      "MOTIF_NAME\n",
      "8\n",
      "[nan 'ENSPFM0497' 'ENSPFM0042' 'ENSPFM0239' 'ENSPFM0598']\n",
      "ENSPFM0497\n",
      "========================================\n",
      "42\n",
      "TRANSCRIPTION_FACTORS\n",
      "8\n",
      "[nan 'SOX6::TBX21' 'CTCF' 'GCM1::MAX' 'TFAP2C::MAX']\n",
      "MEF2A&MEF2D&MEF2B\n",
      "========================================\n",
      "43\n",
      "NMD\n",
      "2\n",
      "[nan 'NMD_escaping_variant']\n",
      "NMD_escaping_variant\n",
      "========================================\n",
      "44\n",
      "SYMBOL_SOURCE\n",
      "4\n",
      "['HGNC' nan 'EntrezGene' 'RFAM']\n",
      "EntrezGene\n",
      "========================================\n",
      "45\n",
      "BAM_EDIT\n",
      "2\n",
      "[nan 'OK']\n",
      "OK\n",
      "========================================\n",
      "46\n",
      "VAR_SYNONYMS\n",
      "25211\n",
      "[nan 'ClinVar::VCV001185186&RCV001543790--COSMIC::COSM3807975'\n",
      " 'ClinVar::VCV001185191&RCV001543795--COSMIC::COSM146863'\n",
      " 'COSMIC::COSM4594799' 'COSMIC::COSM5546535']\n",
      "ClinVar::RCV000716073&VCV000587963&RCV000715753&VCV000587879&RCV000715497&VCV000587833&RCV000717985&VCV000588716&RCV000717468&VCV000588473&RCV001575862&RCV001527980&RCV001529698&RCV001559429&RCV001572885&RCV002249437&RCV001668223&RCV002493272&RCV002490760&RCV002314562&RCV002313653&RCV002312443&RCV002312367&RCV002312322&RCV002312020&RCV002311825&RCV002311824&RCV002298473&RCV002315506&RCV002367693&RCV001683642&RCV001727797&RCV001662789&RCV000717578&RCV000716277&RCV000715616&RCV000715206&RCV000611767&RCV000513731&RCV000504347&VCV000436801&RCV000503821&VCV000436797&RCV000502291&VCV000436800&RCV000192369&VCV000212231&RCV000194391&VCV000212230&RCV000114296&VCV000126353&RCV000114295&VCV000126352&RCV001262347\n",
      "========================================\n",
      "47\n",
      "cDNA_position\n",
      "14066\n",
      "[nan '2105' '2100' '301' '251']\n",
      "11335-11336\n",
      "========================================\n",
      "48\n",
      "IMPACT\n",
      "4\n",
      "['MODIFIER' 'LOW' 'MODERATE' 'HIGH']\n",
      "MODIFIER\n",
      "========================================\n"
     ]
    }
   ],
   "source": [
    "for i in range(len(object_columns)):\n",
    "    print(i)\n",
    "    print(object_columns[i])\n",
    "    uniq = unvcf_df_csq[object_columns[i]].unique()\n",
    "    print(len(uniq))\n",
    "    print(uniq[:5])\n",
    "    x = np.argmax([0 if isinstance(y, float) else len(y) for y in uniq])\n",
    "    print(uniq[x])\n",
    "    print(\"========================================\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "CSQ:\n",
    "- object columns:\n",
    "    - UNIPARC - no arrays - 18096 unique\n",
    "    - APPRIS - no arrays - 8 unique\n",
    "    - SYMBOL - no arrays - 23275 unique\n",
    "    - EXON - no arrays - 2107 unique\n",
    "    - Amino_acids - no arrays - 794 unique\n",
    "    - SOURCE - no arrays - 3 unique\n",
    "    - USED_REF - no arays - 2777 unique\n",
    "    - PHENOTYPES - looks to be an **array** sep=+ ? - _54932 unique_\n",
    "    - CLIN_SIG - **arrays** sep=& - _184 unique_\n",
    "    - Allele - no arrays - 2350 unique\n",
    "    - HGVSc - no arrays - 207584 unique\n",
    "    - UNIPROT_ISOFORM - no arrays - 9307 unique\n",
    "    - CCDS - no arrays - 15897 unique\n",
    "    - ENSP - no arrays - 18398\n",
    "    - CDS_position - no arrays? (numbers linked xxxx-yyyy) - 6519 unique\n",
    "    - SOMATIC - **arrays**? (number linked x&x&x&x&y...) - _16 unique_\n",
    "    - DOMAINS - **arrays** sep=& - _20172 unique_\n",
    "    - Protein_position - no arrays? (numbers linked xxxx-yyyy) - 3401 unique\n",
    "    - BIOTYPE - no arrays - 41 unique\n",
    "    - MANE_PLUS_CLINICAL - no arrays - 17 unique\n",
    "    - GIVEN_REF - no arrays - 2779 unique\n",
    "    - PUBMED - **arrays** sep=& - _12572 unique_\n",
    "    - PolyPhen - no arrays but can be split! into a category and float (probablitity?) e.g. benign(0.737) - _912 unique_\n",
    "    - HIGH_INF_POS - no arrays - 3 unique\n",
    "    - CANONICAL - no arrays - 2 unique\n",
    "    - MANE_SELECT - no arrays - 11361 unique\n",
    "    - EVE_CLASS - no arrays (nan, uncertain, benign, pathogenic) - 4 unique\n",
    "    - INTRON - no arrays - 3346 unique\n",
    "    - TREMBL - **arrays** sep=& - _5451 unique_\n",
    "    - Consequence - **arrays** sep=& - _94 unique_\n",
    "    - HGVSp - **arrays**? - a lot of values concatenated (HisHisHisHis...) - _29180 unique_\n",
    "    - Codons - no arrays lower and uppercase values? - 1430 unique\n",
    "    - Feature - no arrays - 29671 unique\n",
    "    - FLAGS - **arrays** sep=& - _5 unique_\n",
    "    - PHENO -  **arrays**? (number linked x&x&x&x&y...) - _17 unique_\n",
    "    - SIFT - no arrays but can be split! into a category and float (probablitity?) e.g. tolerated(0.55) - _205 unique_\n",
    "    - SpliceAI_pred_SYMBOL - no arrays - 15393 unique\n",
    "    - HGNC_ID - no arrays but can be split? category:number - _22706 unique_\n",
    "    - Existing_variation - **arrays** sep=& - _268961 unique_\n",
    "    - SWISSPROT - **arrays** sep=& - _15309 unique_\n",
    "    - Feature_type - no arrays - 4 unique\n",
    "    - MOTIF_NAME - no arrays - 8 unique\n",
    "    - TRANSCRIPTION_FACTORS - **arrays** sep=& - _8 unique_\n",
    "    - NMD - no arrays (nan, NMD_escaping_variant) - 2 unique\n",
    "    - SYMBOL_SOURCE - no arrays - 4 unique\n",
    "    - BAM_EDIT - no arrays (nan, OK) - 2 unique\n",
    "    - VAR_SYNONYMS - **arrays** specific format sep=& 2 fields separated with :: - _25211 unique_\n",
    "    - cDNA_position - no arrays? (numbers linked xxxx-yyyy) - _14066 unique_\n",
    "    - IMPACT - no arrays - 4 unique"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "pathogen",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
