{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "BcWEw0OUZ8gD"
      },
      "source": [
        "### Reading all VCF data and saving it as CSV files"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "8J7DTN3QB82N"
      },
      "source": [
        "This notebook is here to convert VCF files to something more familiar and easier to use - CSV files. However, I do not claim to have enough expertise in working with VCF files to say if this makes sense and if I have done it correctly here. As such, I would like to ask other team members to review this code, check if such transformation even makes sense and improve this notebook as they see fit."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 592
        },
        "id": "2Hw4PIsmY28F",
        "outputId": "1e60c8db-c72e-4351-fc85-04058ba27c0c"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Collecting setuptools<58\n",
            "  Downloading setuptools-57.5.0-py3-none-any.whl (819 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m819.3/819.3 kB\u001b[0m \u001b[31m12.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hInstalling collected packages: setuptools\n",
            "  Attempting uninstall: setuptools\n",
            "    Found existing installation: setuptools 67.7.2\n",
            "    Uninstalling setuptools-67.7.2:\n",
            "      Successfully uninstalled setuptools-67.7.2\n",
            "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
            "ipython 7.34.0 requires jedi>=0.16, which is not installed.\n",
            "arviz 0.15.1 requires setuptools>=60.0.0, but you have setuptools 57.5.0 which is incompatible.\n",
            "cvxpy 1.3.2 requires setuptools>65.5.1, but you have setuptools 57.5.0 which is incompatible.\u001b[0m\u001b[31m\n",
            "\u001b[0mSuccessfully installed setuptools-57.5.0\n"
          ]
        },
        {
          "data": {
            "application/vnd.colab-display-data+json": {
              "pip_warning": {
                "packages": [
                  "_distutils_hack",
                  "pkg_resources",
                  "setuptools"
                ]
              }
            }
          },
          "metadata": {},
          "output_type": "display_data"
        },
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\u001b[33mWARNING: Skipping pyvcf as it is not installed.\u001b[0m\u001b[33m\n",
            "\u001b[0mCollecting pyvcf\n",
            "  Downloading PyVCF-0.6.8.tar.gz (34 kB)\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: setuptools in /usr/local/lib/python3.10/dist-packages (from pyvcf) (57.5.0)\n",
            "Building wheels for collected packages: pyvcf\n",
            "  Building wheel for pyvcf (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for pyvcf: filename=PyVCF-0.6.8-cp310-cp310-linux_x86_64.whl size=174265 sha256=6f62a567c25d33757750fddae1d38b530818cdd133b560904fdc38b1de4b4d72\n",
            "  Stored in directory: /root/.cache/pip/wheels/da/88/3b/b691691894da44fc597c93ba7f953a46d1413ed7411906a6d2\n",
            "Successfully built pyvcf\n",
            "Installing collected packages: pyvcf\n",
            "Successfully installed pyvcf-0.6.8\n"
          ]
        }
      ],
      "source": [
        "!pip install \"setuptools<58\" --upgrade\n",
        "!pip uninstall pyvcf\n",
        "!pip install pyvcf"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "id": "PDCDYXsAaNKR"
      },
      "outputs": [],
      "source": [
        "file1 = \"/content/drive/MyDrive/Pathogenicity_Data/EE_015.vcf.gz\"\n",
        "file2 = \"/content/drive/MyDrive/Pathogenicity_Data/EE_050.vcf.gz\"\n",
        "file3 = \"/content/drive/MyDrive/Pathogenicity_Data/EE_069.vcf.gz\""
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {
        "id": "FnIOz7lhl5gg"
      },
      "outputs": [],
      "source": [
        "import vcf\n",
        "import pandas as pd"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 58,
      "metadata": {},
      "outputs": [],
      "source": [
        "def convert_to_csv_test(vcf_file):\n",
        "    data = []\n",
        "    data2 = []\n",
        "\n",
        "    with open(vcf_file, 'r') as vcf_file_binary:\n",
        "        vcf_reader = vcf.Reader(vcf_file_binary)\n",
        "        column_headers = [\"CHROM\", \"POS\", \"ID\", \"REF\", \"ALT\", \"QUAL\", \"FILTER\"]\n",
        "\n",
        "        column_headers.extend(list(vcf_reader.infos.keys())[:-1])\n",
        "        column_csq_headers = vcf_reader.infos[\"CSQ\"].desc[50:].split(\"|\")\n",
        "\n",
        "        for record in vcf_reader:\n",
        "            row = []\n",
        "            row.extend([record.CHROM, record.POS, record.ID, record.REF, record.ALT, record.QUAL, record.FILTER])\n",
        "\n",
        "            for field in vcf_reader.infos.keys():\n",
        "                if field != \"CSQ\":\n",
        "                    info_value = record.INFO.get(field)\n",
        "                    row.append(info_value)\n",
        "                else:\n",
        "                    data2.append(record.INFO.get(field)[0].split(\"|\"))\n",
        "\n",
        "            data.append(row)\n",
        "\n",
        "    df1 = pd.DataFrame(data, columns=column_headers)\n",
        "    df2 = pd.DataFrame(data2, columns=column_csq_headers)\n",
        "    return df1, df2"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 59,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "KEY: CHROM                     VAL:chr1\n",
            "KEY: POS                       VAL:15820\n",
            "KEY: ID                        VAL:rs2691315\n",
            "KEY: REF                       VAL:G\n",
            "KEY: ALT                       VAL:[T]\n",
            "KEY: QUAL                      VAL:None\n",
            "KEY: FILTER                    VAL:[]\n",
            "KEY: ACMG_class                VAL:['Uncertain%40Significance']\n",
            "KEY: ACMG_coding_impact        VAL:['non%40coding']\n",
            "KEY: ACMG_gene                 VAL:None\n",
            "KEY: ACMG_rules                VAL:['BP4']\n",
            "KEY: ACMG_score                VAL:[0.193]\n",
            "KEY: ACMG_transcript           VAL:None\n",
            "KEY: AMP_matches               VAL:['V']\n",
            "KEY: AMP_rules                 VAL:['Freq_II%2CType_IV%2CPred_IV']\n",
            "KEY: AMP_score                 VAL:None\n",
            "KEY: AMP_tier                  VAL:['Tier%40IV']\n",
            "KEY: AMP_total_samples         VAL:[0]\n",
            "KEY: AS_FilterStatus           VAL:SITE\n",
            "KEY: AS_SB_TABLE               VAL:12%2C23|7%2C2\n",
            "KEY: CGDinheritance            VAL:None\n",
            "KEY: ClinVarClass              VAL:None\n",
            "KEY: ClinVarDisease            VAL:None\n",
            "KEY: DANN_score                VAL:None\n",
            "KEY: DP                        VAL:45\n",
            "KEY: ECNT                      VAL:1\n",
            "KEY: GERMQ                     VAL:55\n",
            "KEY: Gene                      VAL:['WASH7P']\n",
            "KEY: MBQ                       VAL:[29]\n",
            "KEY: MFRL                      VAL:[219]\n",
            "KEY: MMQ                       VAL:[34]\n",
            "KEY: MPOS                      VAL:[30]\n",
            "KEY: MutationTaster_pred       VAL:None\n",
            "KEY: MutationTaster_score      VAL:None\n",
            "KEY: POPAF                     VAL:[7.3]\n",
            "KEY: RPA                       VAL:None\n",
            "KEY: RU                        VAL:None\n",
            "KEY: SIFT_score                VAL:None\n",
            "KEY: STR                       VAL:None\n",
            "KEY: STRQ                      VAL:None\n",
            "KEY: TLOD                      VAL:[10.9]\n",
            "KEY: coding_impact             VAL:None\n",
            "KEY: cosmicFathMMPrediction    VAL:None\n",
            "KEY: cosmicFathMMScore         VAL:None\n",
            "KEY: function                  VAL:['non-coding%40exon']\n",
            "KEY: gnomadExomesEthnic_AC_Hom VAL:None\n",
            "KEY: gnomadExomes_AC           VAL:None\n",
            "KEY: gnomadExomes_AC_Hemi      VAL:None\n",
            "KEY: gnomadExomes_AC_Hom       VAL:None\n",
            "KEY: gnomadExomes_AF           VAL:None\n",
            "KEY: gnomadExomes_AF_ethnic    VAL:None\n",
            "KEY: gnomadExomes_AN           VAL:None\n",
            "KEY: gnomadGenomesAC           VAL:[18059]\n",
            "KEY: gnomadGenomesAN           VAL:[82218]\n",
            "KEY: gnomadGenomesEthnic_AC_Hom VAL:[233]\n",
            "KEY: gnomadGenomes_AC_Hemi     VAL:None\n",
            "KEY: gnomadGenomes_AC_Hom      VAL:[18059]\n",
            "KEY: gnomadGenomes_AF          VAL:[0.219648]\n",
            "KEY: gnomadGenomes_AF_ethnic   VAL:[0.0938074]\n",
            "KEY: hgvs                      VAL:None\n",
            "-----------------------------------------------------\n",
            "KEY: Allele                    VAL:T\n",
            "KEY: Consequence               VAL:downstream_gene_variant\n",
            "KEY: IMPACT                    VAL:MODIFIER\n",
            "KEY: SYMBOL                    VAL:DDX11L1\n",
            "KEY: Gene                      VAL:ENSG00000223972\n",
            "KEY: Feature_type              VAL:Transcript\n",
            "KEY: Feature                   VAL:ENST00000450305.2\n",
            "KEY: BIOTYPE                   VAL:transcribed_unprocessed_pseudogene\n",
            "KEY: EXON                      VAL:None\n",
            "KEY: INTRON                    VAL:None\n",
            "KEY: HGVSc                     VAL:None\n",
            "KEY: HGVSp                     VAL:None\n",
            "KEY: cDNA_position             VAL:None\n",
            "KEY: CDS_position              VAL:None\n",
            "KEY: Protein_position          VAL:None\n",
            "KEY: Amino_acids               VAL:None\n",
            "KEY: Codons                    VAL:None\n",
            "KEY: Existing_variation        VAL:rs2691315\n",
            "KEY: DISTANCE                  VAL:2150\n",
            "KEY: STRAND                    VAL:1\n",
            "KEY: FLAGS                     VAL:None\n",
            "KEY: SYMBOL_SOURCE             VAL:HGNC\n",
            "KEY: HGNC_ID                   VAL:HGNC:37102\n",
            "KEY: CANONICAL                 VAL:YES\n",
            "KEY: MANE_SELECT               VAL:None\n",
            "KEY: MANE_PLUS_CLINICAL        VAL:None\n",
            "KEY: TSL                       VAL:None\n",
            "KEY: APPRIS                    VAL:None\n",
            "KEY: CCDS                      VAL:None\n",
            "KEY: ENSP                      VAL:None\n",
            "KEY: SWISSPROT                 VAL:None\n",
            "KEY: TREMBL                    VAL:None\n",
            "KEY: UNIPARC                   VAL:None\n",
            "KEY: UNIPROT_ISOFORM           VAL:None\n",
            "KEY: REFSEQ_MATCH              VAL:None\n",
            "KEY: SOURCE                    VAL:Ensembl\n",
            "KEY: REFSEQ_OFFSET             VAL:None\n",
            "KEY: GIVEN_REF                 VAL:G\n",
            "KEY: USED_REF                  VAL:G\n",
            "KEY: BAM_EDIT                  VAL:None\n",
            "KEY: SIFT                      VAL:None\n",
            "KEY: PolyPhen                  VAL:None\n",
            "KEY: DOMAINS                   VAL:None\n",
            "KEY: HGVS_OFFSET               VAL:None\n",
            "KEY: AF                        VAL:0.4105\n",
            "KEY: AFR_AF                    VAL:0.4849\n",
            "KEY: AMR_AF                    VAL:0.2939\n",
            "KEY: EAS_AF                    VAL:0.6052\n",
            "KEY: EUR_AF                    VAL:0.2714\n",
            "KEY: SAS_AF                    VAL:0.3354\n",
            "KEY: gnomADe_AF                VAL:None\n",
            "KEY: gnomADe_AFR_AF            VAL:None\n",
            "KEY: gnomADe_AMR_AF            VAL:None\n",
            "KEY: gnomADe_ASJ_AF            VAL:None\n",
            "KEY: gnomADe_EAS_AF            VAL:None\n",
            "KEY: gnomADe_FIN_AF            VAL:None\n",
            "KEY: gnomADe_NFE_AF            VAL:None\n",
            "KEY: gnomADe_OTH_AF            VAL:None\n",
            "KEY: gnomADe_SAS_AF            VAL:None\n",
            "KEY: gnomADg_AF                VAL:0.2196\n",
            "KEY: gnomADg_AFR_AF            VAL:0.4324\n",
            "KEY: gnomADg_AMI_AF            VAL:0.1146\n",
            "KEY: gnomADg_AMR_AF            VAL:0.1638\n",
            "KEY: gnomADg_ASJ_AF            VAL:0.1285\n",
            "KEY: gnomADg_EAS_AF            VAL:0.06176\n",
            "KEY: gnomADg_FIN_AF            VAL:0.06299\n",
            "KEY: gnomADg_MID_AF            VAL:0.2151\n",
            "KEY: gnomADg_NFE_AF            VAL:0.09381\n",
            "KEY: gnomADg_OTH_AF            VAL:0.2235\n",
            "KEY: gnomADg_SAS_AF            VAL:0.1329\n",
            "KEY: CLIN_SIG                  VAL:None\n",
            "KEY: SOMATIC                   VAL:None\n",
            "KEY: PHENO                     VAL:None\n",
            "KEY: PUBMED                    VAL:None\n",
            "KEY: VAR_SYNONYMS              VAL:None\n",
            "KEY: MOTIF_NAME                VAL:None\n",
            "KEY: MOTIF_POS                 VAL:None\n",
            "KEY: HIGH_INF_POS              VAL:None\n",
            "KEY: MOTIF_SCORE_CHANGE        VAL:None\n",
            "KEY: TRANSCRIPTION_FACTORS     VAL:None\n",
            "KEY: EVE_CLASS                 VAL:None\n",
            "KEY: EVE_SCORE                 VAL:None\n",
            "KEY: CADD_PHRED                VAL:16.02\n",
            "KEY: CADD_RAW                  VAL:1.543896\n",
            "KEY: SpliceAI_pred_DP_AG       VAL:None\n",
            "KEY: SpliceAI_pred_DP_AL       VAL:None\n",
            "KEY: SpliceAI_pred_DP_DG       VAL:None\n",
            "KEY: SpliceAI_pred_DP_DL       VAL:None\n",
            "KEY: SpliceAI_pred_DS_AG       VAL:None\n",
            "KEY: SpliceAI_pred_DS_AL       VAL:None\n",
            "KEY: SpliceAI_pred_DS_DG       VAL:None\n",
            "KEY: SpliceAI_pred_DS_DL       VAL:None\n",
            "KEY: SpliceAI_pred_SYMBOL      VAL:None\n",
            "KEY: LOEUF                     VAL:None\n",
            "KEY: PHENOTYPES                VAL:None\n",
            "KEY: NMD                       VAL:None\n"
          ]
        }
      ],
      "source": [
        "df1, df2 = convert_to_csv_test(\"data/EE_sample/EE_sample.vcf\")\n",
        "for i, x in enumerate(df1.iloc[0]):\n",
        "    print(f\"KEY: {df1.columns[i]:25} VAL:{x}\")\n",
        "\n",
        "print(\"-----------------------------------------------------\")\n",
        "for i, x in enumerate(df2.iloc[0]):\n",
        "    print(f\"KEY: {df2.columns[i]:25} VAL:{x if x != '' else None}\")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 56,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>CHROM</th>\n",
              "      <th>POS</th>\n",
              "      <th>ID</th>\n",
              "      <th>REF</th>\n",
              "      <th>ALT</th>\n",
              "      <th>QUAL</th>\n",
              "      <th>FILTER</th>\n",
              "      <th>ACMG_class</th>\n",
              "      <th>ACMG_coding_impact</th>\n",
              "      <th>ACMG_gene</th>\n",
              "      <th>...</th>\n",
              "      <th>gnomadExomes_AF_ethnic</th>\n",
              "      <th>gnomadExomes_AN</th>\n",
              "      <th>gnomadGenomesAC</th>\n",
              "      <th>gnomadGenomesAN</th>\n",
              "      <th>gnomadGenomesEthnic_AC_Hom</th>\n",
              "      <th>gnomadGenomes_AC_Hemi</th>\n",
              "      <th>gnomadGenomes_AC_Hom</th>\n",
              "      <th>gnomadGenomes_AF</th>\n",
              "      <th>gnomadGenomes_AF_ethnic</th>\n",
              "      <th>hgvs</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>chr1</td>\n",
              "      <td>15820</td>\n",
              "      <td>rs2691315</td>\n",
              "      <td>G</td>\n",
              "      <td>[T]</td>\n",
              "      <td>None</td>\n",
              "      <td>[]</td>\n",
              "      <td>[Uncertain%40Significance]</td>\n",
              "      <td>[non%40coding]</td>\n",
              "      <td>None</td>\n",
              "      <td>...</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>[18059]</td>\n",
              "      <td>[82218]</td>\n",
              "      <td>[233]</td>\n",
              "      <td>None</td>\n",
              "      <td>[18059]</td>\n",
              "      <td>[0.219648]</td>\n",
              "      <td>[0.0938074]</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>chr1</td>\n",
              "      <td>17385</td>\n",
              "      <td>rs201535981</td>\n",
              "      <td>G</td>\n",
              "      <td>[A]</td>\n",
              "      <td>None</td>\n",
              "      <td>[]</td>\n",
              "      <td>[Uncertain%40Significance]</td>\n",
              "      <td>[non%40coding]</td>\n",
              "      <td>None</td>\n",
              "      <td>...</td>\n",
              "      <td>[0.199722]</td>\n",
              "      <td>[53936]</td>\n",
              "      <td>[21384]</td>\n",
              "      <td>[111582]</td>\n",
              "      <td>[1]</td>\n",
              "      <td>None</td>\n",
              "      <td>[21384]</td>\n",
              "      <td>[0.191644]</td>\n",
              "      <td>[0.213724]</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>chr1</td>\n",
              "      <td>17697</td>\n",
              "      <td>rs71260069</td>\n",
              "      <td>G</td>\n",
              "      <td>[C]</td>\n",
              "      <td>None</td>\n",
              "      <td>[]</td>\n",
              "      <td>[Benign]</td>\n",
              "      <td>[non%40coding]</td>\n",
              "      <td>None</td>\n",
              "      <td>...</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>[17539]</td>\n",
              "      <td>[116050]</td>\n",
              "      <td>[2]</td>\n",
              "      <td>None</td>\n",
              "      <td>[17539]</td>\n",
              "      <td>[0.151133]</td>\n",
              "      <td>[0.178196]</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>chr1</td>\n",
              "      <td>133129</td>\n",
              "      <td>rs367730352</td>\n",
              "      <td>G</td>\n",
              "      <td>[A]</td>\n",
              "      <td>None</td>\n",
              "      <td>[]</td>\n",
              "      <td>[Uncertain%40Significance]</td>\n",
              "      <td>[non%40coding]</td>\n",
              "      <td>None</td>\n",
              "      <td>...</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>[72154]</td>\n",
              "      <td>[131698]</td>\n",
              "      <td>[7468]</td>\n",
              "      <td>None</td>\n",
              "      <td>[72154]</td>\n",
              "      <td>[0.547875]</td>\n",
              "      <td>[0.597389]</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>chr1</td>\n",
              "      <td>183629</td>\n",
              "      <td>rs71267774</td>\n",
              "      <td>G</td>\n",
              "      <td>[A]</td>\n",
              "      <td>None</td>\n",
              "      <td>[]</td>\n",
              "      <td>[Benign]</td>\n",
              "      <td>[non%40coding]</td>\n",
              "      <td>None</td>\n",
              "      <td>...</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>[14902]</td>\n",
              "      <td>[111796]</td>\n",
              "      <td>[1]</td>\n",
              "      <td>None</td>\n",
              "      <td>[14902]</td>\n",
              "      <td>[0.133296]</td>\n",
              "      <td>[0.220891]</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>chr1</td>\n",
              "      <td>184267</td>\n",
              "      <td>rs1396210256</td>\n",
              "      <td>C</td>\n",
              "      <td>[T]</td>\n",
              "      <td>None</td>\n",
              "      <td>[]</td>\n",
              "      <td>[Likely%40Benign]</td>\n",
              "      <td>[non%40coding]</td>\n",
              "      <td>None</td>\n",
              "      <td>...</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>[887]</td>\n",
              "      <td>[144780]</td>\n",
              "      <td>[0]</td>\n",
              "      <td>None</td>\n",
              "      <td>[887]</td>\n",
              "      <td>[0.00612654]</td>\n",
              "      <td>[0.00631038]</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>6 rows × 60 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "  CHROM     POS            ID REF  ALT  QUAL FILTER  \\\n",
              "0  chr1   15820     rs2691315   G  [T]  None     []   \n",
              "1  chr1   17385   rs201535981   G  [A]  None     []   \n",
              "2  chr1   17697    rs71260069   G  [C]  None     []   \n",
              "3  chr1  133129   rs367730352   G  [A]  None     []   \n",
              "4  chr1  183629    rs71267774   G  [A]  None     []   \n",
              "5  chr1  184267  rs1396210256   C  [T]  None     []   \n",
              "\n",
              "                   ACMG_class ACMG_coding_impact ACMG_gene  ...  \\\n",
              "0  [Uncertain%40Significance]     [non%40coding]      None  ...   \n",
              "1  [Uncertain%40Significance]     [non%40coding]      None  ...   \n",
              "2                    [Benign]     [non%40coding]      None  ...   \n",
              "3  [Uncertain%40Significance]     [non%40coding]      None  ...   \n",
              "4                    [Benign]     [non%40coding]      None  ...   \n",
              "5           [Likely%40Benign]     [non%40coding]      None  ...   \n",
              "\n",
              "  gnomadExomes_AF_ethnic gnomadExomes_AN gnomadGenomesAC gnomadGenomesAN  \\\n",
              "0                   None            None         [18059]         [82218]   \n",
              "1             [0.199722]         [53936]         [21384]        [111582]   \n",
              "2                   None            None         [17539]        [116050]   \n",
              "3                   None            None         [72154]        [131698]   \n",
              "4                   None            None         [14902]        [111796]   \n",
              "5                   None            None           [887]        [144780]   \n",
              "\n",
              "  gnomadGenomesEthnic_AC_Hom gnomadGenomes_AC_Hemi gnomadGenomes_AC_Hom  \\\n",
              "0                      [233]                  None              [18059]   \n",
              "1                        [1]                  None              [21384]   \n",
              "2                        [2]                  None              [17539]   \n",
              "3                     [7468]                  None              [72154]   \n",
              "4                        [1]                  None              [14902]   \n",
              "5                        [0]                  None                [887]   \n",
              "\n",
              "  gnomadGenomes_AF gnomadGenomes_AF_ethnic  hgvs  \n",
              "0       [0.219648]             [0.0938074]  None  \n",
              "1       [0.191644]              [0.213724]  None  \n",
              "2       [0.151133]              [0.178196]  None  \n",
              "3       [0.547875]              [0.597389]  None  \n",
              "4       [0.133296]              [0.220891]  None  \n",
              "5     [0.00612654]            [0.00631038]  None  \n",
              "\n",
              "[6 rows x 60 columns]"
            ]
          },
          "execution_count": 56,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "df1"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 57,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Allele</th>\n",
              "      <th>Consequence</th>\n",
              "      <th>IMPACT</th>\n",
              "      <th>SYMBOL</th>\n",
              "      <th>Gene</th>\n",
              "      <th>Feature_type</th>\n",
              "      <th>Feature</th>\n",
              "      <th>BIOTYPE</th>\n",
              "      <th>EXON</th>\n",
              "      <th>INTRON</th>\n",
              "      <th>...</th>\n",
              "      <th>SpliceAI_pred_DP_DG</th>\n",
              "      <th>SpliceAI_pred_DP_DL</th>\n",
              "      <th>SpliceAI_pred_DS_AG</th>\n",
              "      <th>SpliceAI_pred_DS_AL</th>\n",
              "      <th>SpliceAI_pred_DS_DG</th>\n",
              "      <th>SpliceAI_pred_DS_DL</th>\n",
              "      <th>SpliceAI_pred_SYMBOL</th>\n",
              "      <th>LOEUF</th>\n",
              "      <th>PHENOTYPES</th>\n",
              "      <th>NMD</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>T</td>\n",
              "      <td>downstream_gene_variant</td>\n",
              "      <td>MODIFIER</td>\n",
              "      <td>DDX11L1</td>\n",
              "      <td>ENSG00000223972</td>\n",
              "      <td>Transcript</td>\n",
              "      <td>ENST00000450305.2</td>\n",
              "      <td>transcribed_unprocessed_pseudogene</td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td>...</td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>A</td>\n",
              "      <td>downstream_gene_variant</td>\n",
              "      <td>MODIFIER</td>\n",
              "      <td>DDX11L1</td>\n",
              "      <td>ENSG00000223972</td>\n",
              "      <td>Transcript</td>\n",
              "      <td>ENST00000450305.2</td>\n",
              "      <td>transcribed_unprocessed_pseudogene</td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td>...</td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>C</td>\n",
              "      <td>downstream_gene_variant</td>\n",
              "      <td>MODIFIER</td>\n",
              "      <td>DDX11L1</td>\n",
              "      <td>ENSG00000223972</td>\n",
              "      <td>Transcript</td>\n",
              "      <td>ENST00000450305.2</td>\n",
              "      <td>transcribed_unprocessed_pseudogene</td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td>...</td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>A</td>\n",
              "      <td>non_coding_transcript_exon_variant</td>\n",
              "      <td>MODIFIER</td>\n",
              "      <td>CICP27</td>\n",
              "      <td>ENSG00000233750</td>\n",
              "      <td>Transcript</td>\n",
              "      <td>ENST00000442987.3</td>\n",
              "      <td>processed_pseudogene</td>\n",
              "      <td>1/1</td>\n",
              "      <td></td>\n",
              "      <td>...</td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>A</td>\n",
              "      <td>downstream_gene_variant</td>\n",
              "      <td>MODIFIER</td>\n",
              "      <td>MIR6859-2</td>\n",
              "      <td>ENSG00000273874</td>\n",
              "      <td>Transcript</td>\n",
              "      <td>ENST00000612080.1</td>\n",
              "      <td>miRNA</td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td>...</td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>T</td>\n",
              "      <td>downstream_gene_variant</td>\n",
              "      <td>MODIFIER</td>\n",
              "      <td>MIR6859-2</td>\n",
              "      <td>ENSG00000273874</td>\n",
              "      <td>Transcript</td>\n",
              "      <td>ENST00000612080.1</td>\n",
              "      <td>miRNA</td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td>...</td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "      <td></td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>6 rows × 96 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "  Allele                         Consequence    IMPACT     SYMBOL  \\\n",
              "0      T             downstream_gene_variant  MODIFIER    DDX11L1   \n",
              "1      A             downstream_gene_variant  MODIFIER    DDX11L1   \n",
              "2      C             downstream_gene_variant  MODIFIER    DDX11L1   \n",
              "3      A  non_coding_transcript_exon_variant  MODIFIER     CICP27   \n",
              "4      A             downstream_gene_variant  MODIFIER  MIR6859-2   \n",
              "5      T             downstream_gene_variant  MODIFIER  MIR6859-2   \n",
              "\n",
              "              Gene Feature_type            Feature  \\\n",
              "0  ENSG00000223972   Transcript  ENST00000450305.2   \n",
              "1  ENSG00000223972   Transcript  ENST00000450305.2   \n",
              "2  ENSG00000223972   Transcript  ENST00000450305.2   \n",
              "3  ENSG00000233750   Transcript  ENST00000442987.3   \n",
              "4  ENSG00000273874   Transcript  ENST00000612080.1   \n",
              "5  ENSG00000273874   Transcript  ENST00000612080.1   \n",
              "\n",
              "                              BIOTYPE EXON INTRON  ... SpliceAI_pred_DP_DG  \\\n",
              "0  transcribed_unprocessed_pseudogene              ...                       \n",
              "1  transcribed_unprocessed_pseudogene              ...                       \n",
              "2  transcribed_unprocessed_pseudogene              ...                       \n",
              "3                processed_pseudogene  1/1         ...                       \n",
              "4                               miRNA              ...                       \n",
              "5                               miRNA              ...                       \n",
              "\n",
              "  SpliceAI_pred_DP_DL SpliceAI_pred_DS_AG SpliceAI_pred_DS_AL  \\\n",
              "0                                                               \n",
              "1                                                               \n",
              "2                                                               \n",
              "3                                                               \n",
              "4                                                               \n",
              "5                                                               \n",
              "\n",
              "  SpliceAI_pred_DS_DG SpliceAI_pred_DS_DL SpliceAI_pred_SYMBOL LOEUF  \\\n",
              "0                                                                      \n",
              "1                                                                      \n",
              "2                                                                      \n",
              "3                                                                      \n",
              "4                                                                      \n",
              "5                                                                      \n",
              "\n",
              "  PHENOTYPES NMD  \n",
              "0                 \n",
              "1                 \n",
              "2                 \n",
              "3                 \n",
              "4                 \n",
              "5                 \n",
              "\n",
              "[6 rows x 96 columns]"
            ]
          },
          "execution_count": 57,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "df2"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "df1.to_csv('data/EE_sample/EE_sample.csv', index=False)\n",
        "df2.to_csv('data/EE_sample/EE_sample_csq.csv', index=False)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 8,
      "metadata": {
        "id": "9A6LWWvFaskW"
      },
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>CHROM</th>\n",
              "      <th>POS</th>\n",
              "      <th>ID</th>\n",
              "      <th>REF</th>\n",
              "      <th>ALT</th>\n",
              "      <th>QUAL</th>\n",
              "      <th>FILTER</th>\n",
              "      <th>FORMAT</th>\n",
              "      <th>ACMG_class</th>\n",
              "      <th>ACMG_coding_impact</th>\n",
              "      <th>...</th>\n",
              "      <th>gnomadGenomesAC</th>\n",
              "      <th>gnomadGenomesAN</th>\n",
              "      <th>gnomadGenomesEthnic_AC_Hom</th>\n",
              "      <th>gnomadGenomes_AC_Hemi</th>\n",
              "      <th>gnomadGenomes_AC_Hom</th>\n",
              "      <th>gnomadGenomes_AF</th>\n",
              "      <th>gnomadGenomes_AF_ethnic</th>\n",
              "      <th>hgvs</th>\n",
              "      <th>CSQ</th>\n",
              "      <th>SAMPLE</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>chr1</td>\n",
              "      <td>15820</td>\n",
              "      <td>rs2691315</td>\n",
              "      <td>G</td>\n",
              "      <td>[T]</td>\n",
              "      <td>None</td>\n",
              "      <td>[]</td>\n",
              "      <td>GT:AB:AD:AF:DP:F1R2:F2R1:SB</td>\n",
              "      <td>[Uncertain%40Significance]</td>\n",
              "      <td>[non%40coding]</td>\n",
              "      <td>...</td>\n",
              "      <td>[18059]</td>\n",
              "      <td>[82218]</td>\n",
              "      <td>[233]</td>\n",
              "      <td>None</td>\n",
              "      <td>[18059]</td>\n",
              "      <td>[0.219648]</td>\n",
              "      <td>[0.0938074]</td>\n",
              "      <td>None</td>\n",
              "      <td>[T|downstream_gene_variant|MODIFIER|DDX11L1|EN...</td>\n",
              "      <td>0/1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>chr1</td>\n",
              "      <td>17385</td>\n",
              "      <td>rs201535981</td>\n",
              "      <td>G</td>\n",
              "      <td>[A]</td>\n",
              "      <td>None</td>\n",
              "      <td>[]</td>\n",
              "      <td>GT:AB:AD:AF:DP:F1R2:F2R1:SB</td>\n",
              "      <td>[Uncertain%40Significance]</td>\n",
              "      <td>[non%40coding]</td>\n",
              "      <td>...</td>\n",
              "      <td>[21384]</td>\n",
              "      <td>[111582]</td>\n",
              "      <td>[1]</td>\n",
              "      <td>None</td>\n",
              "      <td>[21384]</td>\n",
              "      <td>[0.191644]</td>\n",
              "      <td>[0.213724]</td>\n",
              "      <td>None</td>\n",
              "      <td>[A|downstream_gene_variant|MODIFIER|DDX11L1|EN...</td>\n",
              "      <td>0/1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>chr1</td>\n",
              "      <td>17697</td>\n",
              "      <td>rs71260069</td>\n",
              "      <td>G</td>\n",
              "      <td>[C]</td>\n",
              "      <td>None</td>\n",
              "      <td>[]</td>\n",
              "      <td>GT:AB:AD:AF:DP:F1R2:F2R1:SB</td>\n",
              "      <td>[Benign]</td>\n",
              "      <td>[non%40coding]</td>\n",
              "      <td>...</td>\n",
              "      <td>[17539]</td>\n",
              "      <td>[116050]</td>\n",
              "      <td>[2]</td>\n",
              "      <td>None</td>\n",
              "      <td>[17539]</td>\n",
              "      <td>[0.151133]</td>\n",
              "      <td>[0.178196]</td>\n",
              "      <td>None</td>\n",
              "      <td>[C|downstream_gene_variant|MODIFIER|DDX11L1|EN...</td>\n",
              "      <td>0/1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>chr1</td>\n",
              "      <td>133129</td>\n",
              "      <td>rs367730352</td>\n",
              "      <td>G</td>\n",
              "      <td>[A]</td>\n",
              "      <td>None</td>\n",
              "      <td>[]</td>\n",
              "      <td>GT:AB:AD:AF:DP:F1R2:F2R1:SB</td>\n",
              "      <td>[Uncertain%40Significance]</td>\n",
              "      <td>[non%40coding]</td>\n",
              "      <td>...</td>\n",
              "      <td>[72154]</td>\n",
              "      <td>[131698]</td>\n",
              "      <td>[7468]</td>\n",
              "      <td>None</td>\n",
              "      <td>[72154]</td>\n",
              "      <td>[0.547875]</td>\n",
              "      <td>[0.597389]</td>\n",
              "      <td>None</td>\n",
              "      <td>[A|non_coding_transcript_exon_variant|MODIFIER...</td>\n",
              "      <td>0/1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>chr1</td>\n",
              "      <td>183629</td>\n",
              "      <td>rs71267774</td>\n",
              "      <td>G</td>\n",
              "      <td>[A]</td>\n",
              "      <td>None</td>\n",
              "      <td>[]</td>\n",
              "      <td>GT:AB:AD:AF:DP:F1R2:F2R1:SB</td>\n",
              "      <td>[Benign]</td>\n",
              "      <td>[non%40coding]</td>\n",
              "      <td>...</td>\n",
              "      <td>[14902]</td>\n",
              "      <td>[111796]</td>\n",
              "      <td>[1]</td>\n",
              "      <td>None</td>\n",
              "      <td>[14902]</td>\n",
              "      <td>[0.133296]</td>\n",
              "      <td>[0.220891]</td>\n",
              "      <td>None</td>\n",
              "      <td>[A|downstream_gene_variant|MODIFIER|MIR6859-2|...</td>\n",
              "      <td>0/1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>102512</th>\n",
              "      <td>chrY</td>\n",
              "      <td>56856512</td>\n",
              "      <td>rs76439603</td>\n",
              "      <td>A</td>\n",
              "      <td>[C]</td>\n",
              "      <td>None</td>\n",
              "      <td>[]</td>\n",
              "      <td>GT:AB:AD:AF:DP:F1R2:F2R1:PGT:PID:PS:SB</td>\n",
              "      <td>[Uncertain%40Significance]</td>\n",
              "      <td>[non%40coding]</td>\n",
              "      <td>...</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>[C|downstream_gene_variant|MODIFIER|CTBP2P1|EN...</td>\n",
              "      <td>0/1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>102513</th>\n",
              "      <td>chrY</td>\n",
              "      <td>56856517</td>\n",
              "      <td>rs75739861</td>\n",
              "      <td>A</td>\n",
              "      <td>[G]</td>\n",
              "      <td>None</td>\n",
              "      <td>[]</td>\n",
              "      <td>GT:AB:AD:AF:DP:F1R2:F2R1:PGT:PID:PS:SB</td>\n",
              "      <td>[Uncertain%40Significance]</td>\n",
              "      <td>[non%40coding]</td>\n",
              "      <td>...</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>[G|downstream_gene_variant|MODIFIER|CTBP2P1|EN...</td>\n",
              "      <td>0/1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>102514</th>\n",
              "      <td>chrY</td>\n",
              "      <td>56856554</td>\n",
              "      <td>rs77088808</td>\n",
              "      <td>T</td>\n",
              "      <td>[G]</td>\n",
              "      <td>None</td>\n",
              "      <td>[]</td>\n",
              "      <td>GT:AB:AD:AF:DP:F1R2:F2R1:SB</td>\n",
              "      <td>[Uncertain%40Significance]</td>\n",
              "      <td>[non%40coding]</td>\n",
              "      <td>...</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>[G|downstream_gene_variant|MODIFIER|CTBP2P1|EN...</td>\n",
              "      <td>0/1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>102515</th>\n",
              "      <td>chrY</td>\n",
              "      <td>56856581</td>\n",
              "      <td>rs9320083</td>\n",
              "      <td>G</td>\n",
              "      <td>[A]</td>\n",
              "      <td>None</td>\n",
              "      <td>[]</td>\n",
              "      <td>GT:AB:AD:AF:DP:F1R2:F2R1:SB</td>\n",
              "      <td>[Uncertain%40Significance]</td>\n",
              "      <td>[non%40coding]</td>\n",
              "      <td>...</td>\n",
              "      <td>[2]</td>\n",
              "      <td>[135]</td>\n",
              "      <td>[0]</td>\n",
              "      <td>None</td>\n",
              "      <td>[2]</td>\n",
              "      <td>[0.0148148]</td>\n",
              "      <td>[0.0]</td>\n",
              "      <td>None</td>\n",
              "      <td>[A|downstream_gene_variant|MODIFIER|CTBP2P1|EN...</td>\n",
              "      <td>0/1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>102516</th>\n",
              "      <td>chrY</td>\n",
              "      <td>56856587</td>\n",
              "      <td>rs7067511</td>\n",
              "      <td>A</td>\n",
              "      <td>[C]</td>\n",
              "      <td>None</td>\n",
              "      <td>[]</td>\n",
              "      <td>GT:AB:AD:AF:DP:F1R2:F2R1:SB</td>\n",
              "      <td>[Uncertain%40Significance]</td>\n",
              "      <td>[non%40coding]</td>\n",
              "      <td>...</td>\n",
              "      <td>[123]</td>\n",
              "      <td>[123]</td>\n",
              "      <td>[0]</td>\n",
              "      <td>None</td>\n",
              "      <td>[123]</td>\n",
              "      <td>[1.0]</td>\n",
              "      <td>[1.0]</td>\n",
              "      <td>None</td>\n",
              "      <td>[C|downstream_gene_variant|MODIFIER|CTBP2P1|EN...</td>\n",
              "      <td>0/1</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>102517 rows × 63 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "       CHROM       POS           ID REF  ALT  QUAL FILTER  \\\n",
              "0       chr1     15820    rs2691315   G  [T]  None     []   \n",
              "1       chr1     17385  rs201535981   G  [A]  None     []   \n",
              "2       chr1     17697   rs71260069   G  [C]  None     []   \n",
              "3       chr1    133129  rs367730352   G  [A]  None     []   \n",
              "4       chr1    183629   rs71267774   G  [A]  None     []   \n",
              "...      ...       ...          ...  ..  ...   ...    ...   \n",
              "102512  chrY  56856512   rs76439603   A  [C]  None     []   \n",
              "102513  chrY  56856517   rs75739861   A  [G]  None     []   \n",
              "102514  chrY  56856554   rs77088808   T  [G]  None     []   \n",
              "102515  chrY  56856581    rs9320083   G  [A]  None     []   \n",
              "102516  chrY  56856587    rs7067511   A  [C]  None     []   \n",
              "\n",
              "                                        FORMAT                  ACMG_class  \\\n",
              "0                  GT:AB:AD:AF:DP:F1R2:F2R1:SB  [Uncertain%40Significance]   \n",
              "1                  GT:AB:AD:AF:DP:F1R2:F2R1:SB  [Uncertain%40Significance]   \n",
              "2                  GT:AB:AD:AF:DP:F1R2:F2R1:SB                    [Benign]   \n",
              "3                  GT:AB:AD:AF:DP:F1R2:F2R1:SB  [Uncertain%40Significance]   \n",
              "4                  GT:AB:AD:AF:DP:F1R2:F2R1:SB                    [Benign]   \n",
              "...                                        ...                         ...   \n",
              "102512  GT:AB:AD:AF:DP:F1R2:F2R1:PGT:PID:PS:SB  [Uncertain%40Significance]   \n",
              "102513  GT:AB:AD:AF:DP:F1R2:F2R1:PGT:PID:PS:SB  [Uncertain%40Significance]   \n",
              "102514             GT:AB:AD:AF:DP:F1R2:F2R1:SB  [Uncertain%40Significance]   \n",
              "102515             GT:AB:AD:AF:DP:F1R2:F2R1:SB  [Uncertain%40Significance]   \n",
              "102516             GT:AB:AD:AF:DP:F1R2:F2R1:SB  [Uncertain%40Significance]   \n",
              "\n",
              "       ACMG_coding_impact  ... gnomadGenomesAC gnomadGenomesAN  \\\n",
              "0          [non%40coding]  ...         [18059]         [82218]   \n",
              "1          [non%40coding]  ...         [21384]        [111582]   \n",
              "2          [non%40coding]  ...         [17539]        [116050]   \n",
              "3          [non%40coding]  ...         [72154]        [131698]   \n",
              "4          [non%40coding]  ...         [14902]        [111796]   \n",
              "...                   ...  ...             ...             ...   \n",
              "102512     [non%40coding]  ...            None            None   \n",
              "102513     [non%40coding]  ...            None            None   \n",
              "102514     [non%40coding]  ...            None            None   \n",
              "102515     [non%40coding]  ...             [2]           [135]   \n",
              "102516     [non%40coding]  ...           [123]           [123]   \n",
              "\n",
              "       gnomadGenomesEthnic_AC_Hom gnomadGenomes_AC_Hemi gnomadGenomes_AC_Hom  \\\n",
              "0                           [233]                  None              [18059]   \n",
              "1                             [1]                  None              [21384]   \n",
              "2                             [2]                  None              [17539]   \n",
              "3                          [7468]                  None              [72154]   \n",
              "4                             [1]                  None              [14902]   \n",
              "...                           ...                   ...                  ...   \n",
              "102512                       None                  None                 None   \n",
              "102513                       None                  None                 None   \n",
              "102514                       None                  None                 None   \n",
              "102515                        [0]                  None                  [2]   \n",
              "102516                        [0]                  None                [123]   \n",
              "\n",
              "       gnomadGenomes_AF gnomadGenomes_AF_ethnic  hgvs  \\\n",
              "0            [0.219648]             [0.0938074]  None   \n",
              "1            [0.191644]              [0.213724]  None   \n",
              "2            [0.151133]              [0.178196]  None   \n",
              "3            [0.547875]              [0.597389]  None   \n",
              "4            [0.133296]              [0.220891]  None   \n",
              "...                 ...                     ...   ...   \n",
              "102512             None                    None  None   \n",
              "102513             None                    None  None   \n",
              "102514             None                    None  None   \n",
              "102515      [0.0148148]                   [0.0]  None   \n",
              "102516            [1.0]                   [1.0]  None   \n",
              "\n",
              "                                                      CSQ SAMPLE  \n",
              "0       [T|downstream_gene_variant|MODIFIER|DDX11L1|EN...    0/1  \n",
              "1       [A|downstream_gene_variant|MODIFIER|DDX11L1|EN...    0/1  \n",
              "2       [C|downstream_gene_variant|MODIFIER|DDX11L1|EN...    0/1  \n",
              "3       [A|non_coding_transcript_exon_variant|MODIFIER...    0/1  \n",
              "4       [A|downstream_gene_variant|MODIFIER|MIR6859-2|...    0/1  \n",
              "...                                                   ...    ...  \n",
              "102512  [C|downstream_gene_variant|MODIFIER|CTBP2P1|EN...    0/1  \n",
              "102513  [G|downstream_gene_variant|MODIFIER|CTBP2P1|EN...    0/1  \n",
              "102514  [G|downstream_gene_variant|MODIFIER|CTBP2P1|EN...    0/1  \n",
              "102515  [A|downstream_gene_variant|MODIFIER|CTBP2P1|EN...    0/1  \n",
              "102516  [C|downstream_gene_variant|MODIFIER|CTBP2P1|EN...    0/1  \n",
              "\n",
              "[102517 rows x 63 columns]"
            ]
          },
          "execution_count": 8,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "def convert_to_csv(vcf_file):\n",
        "    data = []\n",
        "\n",
        "    with open(vcf_file, 'rb') as vcf_file_binary:\n",
        "        vcf_reader = vcf.Reader(vcf_file_binary)\n",
        "        column_headers = [\"CHROM\", \"POS\", \"ID\", \"REF\", \"ALT\", \"QUAL\", \"FILTER\", \"FORMAT\"]\n",
        "\n",
        "        column_headers.extend(vcf_reader.infos.keys())\n",
        "        column_headers.append(\"SAMPLE\")\n",
        "\n",
        "        for record in vcf_reader:\n",
        "            row = []\n",
        "            row.extend([record.CHROM, record.POS, record.ID, record.REF, record.ALT, record.QUAL, record.FILTER, record.FORMAT])\n",
        "\n",
        "            for field in vcf_reader.infos.keys():\n",
        "                info_value = record.INFO.get(field)\n",
        "                row.append(info_value)\n",
        "\n",
        "            for sample in record.samples:\n",
        "                row.append(sample['GT'])\n",
        "\n",
        "            data.append(row)\n",
        "\n",
        "    df = pd.DataFrame(data, columns=column_headers)\n",
        "    return df"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "C3KLa29mnxuL"
      },
      "outputs": [],
      "source": [
        "EE_015=convert_to_csv(file1)\n",
        "EE_015.to_csv('/content/drive/MyDrive/Pathogenicity_Data/EE_015.csv', index=False)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "2bxn-U8QnybO"
      },
      "outputs": [],
      "source": [
        "EE_050=convert_to_csv(file2)\n",
        "EE_050.to_csv('/content/drive/MyDrive/Pathogenicity_Data/EE_050.csv', index=False)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {
        "id": "zVZiwbTDcims"
      },
      "outputs": [],
      "source": [
        "EE_069=convert_to_csv(file3)\n",
        "EE_069.to_csv('/content/drive/MyDrive/Pathogenicity_Data/EE_069.csv', index=False)"
      ]
    }
  ],
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.11.6"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}
